<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8" /> 
<meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover"> 
<title>Groovy NGS: Working With Variants</title>
<link rel="stylesheet" type="text/css" href="/groovy-ngs-utils/userguide/static/css/katex.min.css">
<link rel="stylesheet" type="text/css" href="/groovy-ngs-utils/userguide/static/main.css">
<link rel="stylesheet" type="text/css" href="/groovy-ngs-utils/userguide/static/css/global-overrides.css">
</head>
<link rel="shortcut icon" href="/groovy-ngs-utils/userguide/favicon.png"type="image/ico"/>
<body class="theme-znai-dark">
<script>(function() {
    var themeNameKey = 'znaiTheme';
    var darkThemeName = 'znai-dark';
    var lightThemeName = 'default';

    var znaiTheme = {
        changeHandlers: [],
        addChangeHandler(handler) {
            this.changeHandlers.push(handler);
        },
        removeChangeHandler(handler) {
            var idx = this.changeHandlers.indexOf(handler);
            this.changeHandlers.splice(idx, 1);
        },
        set(name) {
            this.name = name;
            document.body.className = 'theme-' + name;

            var idx = 0;
            var len = this.changeHandlers.length;
            for (; idx < len; idx++) {
                this.changeHandlers[idx](name);
            }
        },
        setExplicitly(name) {
            storeThemeName(name);
            this.set(name);
        },
        setExplicitlyIfNotSetAlready(name) {
            const themeName = getStoredThemeName();
            if (themeName) {
                return
            }

            this.setExplicitly(name)
        },
        toggle() {
            this.setExplicitly(this.name === lightThemeName ? darkThemeName : lightThemeName)
        }
    };

    var mediaThemeName = setLightMatchMediaListenerAndGetThemeName()
    var themeName = getStoredThemeName() || mediaThemeName;
    znaiTheme.set(themeName);

    window.znaiTheme = znaiTheme;

    function getStoredThemeName() {
        return localStorage.getItem(themeNameKey);
    }

    function storeThemeName(name) {
        return localStorage.setItem(themeNameKey, name);
    }

    function setLightMatchMediaListenerAndGetThemeName() {
        if (!window.matchMedia) {
            return darkThemeName;
        }

        var lightQuery = window.matchMedia('(prefers-color-scheme: light)');
        lightQuery.addListener(function (e) {
            const newThemeName = e.matches ? lightThemeName : darkThemeName;
            znaiTheme.setExplicitly(newThemeName);
        });

        return lightQuery.matches ? lightThemeName : darkThemeName;
    }
})()</script>
<div id="znai"><div id="znai-initial-page-loading" style="margin: -20px 0 0 -20px; padding: 0 40px 40px 0; width: 100vw; height: 100vh; display: flex; justify-content: center; align-items: center">
    <div></div>
</div><section style="max-width: 640px; margin-left: auto; margin-right: auto;">
<article>
<header><h1>General Notes</h1></header>
<p>Variants are one of the core focuses of many genomic analysis, so Groovy NGS goes to significant effort to streamline and optimise access to variant information. This is one of the exceptions where, even though HTSJDK does provide a https://samtools.github.io/htsjdk/javadoc/htsjdk/htsjdk/variant/variantcontext/VariantContext.html VariantContext class, Groovy NGS provides a different, substitute class as a complementary option. This is because HTSJDK focuses on completely representing all features from the VCF specification and also ensuring high levels of safety and robustness in how it is used. Groovy NGS, therefore, provides an alternative that is highly ergonomic, but makes some important simplifiying assumptions and only presents a commonly used subset of the VCF specification.The most important simplifying assumption is that many operations, there is a &quot;simple&quot; version that returns results for the first allele and / or the first sample in a VCF. This assumption means that many analyses where these assumptions hold true can be expressed with vastly simpler code and especially for interactive analyses (such as in Jupyter Notebooks), this adds a great level of convenience.Warning: you may miss variants when using default methods with multi-allelic variants. In general, you should apply Groovy NGS classes to VCFs that have been &quot;normalised&quot; and reduced to primitives, for example, by using https://samtools.github.io/bcftools/bcftools.html#norm bcftools norm.</p>
</article>

<article>
<header><h1>Variant Properties</h1></header>
<p>The key class for working with variants in Groovy NGS is the https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/Variant.html Variant class. Conceptually, this class represents a single line in a VCF file, and hence it can include multiple alleles and multiple samples at a single site. Alleles The base sequence of default alternate allele is accessible as the alt property. This value is exactly as presented in the first element of the ALT column from the VCF file. Therefore, for indels it includes the context base that is generally present for those variant types. The full list of alternate allele bases is accessible via the alts property.These properties tell you only the base sequence of the allele. For more information, you can access a full list of Allele objects as the alleles property. These allow you to directly query the type of the variant, start and end position and other attributes. Dosage GNGS refers to the number of copies of a variant that are present as the dosage . Therefore, in a diploid genome, having genotype 1/1 will yield a dosage of 2 for the alternate allele, while having genotype 1/0 will give dosage of 1. This generally makes calculations more straightforward than dealing with the exact genotype, especially when accessing the dosage for the first sample and allele (directly as the dosage property).To see the dosage of the default allele for every sample, you can query the dosages property. For example assuming the proband is the first sample in a trio VCF, we can identify de novo variants as follows: groovy vcf.grep { it.dosages[0] &gt; 0 &amp;&amp; it.dosages[1]==0 &amp;&amp; it.dosages[2] == 0 } Depth The total read depth at the site of a variant can be queried using the totalDepth property. To get the depth of the first allele, you may query the altDepth property. To see the alternate allele depths for other alleles, you can call getAlleleDepths(alleleIndex) where providing 0 specifies the reference allele.Often we are interested in the proportion of reads supporting a variant (refererred to as the variant allele frequency . This is returned as the vaf property for the default allele and sample. To query it for arbitrary alelel and sample, call the https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/Variant.html#getVaf(int,%20int) getVaf(alleleIndex, sampleIndex) method.</p>
</article>

<article>
<header><h1>Variants as Regions</h1></header>
<p>Like other entities that belong to a specific location in the genome, Variants implement the IRegion interface which allows variants to be passed to many functions for region based operations as a parameter.For example, to test if a particular variant lies within a region of interest: Input groovy roi = new Region(&quot;chr22:16591390-16591800&quot;) println(&quot;Does variant $v overlap $roi? : &quot; + roi.overlaps(v)) Output text Does variant chr22:16591593 A/G overlap chr22:16591390-16591800? : true</p>
</article>

<article>
<header><h1>Querying VCFs for Presence of Variants</h1></header>
<p>Finding out whether a particular variant is contained in a VCF is easy because the VCF class implements the Groovy https://groovy-lang.org/operators.html#_membership_operator membership operator which enables use of the in keyword. Hence we can query if a VCF contains a particular variant by using code such as: groovy vcf = VCF.parse('test.vcf') variant = vcf[0] if(variant in vcf) println &quot;Variant $variant is in the VCF!&quot; Note: in testing whether a variant is &quot;in&quot; a VCF, the position and allele are compared. However, the dosage (zygosity) are not compared. Therefore, querying if a homozygous variant is present in a VCF that contains the VCF in heterozygous form will return true . In fact, even if the variant is present but genotyped as homozygous reference, the query will return true . To ascertain whether any sample has non-zero dosage for the variant, use the https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/VCF.html#find(gngs.Variant) find to retrieve the variant and then directly query the dosages. For example: vcf.find(v).dosages.any { it &gt; 0 } Warning: this function may behave in unexpected ways when multi-allelic variants are used as query or target VCF. The find and in functions will match a variant in a VCF if any allele matches any allele in the query.</p>
</article>

<article>
<header><h1>Info Fields</h1></header>
<p>Info fields for each https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/Variant.html Variant are parsed and accessible as a Map keyed on the name. Values are represented as String s and are not further parsed into typed entities matching the INFO fields declared in the VCF header.</p>
</article>

<article>
<header><h1>Annotated Variants</h1></header>
<p>The https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/Variant.html Variant class has built in support for parsing https://asia.ensembl.org/info/docs/tools/vep/index.html VEP annotations. These are accessed via the vepInfo property. As VEP can create multiple annotations, the result is returned as a List , each entry which is a Map&lt;String,Object&gt; of VEP annotations.</p>
</article>

<article>
<header><h1>Updating Variant Attributes</h1></header>
<p>GNGS contains limited support for updating variants, allowing a variant that has been read from a VCF file to be modified and then written out with the changes intact. Due to how Variant entries are stored, it is critical that when updating a variant the update is performed within a update closure: groovy v.update { v.info.MYINFO = &quot;a new INFO property&quot; } When updated this way, changes to INFO and genotype fields will be written out if made as part of a filter operation or the VCF is written using the https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/VCF.html#print(java.lang.Appendable) print method.</p>
</article>
</section>
</div>
<script type="text/javascript" src="/groovy-ngs-utils/userguide/footer.js"></script>
<script type="text/javascript" src="/groovy-ngs-utils/userguide/toc.js"></script>
<script type="text/javascript" src="/groovy-ngs-utils/userguide/assets.js"></script>
<script type="text/javascript" src="/groovy-ngs-utils/userguide/static/main.js"></script>
<script type="text/javascript" src="/groovy-ngs-utils/userguide/search-index.js"></script>
<script>
document.getElementById('znai').innerHTML = '';
/*<!--*/
ReactDOM.render(React.createElement(Documentation, {
  "docMeta" : {
    "viewOn" : {
      "link" : "https://github.com/ssadedin/groovy-ngs-utils",
      "title" : "Groovy NGS"
    },
    "id" : "groovy-ngs-utils/userguide",
    "title" : "Groovy NGS",
    "type" : "Guide",
    "previewEnabled" : false
  },
  "page" : {
    "type" : "Page",
    "content" : [ {
      "id" : "general-notes",
      "title" : "General Notes",
      "type" : "Section",
      "content" : [ {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "Variants are one of the core focuses of many genomic analysis, so Groovy NGS goes to significant effort to",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "streamline and optimise access to variant information. This is one of the exceptions where,",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "even though HTSJDK does provide a ",
          "type" : "SimpleText"
        }, {
          "url" : "https://samtools.github.io/htsjdk/javadoc/htsjdk/htsjdk/variant/variantcontext/VariantContext.html",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "VariantContext",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " class,",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "Groovy NGS provides a different, substitute class as a complementary option. This is because  HTSJDK focuses on completely",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "representing all features from the VCF specification and also ensuring high levels of safety and robustness in how it is used.",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "Groovy NGS, therefore, provides an alternative that is highly ergonomic, but makes some important simplifiying assumptions and",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "only presents a commonly used subset of the VCF specification.",
          "type" : "SimpleText"
        } ]
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "The most important simplifying assumption is that many operations, there is a \"simple\" version that returns results for the first",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "allele and / or the first sample in a VCF. This assumption means that many analyses where these assumptions hold true",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "can be expressed with vastly simpler code and especially for interactive analyses (such as in Jupyter Notebooks), this",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "adds a great level of convenience.",
          "type" : "SimpleText"
        } ]
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "Warning: you may miss variants when using default methods with multi-allelic variants. In general, you should",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "apply Groovy NGS classes to VCFs that have been \"normalised\" and reduced to primitives, for example, by using",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "url" : "https://samtools.github.io/bcftools/bcftools.html#norm",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "bcftools norm",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : ".",
          "type" : "SimpleText"
        } ]
      } ]
    }, {
      "id" : "variant-properties",
      "title" : "Variant Properties",
      "type" : "Section",
      "content" : [ {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "The key class for working with variants in Groovy NGS is the ",
          "type" : "SimpleText"
        }, {
          "url" : "https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/Variant.html",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "Variant",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : "",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "class. Conceptually, this class represents a single line in a VCF file, and hence it can include multiple alleles",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "and multiple samples at a single site.",
          "type" : "SimpleText"
        } ]
      }, {
        "id" : "variant-properties-alleles",
        "level" : 2,
        "title" : "Alleles",
        "type" : "SubHeading"
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "The base sequence of default alternate allele is accessible as the ",
          "type" : "SimpleText"
        }, {
          "code" : "alt",
          "type" : "InlinedCode"
        }, {
          "text" : " property. This value is exactly",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "as presented in the first element of the ALT column from the VCF file. Therefore, for indels it includes",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "the context base that is generally present for those variant types. The full list of alternate allele",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "bases is accessible via the ",
          "type" : "SimpleText"
        }, {
          "code" : "alts",
          "type" : "InlinedCode"
        }, {
          "text" : " property.",
          "type" : "SimpleText"
        } ]
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "These properties tell you only the base sequence of the allele. For more information, you can access",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "a full list of ",
          "type" : "SimpleText"
        }, {
          "code" : "Allele",
          "type" : "InlinedCode"
        }, {
          "text" : " objects as the ",
          "type" : "SimpleText"
        }, {
          "code" : "alleles",
          "type" : "InlinedCode"
        }, {
          "text" : " property. These allow you to directly query",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "the type of the variant, start and end position and other attributes.",
          "type" : "SimpleText"
        } ]
      }, {
        "id" : "variant-properties-dosage",
        "level" : 2,
        "title" : "Dosage",
        "type" : "SubHeading"
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "GNGS refers to the number of copies of a variant that are present as the ",
          "type" : "SimpleText"
        }, {
          "code" : "dosage",
          "type" : "InlinedCode"
        }, {
          "text" : ". Therefore, in a",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "diploid genome, having genotype ",
          "type" : "SimpleText"
        }, {
          "code" : "1/1",
          "type" : "InlinedCode"
        }, {
          "text" : " will yield a dosage of ",
          "type" : "SimpleText"
        }, {
          "code" : "2",
          "type" : "InlinedCode"
        }, {
          "text" : " for the alternate allele, while",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "having genotype ",
          "type" : "SimpleText"
        }, {
          "code" : "1/0",
          "type" : "InlinedCode"
        }, {
          "text" : " will give dosage of 1. This generally makes calculations more straightforward",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "than dealing with the exact genotype, especially when accessing the dosage for the first sample and",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "allele (directly as the ",
          "type" : "SimpleText"
        }, {
          "code" : "dosage",
          "type" : "InlinedCode"
        }, {
          "text" : " property).",
          "type" : "SimpleText"
        } ]
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "To see the dosage of the default allele for every sample, you can query the ",
          "type" : "SimpleText"
        }, {
          "code" : "dosages",
          "type" : "InlinedCode"
        }, {
          "text" : " property.",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "For example assuming the proband is the first sample in a trio VCF, we can identify de novo variants",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "as follows:",
          "type" : "SimpleText"
        } ]
      }, {
        "lang" : "groovy",
        "snippet" : "vcf.grep {\n    it.dosages[0] > 0 && it.dosages[1]==0 && it.dosages[2] == 0\n}\n",
        "lineNumber" : "",
        "type" : "Snippet"
      }, {
        "id" : "variant-properties-depth",
        "level" : 2,
        "title" : "Depth",
        "type" : "SubHeading"
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "The total read depth at the site of a variant can be queried using the ",
          "type" : "SimpleText"
        }, {
          "code" : "totalDepth",
          "type" : "InlinedCode"
        }, {
          "text" : " property. To",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "get the depth of the first allele, you may query the ",
          "type" : "SimpleText"
        }, {
          "code" : "altDepth",
          "type" : "InlinedCode"
        }, {
          "text" : " property. To see the alternate",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "allele depths for other alleles,  you can call ",
          "type" : "SimpleText"
        }, {
          "code" : "getAlleleDepths(alleleIndex)",
          "type" : "InlinedCode"
        }, {
          "text" : " where providing",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "0 specifies the reference allele.",
          "type" : "SimpleText"
        } ]
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "Often we are interested in the proportion of reads supporting a variant (refererred to as",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "the ",
          "type" : "SimpleText"
        }, {
          "code" : "variant allele frequency",
          "type" : "InlinedCode"
        }, {
          "text" : ". This is returned as the ",
          "type" : "SimpleText"
        }, {
          "code" : "vaf",
          "type" : "InlinedCode"
        }, {
          "text" : " property for the default",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "allele and sample. To query it for arbitrary alelel and sample, call the",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "url" : "https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/Variant.html#getVaf(int,%20int)",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "getVaf(alleleIndex, sampleIndex)",
            "type" : "SimpleText"
          } ]
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "method.",
          "type" : "SimpleText"
        } ]
      } ]
    }, {
      "id" : "variants-as-regions",
      "title" : "Variants as Regions",
      "type" : "Section",
      "content" : [ {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "Like other entities that belong to a specific location in the genome, Variants implement the",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "code" : "IRegion",
          "type" : "InlinedCode"
        }, {
          "text" : " interface which allows variants to be passed to many functions for region based",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "operations as a parameter.",
          "type" : "SimpleText"
        } ]
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "For example, to test if a particular variant lies within a region of interest:",
          "type" : "SimpleText"
        } ]
      }, {
        "columns" : [ {
          "content" : [ {
            "type" : "Paragraph",
            "content" : [ {
              "type" : "StrongEmphasis",
              "content" : [ {
                "text" : "Input",
                "type" : "SimpleText"
              } ]
            } ]
          }, {
            "lang" : "groovy",
            "snippet" : "roi = new Region(\"chr22:16591390-16591800\")\nprintln(\"Does variant $v overlap $roi? : \" + roi.overlaps(v))\n",
            "lineNumber" : "",
            "type" : "Snippet"
          } ]
        }, {
          "content" : [ {
            "type" : "Paragraph",
            "content" : [ {
              "type" : "StrongEmphasis",
              "content" : [ {
                "text" : "Output",
                "type" : "SimpleText"
              } ]
            } ]
          }, {
            "lang" : "text",
            "snippet" : "Does variant chr22:16591593 A/G overlap chr22:16591390-16591800? : true\n",
            "lineNumber" : "",
            "type" : "Snippet"
          } ]
        } ],
        "config" : { },
        "type" : "Columns"
      } ]
    }, {
      "id" : "querying-vcfs-for-presence-of-variants",
      "title" : "Querying VCFs for Presence of Variants",
      "type" : "Section",
      "content" : [ {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "Finding out whether a particular variant is contained in a VCF is easy because the ",
          "type" : "SimpleText"
        }, {
          "code" : "VCF",
          "type" : "InlinedCode"
        }, {
          "text" : " class implements",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "the Groovy ",
          "type" : "SimpleText"
        }, {
          "url" : "https://groovy-lang.org/operators.html#_membership_operator",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "membership operator",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " which enables",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "use of the ",
          "type" : "SimpleText"
        }, {
          "code" : "in",
          "type" : "InlinedCode"
        }, {
          "text" : " keyword. Hence we can query if a VCF contains a particular variant by using code such as:",
          "type" : "SimpleText"
        } ]
      }, {
        "lang" : "groovy",
        "snippet" : "vcf = VCF.parse('test.vcf')\nvariant = vcf[0]\nif(variant in vcf)\n    println \"Variant $variant is in the VCF!\"\n",
        "lineNumber" : "",
        "type" : "Snippet"
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "Note: in testing whether a variant is \"in\" a VCF, the position and allele are compared. However, the dosage (zygosity)",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "are not compared. Therefore, querying if a homozygous variant is present in a VCF that contains the VCF in heterozygous form",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "will return ",
          "type" : "SimpleText"
        }, {
          "code" : "true",
          "type" : "InlinedCode"
        }, {
          "text" : ". In fact, even if the variant is present but genotyped as homozygous reference, the query will return ",
          "type" : "SimpleText"
        }, {
          "code" : "true",
          "type" : "InlinedCode"
        }, {
          "text" : ".",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "To ascertain whether any sample has non-zero dosage for the variant, use the ",
          "type" : "SimpleText"
        }, {
          "url" : "https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/VCF.html#find(gngs.Variant)",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "find",
            "type" : "SimpleText"
          } ]
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "to retrieve the variant and then directly query the dosages. For example: ",
          "type" : "SimpleText"
        }, {
          "code" : "vcf.find(v).dosages.any { it > 0 }",
          "type" : "InlinedCode"
        } ]
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "Warning: this function may behave in unexpected ways when multi-allelic variants are used as query or target VCF. The",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "code" : "find",
          "type" : "InlinedCode"
        }, {
          "text" : "  and ",
          "type" : "SimpleText"
        }, {
          "code" : "in",
          "type" : "InlinedCode"
        }, {
          "text" : " functions will match a variant in a VCF if ",
          "type" : "SimpleText"
        }, {
          "type" : "Emphasis",
          "content" : [ {
            "text" : "any",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " allele matches ",
          "type" : "SimpleText"
        }, {
          "type" : "Emphasis",
          "content" : [ {
            "text" : "any",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " allele in the query.",
          "type" : "SimpleText"
        } ]
      } ]
    }, {
      "id" : "info-fields",
      "title" : "Info Fields",
      "type" : "Section",
      "content" : [ {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "Info fields for each ",
          "type" : "SimpleText"
        }, {
          "url" : "https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/Variant.html",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "Variant",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " are parsed and accessible as",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "a ",
          "type" : "SimpleText"
        }, {
          "code" : "Map",
          "type" : "InlinedCode"
        }, {
          "text" : " keyed on the name. Values are represented as ",
          "type" : "SimpleText"
        }, {
          "code" : "String",
          "type" : "InlinedCode"
        }, {
          "text" : "s and are ",
          "type" : "SimpleText"
        }, {
          "type" : "Emphasis",
          "content" : [ {
            "text" : "not",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " further parsed into typed entities matching the ",
          "type" : "SimpleText"
        }, {
          "code" : "INFO",
          "type" : "InlinedCode"
        }, {
          "text" : " fields declared in the",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "VCF header.",
          "type" : "SimpleText"
        } ]
      } ]
    }, {
      "id" : "annotated-variants",
      "title" : "Annotated Variants",
      "type" : "Section",
      "content" : [ {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "The ",
          "type" : "SimpleText"
        }, {
          "url" : "https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/Variant.html",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "Variant",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " class has built in support for parsing",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "url" : "https://asia.ensembl.org/info/docs/tools/vep/index.html",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "VEP",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " annotations. These are accessed via the",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "code" : "vepInfo",
          "type" : "InlinedCode"
        }, {
          "text" : " property. As VEP can create multiple annotations, the result is returned as a ",
          "type" : "SimpleText"
        }, {
          "code" : "List",
          "type" : "InlinedCode"
        }, {
          "text" : ", each entry which is",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "a ",
          "type" : "SimpleText"
        }, {
          "code" : "Map<String,Object>",
          "type" : "InlinedCode"
        }, {
          "text" : " of VEP annotations.",
          "type" : "SimpleText"
        } ]
      } ]
    }, {
      "id" : "updating-variant-attributes",
      "title" : "Updating Variant Attributes",
      "type" : "Section",
      "content" : [ {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "GNGS contains limited support for updating variants, allowing a variant that has been read from a VCF file to be modified",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "and then written out with the changes intact. Due to how ",
          "type" : "SimpleText"
        }, {
          "code" : "Variant",
          "type" : "InlinedCode"
        }, {
          "text" : " entries are stored, it is critical that when",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "updating a variant the update is performed within a ",
          "type" : "SimpleText"
        }, {
          "code" : "update",
          "type" : "InlinedCode"
        }, {
          "text" : " closure:",
          "type" : "SimpleText"
        } ]
      }, {
        "lang" : "groovy",
        "snippet" : "v.update {\n    v.info.MYINFO = \"a new INFO property\"\n}\n",
        "lineNumber" : "",
        "title" : "Updating a Variant to add an INFO property",
        "anchorId" : "updating-variant-attributes-updating-a-variant-to-add-an-info-property",
        "type" : "Snippet"
      }, {
        "type" : "Paragraph",
        "content" : [ {
          "text" : "When updated this way, changes to INFO and genotype fields will be written out if made as part of a ",
          "type" : "SimpleText"
        }, {
          "code" : "filter",
          "type" : "InlinedCode"
        }, {
          "text" : " operation",
          "type" : "SimpleText"
        }, {
          "type" : "SoftLineBreak"
        }, {
          "text" : "or the VCF is written using the ",
          "type" : "SimpleText"
        }, {
          "url" : "https://ssadedin.github.io/groovy-ngs-utils/doc/gngs/VCF.html#print(java.lang.Appendable)",
          "isFile" : false,
          "type" : "Link",
          "content" : [ {
            "text" : "print",
            "type" : "SimpleText"
          } ]
        }, {
          "text" : " method.",
          "type" : "SimpleText"
        } ]
      } ]
    } ],
    "lastModifiedTime" : 1754207802092,
    "tocItem" : {
      "chapterTitle" : "Chapter 3 Vcf Files",
      "pageTitle" : "Working With Variants",
      "pageMeta" : { },
      "dirName" : "chapter-3-vcf-files",
      "fileName" : "working-with-variants",
      "fileExtension" : "md",
      "viewOnRelativePath" : null,
      "pageSectionIdTitles" : [ {
        "title" : "General Notes",
        "id" : "general-notes"
      }, {
        "title" : "Variant Properties",
        "id" : "variant-properties"
      }, {
        "title" : "Variants as Regions",
        "id" : "variants-as-regions"
      }, {
        "title" : "Querying VCFs for Presence of Variants",
        "id" : "querying-vcfs-for-presence-of-variants"
      }, {
        "title" : "Info Fields",
        "id" : "info-fields"
      }, {
        "title" : "Annotated Variants",
        "id" : "annotated-variants"
      }, {
        "title" : "Updating Variant Attributes",
        "id" : "updating-variant-attributes"
      } ]
    }
  }
}), document.getElementById("znai"));
/*-->*/

</script>

</body>
</html>
