<!--

     Licensed to the Apache Software Foundation (ASF) under one
     or more contributor license agreements.  See the NOTICE file
     distributed with this work for additional information
     regarding copyright ownership.  The ASF licenses this file
     to you under the Apache License, Version 2.0 (the
     "License"); you may not use this file except in compliance
     with the License.  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

     Unless required by applicable law or agreed to in writing,
     software distributed under the License is distributed on an
     "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
     KIND, either express or implied.  See the License for the
     specific language governing permissions and limitations
     under the License.

-->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>

<title>gngs (groovy-ngs-utils API)</title>
<meta name="keywords" content="gngs package">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<link rel ="stylesheet" TYPE="text/css" HREF="../stylesheet.css" title="Style">
<link href="../groovy.ico" type="image/x-icon" rel="shortcut icon">
<link href="../groovy.ico" type="image/x-icon" rel="icon">
<script type="text/javascript">
function windowTitle()
{
    parent.document.title="gngs (groovy-ngs-utils API)";
}
</script>
<noscript>
</noscript>

</head>

<body class="center" onload="windowTitle();">

<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
    <!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
    <!--   -->
</a>
<a name="navbar_top_firstrow"><!-- --></a>
<ul class="navList" title="Navigation">
    <li><a href="../overview-summary.html">Overview</a></li>
    <li class="navBarCell1Rev">Package</li>
    <li>Class</li>
    <li><a href="../deprecated-list.html">Deprecated</a></li>
    <li><a href="../index-all.html">Index</a></li>
    <li><a href="../help-doc.html">Help</a></li>
</ul>
</div>

<div class="subNav">
    <ul class="navList">
        <li><a href="../index.html?gngs/package-summary.html" target="_top">Frames</a></li>
        <li><a href="package-summary.html" target="_top">No Frames</a></li>
    </ul>
</div>

<!-- ========= END OF TOP NAVBAR ========= -->

<div class="header">
    <h1 class="title">Package gngs</h1>
</div>


<h2 title="Package gngs Description" class="title">Package gngs Description</h2>
<div class="header"> All the core classes making up Groovy NGS Utils
 
</div>


<div class="contentContainer">


    <div class="summary">
        <ul class="blockList">
            <li class="blockList">
                <table border="0" cellpadding="3" cellspacing="0" summary="Interface Summary">
                    <caption><span>Interface Summary</span><span class="tabEnd">&nbsp;</span></caption>
                    <tbody>
                    <tr>
                        <th class="colFirst" scope="col">Interface</th>
                        <th class="colLast" scope="col">Description</th>
                    </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CompactReadPair.ReadCompressor.html" title="class in gngs">
                                    CompactReadPair.ReadCompressor
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Constraint.html" title="class in gngs">
                                    Constraint
                                </a></strong>
                            </td>
                            <td>Represents a constraint in a genetic model</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="IRegion.html" title="class in gngs">
                                    IRegion
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ReadPair.html" title="class in gngs">
                                    ReadPair
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>

                    </tbody>
                </table>
            </li>
            </ul>
        </div>
    



    <div class="summary">
        <ul class="blockList">
            <li class="blockList">
                <table border="0" cellpadding="3" cellspacing="0" summary="Class Summary">
                    <caption><span>Class Summary</span><span class="tabEnd">&nbsp;</span></caption>
                    <tbody>
                    <tr>
                        <th class="colFirst" scope="col">Class</th>
                        <th class="colLast" scope="col">Description</th>
                    </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="AcknowledgeableMessage.html" title="class in gngs">
                                    AcknowledgeableMessage
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Align.html" title="class in gngs">
                                    Align
                                </a></strong>
                            </td>
                            <td>Utility for helping to perform alignments with BioJava</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="AltContig.html" title="class in gngs">
                                    AltContig
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="AltContigIndex.html" title="class in gngs">
                                    AltContigIndex
                                </a></strong>
                            </td>
                            <td>Parses an alt-index in SAM format as provided with GRCh38 to define the alt contigs,
 including extracting the regions in the primary assembly covered by each alt contig</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="AsciiWriter.html" title="class in gngs">
                                    AsciiWriter
                                </a></strong>
                            </td>
                            <td>Copied from HTSJDK but with constructor modified so that the buffer size can be customized</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="AsyncMateResolveWorker.html" title="class in gngs">
                                    AsyncMateResolveWorker
                                </a></strong>
                            </td>
                            <td>A background task that can read from a queue of unpaired reads and resolve their
 pairs in the background, as well as resolve other reads that are in the same
 region as the pair that is resolved.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="BED.html" title="class in gngs">
                                    BED
                                </a></strong>
                            </td>
                            <td>Support for parsing, filtering, and manipuulating BED files.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="BaseCompactor.html" title="class in gngs">
                                    BaseCompactor
                                </a></strong>
                            </td>
                            <td>A utility to compact bases to a two-per-byte representation</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="BaseCounts.html" title="class in gngs">
                                    BaseCounts
                                </a></strong>
                            </td>
                            <td>Stores a set of base counts for a base read at a given position,
 and supports scoring metrics.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="BatchedAcknowledgeableMessage.html" title="class in gngs">
                                    BatchedAcknowledgeableMessage
                                </a></strong>
                            </td>
                            <td>A batched message automatically accumulates a fixed number of messagse before sending 
 on to a <a href='../gngs/RegulatingActor.html' title='RegulatingActor'>RegulatingActor</a>.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CNVDatabase.html" title="class in gngs">
                                    CNVDatabase
                                </a></strong>
                            </td>
                            <td>Abstract interface that can be implemented by different
 CNV sources.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CNVFrequency.html" title="class in gngs">
                                    CNVFrequency
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Cli.html" title="class in gngs">
                                    Cli
                                </a></strong>
                            </td>
                            <td>This class is a simple utility that configures a CliBuilder how I like it and adds a few
 simple options that I like.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CliOptions.html" title="class in gngs">
                                    CliOptions
                                </a></strong>
                            </td>
                            <td>Wraps a standard OptionAccessor but allows for values to be overridden, enabling
 classes to be used more generically since OptionAccessor is tied closely to 
 command line.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CompactReadPair.html" title="class in gngs">
                                    CompactReadPair
                                </a></strong>
                            </td>
                            <td>Stores one read of a pair and links to the second read in a way that offers
 a useful amount of compression for applications that need to store many
 reads in memory.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Consensus.html" title="class in gngs">
                                    Consensus
                                </a></strong>
                            </td>
                            <td>Support to determine a consensus sequence from a set of sequences</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CoverageBlock.html" title="class in gngs">
                                    CoverageBlock
                                </a></strong>
                            </td>
                            <td>A run of contiguous bases within a band of coverage depth
 (eg: a run of low coverage bases).</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CoverageGaps.html" title="class in gngs">
                                    CoverageGaps
                                </a></strong>
                            </td>
                            <td>Discovers blocks of low coverage in a file output in BEDTools Coverage format,
 or alternatively <a href='../gngs/tools/MultiCov.html' title='MultiCov'>MultiCov</a> output.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CoverageStats.html" title="class in gngs">
                                    CoverageStats
                                </a></strong>
                            </td>
                            <td>An efficient method to calculate percentiles of coverage values
 that doesn't require holding them all in memory or sorting them.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="DGV.html" title="class in gngs">
                                    DGV
                                </a></strong>
                            </td>
                            <td>A <a href='../gngs/CNVDatabase.html' title='CNVDatabase'>CNVDatabase</a> implementation for CNVs from the Database of
 Genomic Variants (sourced via UCSC).</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="DecipherCNVs.html" title="class in gngs">
                                    DecipherCNVs
                                </a></strong>
                            </td>
                            <td>A <a href='../gngs/CNVDatabase.html' title='CNVDatabase'>CNVDatabase</a> implementation for CNVs from Decipher 
 Developmental Delay project.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Docs.html" title="class in gngs">
                                    Docs
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ExecutedProcess.html" title="class in gngs">
                                    ExecutedProcess
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="FASTA.html" title="class in gngs">
                                    FASTA
                                </a></strong>
                            </td>
                            <td>Simple utilities for manipulating FASTA files.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="FASTAIndex.html" title="class in gngs">
                                    FASTAIndex
                                </a></strong>
                            </td>
                            <td>A very simplistic FASTA index that allows lookup of sequence names by their
 sequence content using a fixed length prefix seed. </td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="FASTHRead.html" title="class in gngs">
                                    FASTHRead
                                </a></strong>
                            </td>
                            <td>Models a single FASTH read, including the header (with name), barcodes, bases and 
 quality information.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="FASTQ.html" title="class in gngs">
                                    FASTQ
                                </a></strong>
                            </td>
                            <td>Simple support for reading fastq files.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="FASTQRead.html" title="class in gngs">
                                    FASTQRead
                                </a></strong>
                            </td>
                            <td>Models a single FASTQ read, including the header (with name), bases and 
 quality information.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="FormatMetaData.html" title="class in gngs">
                                    FormatMetaData
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="GATKIntervalSummary.html" title="class in gngs">
                                    GATKIntervalSummary
                                </a></strong>
                            </td>
                            <td>Reads output of GATK DepthOfCoverage interval summary and
 makes it available as a Regions object</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="GATKMeans.html" title="class in gngs">
                                    GATKMeans
                                </a></strong>
                            </td>
                            <td>Estimates sample means by reading interval summary file output by GATK</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="GRange.html" title="class in gngs">
                                    GRange
                                </a></strong>
                            </td>
                            <td>Range over an interval</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="GRegion.html" title="class in gngs">
                                    GRegion
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="GenomeResource.html" title="class in gngs">
                                    GenomeResource
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="GnomADCNVDatabase.html" title="class in gngs">
                                    GnomADCNVDatabase
                                </a></strong>
                            </td>
                            <td>A <a href='../gngs/CNVDatabase.html' title='CNVDatabase'>CNVDatabase</a> implementation for CNVs from the gnomAD
 sv dataset.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Hash.html" title="class in gngs">
                                    Hash
                                </a></strong>
                            </td>
                            <td>Simple convenience wrapper that calculates the sha1 hash for a class and returns
 it formatted in a standard way.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="IlluminaFileNameParser.html" title="class in gngs">
                                    IlluminaFileNameParser
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="IntervalGCCache.html" title="class in gngs">
                                    IntervalGCCache
                                </a></strong>
                            </td>
                            <td>Efficiently supports computing and caching approximate GC content
 for random fragments of DNA in a genome.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="IntervalGCCache.IntervalGCQueryCache.html" title="class in gngs">
                                    IntervalGCCache.IntervalGCQueryCache
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="JavaAdapters.html" title="class in gngs">
                                    JavaAdapters
                                </a></strong>
                            </td>
                            <td>API adapters for calls that do not map cleanly between groovy and java</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="LXPos.html" title="class in gngs">
                                    LXPos
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="LXXPos.html" title="class in gngs">
                                    LXXPos
                                </a></strong>
                            </td>
                            <td>LXXPos models genomic positions including alt and random contigs into a flat 
 space fitting into 63bit unsigned Java long type.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="LiftOver.html" title="class in gngs">
                                    LiftOver
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="MissingMateIndex.html" title="class in gngs">
                                    MissingMateIndex
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="MovingWindowActor.html" title="class in gngs">
                                    MovingWindowActor
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="NumericRegionComparator.html" title="class in gngs">
                                    NumericRegionComparator
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="OMIM.html" title="class in gngs">
                                    OMIM
                                </a></strong>
                            </td>
                            <td>Parser for OMIM genemap files</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="OrderedPairActor.html" title="class in gngs">
                                    OrderedPairActor
                                </a></strong>
                            </td>
                            <td>An actor that adds asynchronous writing to an OrderedPairWriter</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="OrderedPairReader.html" title="class in gngs">
                                    OrderedPairReader
                                </a></strong>
                            </td>
                            <td>A utility class for efficiently iterating all the read pairs in a BAM file.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="OrderedPairWriter.html" title="class in gngs">
                                    OrderedPairWriter
                                </a></strong>
                            </td>
                            <td>When processing reads as pairs it can be tricky to write them out again while maintaining the
 order expected for a "sorted" BAM file. </td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="OverlapTracker.html" title="class in gngs">
                                    OverlapTracker
                                </a></strong>
                            </td>
                            <td>Simple class to track ranges and remove them when they no longer overlap a given 
 position.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Pedigree.html" title="class in gngs">
                                    Pedigree
                                </a></strong>
                            </td>
                            <td>Represents a set of relationships between individuals in a family</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Pedigrees.html" title="class in gngs">
                                    Pedigrees
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Pileup.html" title="class in gngs">
                                    Pileup
                                </a></strong>
                            </td>
                            <td>Overall class containing pileup state</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="PileupIterator.html" title="class in gngs">
                                    PileupIterator
                                </a></strong>
                            </td>
                            <td>Overall class containing pileup state</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="PileupIterator.Pileup.html" title="class in gngs">
                                    PileupIterator.Pileup
                                </a></strong>
                            </td>
                            <td>Inner class that captures pileup state at a specific position</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="PileupState.html" title="class in gngs">
                                    PileupState
                                </a></strong>
                            </td>
                            <td>Additional state about a read that is part of a pileup</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="PrefixTrie.html" title="class in gngs">
                                    PrefixTrie
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ProgressCounter.html" title="class in gngs">
                                    ProgressCounter
                                </a></strong>
                            </td>
                            <td>Simple utility for displaying progress</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="QualCompactor.html" title="class in gngs">
                                    QualCompactor
                                </a></strong>
                            </td>
                            <td>A utility to compact quality scores by run length encoding them</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RangeIndex.html" title="class in gngs">
                                    RangeIndex
                                </a></strong>
                            </td>
                            <td>Indexes a set of ranges so that the ranges crossing any point
 can be identified quickly and easily.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RangedData.html" title="class in gngs">
                                    RangedData
                                </a></strong>
                            </td>
                            <td>RangedData represents a set of genomic regions with data attached. </td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ReadMateLookupActor.html" title="class in gngs">
                                    ReadMateLookupActor
                                </a></strong>
                            </td>
                            <td>An actor that can accept <a href='../gngs/ResolveRequest.html' title='ResolveRequest'>ResolveRequest</a> messages to asynchronously 
 resolve the missing mates of read pairs.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ReadWindow.html" title="class in gngs">
                                    ReadWindow
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RefGenes.html" title="class in gngs">
                                    RefGenes
                                </a></strong>
                            </td>
                            <td>A convenience class that is set up to parse the UCSC RefGene database
 as a RangedData object. </td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Region.html" title="class in gngs">
                                    Region
                                </a></strong>
                            </td>
                            <td>Region of a genome</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RegionComparator.html" title="class in gngs">
                                    RegionComparator
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RegionIterator.html" title="class in gngs">
                                    RegionIterator
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RegionScan.html" title="class in gngs">
                                    RegionScan
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Regions.html" title="class in gngs">
                                    Regions
                                </a></strong>
                            </td>
                            <td>A set of genomic regions, each indexed by chromosome, start and end position.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RegulatingActor.html" title="class in gngs">
                                    RegulatingActor
                                </a></strong>
                            </td>
                            <td>An Actor that exerts back-pressure to regulate the rate of incoming messages.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Relationship.html" title="class in gngs">
                                    Relationship
                                </a></strong>
                            </td>
                            <td>Models a relationship between two people</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RepeatMotif.html" title="class in gngs">
                                    RepeatMotif
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ResolveRequest.html" title="class in gngs">
                                    ResolveRequest
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ResourceDownloader.html" title="class in gngs">
                                    ResourceDownloader
                                </a></strong>
                            </td>
                            <td>Support for downloading data from URLs like UCSC with built-in 
 caching and genome version detection.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SAM.html" title="class in gngs">
                                    SAM
                                </a></strong>
                            </td>
                            <td>Adds various Groovy idioms and convenience features to the 
 Picard SamReader.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SAMRecordCategory.html" title="class in gngs">
                                    SAMRecordCategory
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SAMRecordPair.html" title="class in gngs">
                                    SAMRecordPair
                                </a></strong>
                            </td>
                            <td>Models a read pair, where one of the reads might be missing.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ScannableRegion.html" title="class in gngs">
                                    ScannableRegion
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SexKaryotyper.html" title="class in gngs">
                                    SexKaryotyper
                                </a></strong>
                            </td>
                            <td>An algorithm based on simple heuristics for estimating the sex of a sample from
 sequencing coverage.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SimpleLogFormatter.html" title="class in gngs">
                                    SimpleLogFormatter
                                </a></strong>
                            </td>
                            <td>The default Java log former uses a format that is too verbose, so
 replace it with something more compact.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SnpEffInfo.html" title="class in gngs">
                                    SnpEffInfo
                                </a></strong>
                            </td>
                            <td>Support for parsing annotations from SnpEff</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Subject.html" title="class in gngs">
                                    Subject
                                </a></strong>
                            </td>
                            <td>An individual in a family pedigree</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="TargetedCNVAnnotator.html" title="class in gngs">
                                    TargetedCNVAnnotator
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ToolBase.html" title="class in gngs">
                                    ToolBase
                                </a></strong>
                            </td>
                            <td>A support class that makes implementing a tool based on GNGS very easy 
 while supporting some standard behaviors (such as logging, etc).</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="TwoLeggedOAuthAPI.html" title="class in gngs">
                                    TwoLeggedOAuthAPI
                                </a></strong>
                            </td>
                            <td>With 2 legged OAuth there is no need for acquiring an access token or request token; these
 are presumed known. </td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Utils.html" title="class in gngs">
                                    Utils
                                </a></strong>
                            </td>
                            <td>Miscellaneous utilities that I couldn't think to put anywhere else</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCF.html" title="class in gngs">
                                    VCF
                                </a></strong>
                            </td>
                            <td>The VCF class supports simple parsing, filtering, querying and updating of
 VCF files.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCFIndex.html" title="class in gngs">
                                    VCFIndex
                                </a></strong>
                            </td>
                            <td>Support for querying indexed VCF files via random access.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCFParseContext.html" title="class in gngs">
                                    VCFParseContext
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCFSiteWalker.html" title="class in gngs">
                                    VCFSiteWalker
                                </a></strong>
                            </td>
                            <td>Traverses a set of VCFs site by site, such that all sites in the VCFs are visited
 one at a time.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCFSummaryStats.html" title="class in gngs">
                                    VCFSummaryStats
                                </a></strong>
                            </td>
                            <td>A set of statistics summarizing how a VCF was filtered</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCFWalkPosition.html" title="class in gngs">
                                    VCFWalkPosition
                                </a></strong>
                            </td>
                            <td>A datastructure to hold thhe state of a given VCF that is being walked</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VEPConsequences.html" title="class in gngs">
                                    VEPConsequences
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Variant.html" title="class in gngs">
                                    Variant
                                </a></strong>
                            </td>
                            <td>Represents the genetic state at a specific locus in a genome
 ie: captures the information at a single line of a VCF file.
 </td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Variant.Allele.html" title="class in gngs">
                                    Variant.Allele
                                </a></strong>
                            </td>
                            <td>A specific allele in the context of a Variant in a VCF file</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VariantInjector.html" title="class in gngs">
                                    VariantInjector
                                </a></strong>
                            </td>
                            <td>Injects a variant into a FASTQ read based on recognising the sequence corresponding to 
 the reference sequence for the location of the variant within the read.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="WSCredentials.html" title="class in gngs">
                                    WSCredentials
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="WebService.html" title="class in gngs">
                                    WebService
                                </a></strong>
                            </td>
                            <td>Base class for APIs accessing Web services based on JSON data</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="XA.html" title="class in gngs">
                                    XA
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="XPos.html" title="class in gngs">
                                    XPos
                                </a></strong>
                            </td>
                            <td>A representation of a genomic position as a long</td>
                        </tr>

                    </tbody>
                </table>
            </li>
            </ul>
        </div>
    


    <div class="summary">
        <ul class="blockList">
            <li class="blockList">
                <table border="0" cellpadding="3" cellspacing="0" summary="Enum Summary">
                    <caption><span>Enum Summary</span><span class="tabEnd">&nbsp;</span></caption>
                    <tbody>
                    <tr>
                        <th class="colFirst" scope="col">Enum</th>
                        <th class="colLast" scope="col">Description</th>
                    </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="IlluminaFileNameParser.DIALECT.html" title="class in gngs">
                                    IlluminaFileNameParser.DIALECT
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="RelationshipType.html" title="class in gngs">
                                    RelationshipType
                                </a></strong>
                            </td>
                            <td>Models a relationship between two individuals in a pedigree</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Sex.html" title="class in gngs">
                                    Sex
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>

                    </tbody>
                </table>
            </li>
            </ul>
        </div>
    


    <div class="summary">
        <ul class="blockList">
            <li class="blockList">
                <table border="0" cellpadding="3" cellspacing="0" summary="Exception Summary">
                    <caption><span>Exception Summary</span><span class="tabEnd">&nbsp;</span></caption>
                    <tbody>
                    <tr>
                        <th class="colFirst" scope="col">Exception</th>
                        <th class="colLast" scope="col">Description</th>
                    </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Abort.html" title="class in gngs">
                                    Abort
                                </a></strong>
                            </td>
                            <td>Convenience for aborting loops invoked via closures</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="StopParsingVCFException.html" title="class in gngs">
                                    StopParsingVCFException
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="WebServiceException.html" title="class in gngs">
                                    WebServiceException
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>

                    </tbody>
                </table>
            </li>
            </ul>
        </div>
    




</div>

<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
    <!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="overview-summary.html">Overview</a></li>
        <li class="navBarCell1Rev">Package</li>
        <li>Class</li>
        <li><a href="deprecated-list.html">Deprecated</a></li>
        <li><a href="index-all.html">Index</a></li>
        <li><a href="help-doc.html">Help</a></li>
    </ul>
</div>
<div class="subNav">
    <div>
        <ul class="navList">
            <li><a href="index.html?deprecated-list.html" target="_top">Frames</a></li>
            <li><a href="deprecated-list.html" target="_top">No Frames</a></li>
        </ul>
    </div>

    <div class="aboutLanguage"><em>Groovy Documentation</em></div>
    <a name="skip-navbar_bottom">
        <!--   -->
    </a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

</body>
</html>
