<!--

     Licensed to the Apache Software Foundation (ASF) under one
     or more contributor license agreements.  See the NOTICE file
     distributed with this work for additional information
     regarding copyright ownership.  The ASF licenses this file
     to you under the Apache License, Version 2.0 (the
     "License"); you may not use this file except in compliance
     with the License.  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

     Unless required by applicable law or agreed to in writing,
     software distributed under the License is distributed on an
     "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
     KIND, either express or implied.  See the License for the
     specific language governing permissions and limitations
     under the License.

-->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- **************************************************************** -->
<!-- *  PLEASE KEEP COMPLICATED EXPRESSIONS OUT OF THESE TEMPLATES, * -->
<!-- *  i.e. only iterate & print data where possible. Thanks, Jez. * -->
<!-- **************************************************************** -->

<html>
<head>
    <!-- Generated by groovydoc (3.0.10) on Sun Apr 09 09:06:44 NZST 2023 -->
    <title>SAM</title>
    <meta name="date" content="2023-04-09">
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <link href="../groovy.ico" type="image/x-icon" rel="shortcut icon">
    <link href="../groovy.ico" type="image/x-icon" rel="icon">
    <link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">

<body class="center">
<script type="text/javascript"><!--
if (location.href.indexOf('is-external=true') == -1) {
    parent.document.title="SAM";
}
//-->
</script>
<noscript>
    <div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
    <!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="../overview-summary.html">Overview</a></li>
        <li><a href="package-summary.html">Package</a></li>
        <li class="navBarCell1Rev">Class</li>
        <li><a href="../deprecated-list.html">Deprecated</a></li>
        <li><a href="../index-all.html">Index</a></li>
        <li><a href="../help-doc.html">Help</a></li>
    </ul>
</div>

<div class="subNav">
    <div>
        <ul class="navList">
            <li><a href="../index.html?gngs/SAM" target="_top">Frames</a></li>
            <li><a href="SAM.html" target="_top">No Frames</a></li>
        </ul>
    </div>
    <div>
        <ul class="subNavList">
            <li>Summary:&nbsp;</li>
            Nested&nbsp;&nbsp;&nbsp;Field&nbsp;&nbsp;&nbsp;<li><a href="#property_summary">Property</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#constructor_summary">Constructor</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#method_summary">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
        <ul class="subNavList">
            <li>&nbsp;|&nbsp;Detail:&nbsp;</li>
            Field&nbsp;&nbsp;&nbsp;<li><a href="#prop_detail">Property</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#constructor_detail">Constructor</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#method_detail">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
    </div>
    <a name="skip-navbar_top">
        <!--   -->
    </a></div>
<!-- ========= END OF TOP NAVBAR ========= -->

<!-- ======== START OF CLASS DATA ======== -->
<div class="header">

    <div class="subTitle">Package: <strong>gngs</strong></div>

    <h2 title="[Groovy] Class SAM" class="title">[Groovy] Class SAM</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><ul class="inheritance"></ul></li><li>gngs.SAM
</ul>
<div class="description">
    <ul class="blockList">
        <li class="blockList">


            <!-- todo: direct known subclasses -->
            <hr>
            <br>

<pre>class SAM
extends java.lang.Object</pre>

    <p> Adds various Groovy idioms and convenience features to the 
 Picard SamReader.
 <p>
 There are three major classes of functionality supported:
 
 <li>Iterating through and filtering reads in various ways</li>
 <li>Generating pileups and calculating read depth / coverage</li>
 <li>Accessing meta data (read groups, sample information, etc)</li>
 
 <p>
 For simple looping, the <a href='#eachRead(groovy.lang.Closure)'>eachRead(Closure)</a> static method
 can be used without creating a SAM object at all:
 <pre> SAM.eachRead { SAMRecord r -> println r.readName } </pre>
 (this will read from standard input). More sophisticated use requires the construction of a SAM object, 
 which allows, for example, iteration of read pairs:
 <pre> new SAM("test.bam").eachPair { r1, r2 -> assert r1.readName == r2.readName } </pre>
 <p>
 A region can be optionally passed to iterate over:
 
 <pre> new SAM("test.bam").eachPair("chr1",1000,2000) { r1, r2 -> assert r1.readName == r2.readName }</pre>
 
 Filtering a BAM file to create file containing a subset of reads is supported explicitly:
 
 <pre>new SAM("test.bam").filter("out.bam") { it.mappingQuality > 30 }</pre>
 
 Generating pileups is also straightforward:
 <pre> new SAM("test.bam").pileup("chr1",1000,2000) { p -> 
     println "There are ${p.countOf('A')} A bases at position chr1:$p.position" 
 }</pre>
 See the <a href='../gngs/Pileup.html' title='Pileup'>Pileup</a> class for more information about operations on pileups.
 <p><br>
 <i>Notes:</i>
 <li>Most operations filter out reads with mapping quality 0 by default. To 
 avoid this, set the minMappingQuality property on the SAM object.
 <li>All operations with this class require the BAM file to be indexed.
 <li>SAM and BAM files are treated transparently the same.
 <p>
 <h3>Progress</h3>
 Some operations can automatically print progress as they go. To enable this,
 set a {@link#ProgressCounter} on the #progress attribute. <DL><DT><B>Authors:</B></DT><DD>simon.sadedin@mcri.edu.au</DD></DL></p>

          </li>
    </ul>
</div>

<div class="summary">
    <ul class="blockList">
        <li class="blockList">
        <!-- =========== NESTED CLASS SUMMARY =========== -->
        

        <!-- =========== ENUM CONSTANT SUMMARY =========== -->
        
        <!-- =========== FIELD SUMMARY =========== -->
        

        <!-- =========== PROPERTY SUMMARY =========== -->
        
         <ul class="blockList">
                <li class="blockList"><a name="property_summary"><!--   --></a>
                    <h3>Properties Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Properties Summary table, listing nested classes, and an explanation">
                        <caption><span>Properties</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Type</th>
                            <th class="colLast" scope="col">Name and description</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.io.File</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#indexFile">indexFile</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>int</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#minMappingQuality">minMappingQuality</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>static&nbsp;boolean</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#progress">progress</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.io.File</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#samFile">samFile</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>htsjdk.samtools.SamReader</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#samFileReader">samFileReader</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.io.InputStream</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#samStream">samStream</a></code><br>Only used when created from stream</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.util.List&lt;java.lang.String&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#samples">samples</a></code><br>List of samples in the BAM file. </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>boolean</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#useMemoryMapping">useMemoryMapping</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>boolean</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#verbose">verbose</a></code><br></td>
                        </tr>
                        
                    </table>
                   </ul>
                </li>
            </ul>
        

        <!-- =========== ELEMENT SUMMARY =========== -->
        

        
        <!-- ======== CONSTRUCTOR SUMMARY ======== -->
        <ul class="blockList">
                <li class="blockList"><a name="constructor_summary"><!--   --></a>
                    <h3>Constructor Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructors Summary table">
                        <caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Constructor and description</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst">
                                <code><a href="#SAM()">SAM</a>
                                ()</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst">
                                <code><a href="#SAM(java.io.InputStream)">SAM</a>
                                (java.io.InputStream ips)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst">
                                <code><a href="#SAM(java.lang.String)">SAM</a>
                                (java.lang.String fileName)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst">
                                <code><a href="#SAM(java.io.File)">SAM</a>
                                (java.io.File file)</code><br></td>
                        </tr>
                        
                    </table>
                   </ul>
              </li>
        </ul>
        

        
        <!-- ========== METHOD SUMMARY =========== -->
        <ul class="blockList">
            
            <li class="blockList"><a name="method_summary"><!--   --></a>
                    <h3>Methods Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Methods Summary table">
                        <caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Type Params</th>
                            <th class="colLast" scope="col">Return Type</th>
                            <th class="colLast" scope="col">Name and description</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/Regions.html'>Regions</a></code></td>
                            <td class="colLast"><code><strong><a href="#asType(java.lang.Class)">asType</a></strong>(java.lang.Class clazz)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Map&lt;java.lang.String, java.lang.Integer&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#basesAt(java.lang.String, int)">basesAt</a></strong>(java.lang.String chr, int pos)</code><br>Return a Map with a key for each base observed at the given position,
 with the value being the number of times that base was observed. </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#close()">close</a></strong>()</code><br>Close the underlying SamReader</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#countOf(java.lang.String, int, java.lang.String)">countOf</a></strong>(java.lang.String chr, int pos, java.lang.String baseString)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Map&lt;java.lang.String, java.lang.Long&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#countOnTarget(gngs.Regions)">countOnTarget</a></strong>(<a href='../gngs/Regions.html'>Regions</a> targets)</code><br>Return a set of read counts indicating the counts of reads in that overlap the target region.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#countPairs(gngs.Region)">countPairs</a></strong>(gngs.Region r)</code><br>Count the number of read pairs in the given region</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#countPairs(java.lang.String, int, int)">countPairs</a></strong>(java.lang.String chr, int start, int end)</code><br>Count the number of read pairs in the given region</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#coverage(java.lang.String, int, groovy.lang.Closure)">coverage</a></strong>(java.lang.String chr, int pos, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#coverage(java.lang.String, int, int, groovy.lang.Closure)">coverage</a></strong>(java.lang.String chr, int pos, int end, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>static&nbsp;int</code></td>
                            <td class="colLast"><code><strong><a href="#coverage(htsjdk.samtools.SamReader, java.lang.String, int, int, groovy.lang.Closure, int)">coverage</a></strong>(htsjdk.samtools.SamReader r, java.lang.String chr, int pos, int end, groovy.lang.Closure c, int minMappingQuality)</code><br>Return the number of mapped reads overlapping the given position</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/RegulatingActor.html'>RegulatingActor</a></code></td>
                            <td class="colLast"><code><strong><a href="#coverageAsync(gngs.Region, int, groovy.lang.Closure)">coverageAsync</a></strong>(gngs.Region region, int minMAPQ, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/RegulatingActor.html'>RegulatingActor</a></code></td>
                            <td class="colLast"><code><strong><a href="#coverageAsync(gngs.Regions, int, groovy.lang.Closure)">coverageAsync</a></strong>(<a href='../gngs/Regions.html'>Regions</a> regions, int minMAPQ, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#coverageAsync(gngs.Region, gngs.RegulatingActor, int)">coverageAsync</a></strong>(gngs.Region region, <a href='../gngs/RegulatingActor.html'>RegulatingActor</a> sink, int minMAPQ)</code><br>Stream coverage values as <a href='../gngs/coverage/SampleReadCount.html' title='SampleReadCount'>SampleReadCount</a> objects to the given actor 
 from the given region(s).</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#coverageAsync(gngs.Regions, gngs.RegulatingActor, int)">coverageAsync</a></strong>(<a href='../gngs/Regions.html'>Regions</a> regions, <a href='../gngs/RegulatingActor.html'>RegulatingActor</a> sink, int minMAPQ)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/CoverageStats.html'>CoverageStats</a></code></td>
                            <td class="colLast"><code><strong><a href="#coverageStatistics(gngs.IRegion)">coverageStatistics</a></strong>(<a href='../gngs/IRegion.html'>IRegion</a> r)</code><br>Create a CoverageStats object for the depth of coverage over the given region</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/CoverageStats.html'>CoverageStats</a></code></td>
                            <td class="colLast"><code><strong><a href="#coverageStatistics(java.lang.String, int, int)">coverageStatistics</a></strong>(java.lang.String chr, int pos, int end)</code><br>Create a CoverageStats object for the depth of coverage over the given region</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/CoverageStats.html'>CoverageStats</a></code></td>
                            <td class="colLast"><code><strong><a href="#coverageStatistics(gngs.Regions)">coverageStatistics</a></strong>(<a href='../gngs/Regions.html'>Regions</a> regions)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachPair(groovy.lang.Closure)">eachPair</a></strong>(groovy.lang.Closure c)</code><br>Call the given closure for every pair of reads in a BAM file containing paired end reads.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachPair(java.util.Map, groovy.lang.Closure)">eachPair</a></strong>(java.util.Map options, groovy.lang.Closure c)</code><br>Call the given closure for every pair of reads in a BAM file containing paired end reads.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachPair(java.util.Map, htsjdk.samtools.SAMRecordIterator, groovy.lang.Closure)">eachPair</a></strong>(java.util.Map options, htsjdk.samtools.SAMRecordIterator iter, groovy.lang.Closure c)</code><br>Call the given closure for every pair of reads in a BAM file containing paired end reads.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachPair(gngs.Region, groovy.lang.Closure)">eachPair</a></strong>(gngs.Region r, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachPair(java.lang.String, groovy.lang.Closure)">eachPair</a></strong>(java.lang.String chr, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachPair(java.util.Map, java.lang.String, int, int, groovy.lang.Closure)">eachPair</a></strong>(java.util.Map options, java.lang.String chr, int start, int end, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>static&nbsp;void</code></td>
                            <td class="colLast"><code><strong><a href="#eachRead(groovy.lang.Closure)">eachRead</a></strong>(groovy.lang.Closure c)</code><br>Read a BAM or SAM file from standard input and call the given closure for
 each read contained therein.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachRecord(java.util.Map, groovy.lang.Closure)">eachRecord</a></strong>(java.util.Map options, groovy.lang.Closure c)</code><br>Iterate over each record in the same file in the order they are in the file</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachRecord(RegulatingActor)">eachRecord</a></strong>(<a href='../gngs/RegulatingActor.html' title='RegulatingActor'>RegulatingActor</a>&lt;SAMRecord&gt; actor)</code><br>Iterate over each record in the same file in the order they are in the file</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachRecord(gngs.Region, RegulatingActor)">eachRecord</a></strong>(gngs.Region region, <a href='../gngs/RegulatingActor.html' title='RegulatingActor'>RegulatingActor</a>&lt;java.util.List&lt;SAMRecord&gt;&gt; actor)</code><br>Iterate over each record in the same file in the order they are in the file</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#eachRecord(int, groovy.lang.Closure)">eachRecord</a></strong>(int threads, groovy.lang.Closure c)</code><br>Call the given closure for each read in this alignment, using the
 given number of threads eg:</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#filter(groovy.lang.Closure)">filter</a></strong>(groovy.lang.Closure c)</code><br>Filter the SAM/BAM file to include only the reads for which the 
 given closure returns true. </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#filter(java.util.Map, java.lang.String, groovy.lang.Closure)">filter</a></strong>(java.util.Map options, java.lang.String outputFile, groovy.lang.Closure c)</code><br>Filter the SAM/BAM file to include only the reads for which the given closure returns true</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#filterOrderedPairs(java.util.Map, java.lang.String, groovy.lang.Closure)">filterOrderedPairs</a></strong>(java.util.Map options, java.lang.String outputFileName, groovy.lang.Closure c)</code><br>Execute the given closure with an actor (parallel thread) set to write ordered 
 pairs to the given output file, based on this BAM file.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#genotype(java.lang.String, int)">genotype</a></strong>(java.lang.String chr, int pos)</code><br>Use a simple thresholding approach to genotype SNPs at the given location</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;java.lang.String&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getContigList()">getContigList</a></strong>()</code><br> <DL><DT><B>Returns:</B></DT><DD>a list of contigs in the BAM file in sort order of the BAM file</DD></DL></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/Regions.html'>Regions</a></code></td>
                            <td class="colLast"><code><strong><a href="#getContigRegions()">getContigRegions</a></strong>()</code><br>Return the contigs of this BAM file as a set of regions</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Map&lt;java.lang.String, java.lang.Integer&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getContigs()">getContigs</a></strong>()</code><br> <DL><DT><B>Returns:</B></DT><DD>Map of contig name to size for all the contigs in this SAM</DD></DL></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;SAMReadGroupRecord&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getReadGroups()">getReadGroups</a></strong>()</code><br> <DL><DT><B>Returns:</B></DT><DD>Return the list of read groups present in the SAM file</DD></DL></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;java.lang.String&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getSamples()">getSamples</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>static&nbsp;void</code></td>
                            <td class="colLast"><code><strong><a href="#index(java.io.File)">index</a></strong>(java.io.File bamFile)</code><br>Create an index for the given BAM file</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>float</code></td>
                            <td class="colLast"><code><strong><a href="#meanCoverage(java.lang.String, int, int)">meanCoverage</a></strong>(java.lang.String chr, int pos, int end)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#movingWindow(int, java.lang.String, groovy.lang.Closure, groovy.lang.Closure)">movingWindow</a></strong>(int windowSize, java.lang.String chr, groovy.lang.Closure c, groovy.lang.Closure filterFn)</code><br>Call the given closure for each base position with a moving window of reads over that position</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#movingWindow(int, java.lang.String, int, int, groovy.lang.Closure, groovy.lang.Closure)">movingWindow</a></strong>(int windowSize, java.lang.String chr, int start, int end, groovy.lang.Closure c, groovy.lang.Closure filterFn)</code><br>Call the given closure for each base position with a moving window of reads over that position</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>htsjdk.samtools.SamReader</code></td>
                            <td class="colLast"><code><strong><a href="#newReader(java.util.Map)">newReader</a></strong>(java.util.Map options)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>htsjdk.samtools.SAMFileWriter</code></td>
                            <td class="colLast"><code><strong><a href="#newWriter(java.lang.String)">newWriter</a></strong>(java.lang.String outputFileName)</code><br>Return a new SAMFileWriter configured with the same settings as this 
 SAM. </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/PileupIterator.Pileup.html'>Pileup</a></code></td>
                            <td class="colLast"><code><strong><a href="#pileup(java.lang.String, int)">pileup</a></strong>(java.lang.String chr, int pos)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#pileup(java.lang.String, int, int, groovy.lang.Closure)">pileup</a></strong>(java.lang.String chr, int start, int end, groovy.lang.Closure c)</code><br>Call the given closure once for each base between the 
 start and end positions with a Pileup object representing
 the pileup state at that position.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/PileupIterator.html'>PileupIterator</a></code></td>
                            <td class="colLast"><code><strong><a href="#pileup(java.lang.String, int, int)">pileup</a></strong>(java.lang.String chr, int start, int end)</code><br>Create and return an iterator that iterates over Pileup objects over
 the given range.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/PileupIterator.html'>PileupIterator</a></code></td>
                            <td class="colLast"><code><strong><a href="#pileup(htsjdk.samtools.SamReader, java.lang.String, int, int)">pileup</a></strong>(htsjdk.samtools.SamReader reader, java.lang.String chr, int start, int end)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>htsjdk.samtools.SAMRecord</code></td>
                            <td class="colLast"><code><strong><a href="#queryMate(htsjdk.samtools.SamReader, htsjdk.samtools.SAMRecord)">queryMate</a></strong>(htsjdk.samtools.SamReader r, htsjdk.samtools.SAMRecord r1)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#size()">size</a></strong>()</code><br>Count the total number of reads in the SAM file</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#sniffGenomeBuild()">sniffGenomeBuild</a></strong>()</code><br>Probe the given BAM file to make a guess about what genome build it is 
 generated from.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/Regions.html'>Regions</a></code></td>
                            <td class="colLast"><code><strong><a href="#toPairRegions(java.lang.String, int, int, int)">toPairRegions</a></strong>(java.lang.String chr, int start, int end, int maxSize)</code><br>Return read pairs from this SAM file that overlap the specified region
 as a Regions object - that is, as a set of genomic intervals. </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;QueryInterval&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#toQueryIntervals(gngs.Regions)">toQueryIntervals</a></strong>(<a href='../gngs/Regions.html'>Regions</a> regions)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/Regions.html'>Regions</a></code></td>
                            <td class="colLast"><code><strong><a href="#toRegions(gngs.Region)">toRegions</a></strong>(gngs.Region overlapping)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/Regions.html'>Regions</a></code></td>
                            <td class="colLast"><code><strong><a href="#toRegions(java.lang.String, int, int)">toRegions</a></strong>(java.lang.String chr, int start, int end)</code><br>Return reads from this SAM file that overlap the specified region
 as a Regions object - that is, as a set of genomic intervals.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.Object</code></td>
                            <td class="colLast"><code><strong><a href="#withIterator(gngs.Region, groovy.lang.Closure)">withIterator</a></strong>(gngs.Region region, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.Object</code></td>
                            <td class="colLast"><code><strong><a href="#withIterator(groovy.lang.Closure)">withIterator</a></strong>(groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.Object</code></td>
                            <td class="colLast"><code><strong><a href="#withOrderedPairActor(java.util.Map, java.lang.String, groovy.lang.Closure)">withOrderedPairActor</a></strong>(java.util.Map options, java.lang.String outputFileName, groovy.lang.Closure c)</code><br>Execute the given closure with an actor set to write ordered pairs to the given
 output file, based on this BAM file.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.Object</code></td>
                            <td class="colLast"><code><strong><a href="#withOrderedPairWriter(java.util.Map, java.lang.String, boolean, groovy.lang.Closure)">withOrderedPairWriter</a></strong>(java.util.Map options, java.lang.String outputFileName, boolean sorted, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.Object</code></td>
                            <td class="colLast"><code><strong><a href="#withReader(groovy.lang.Closure)">withReader</a></strong>(groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.Object</code></td>
                            <td class="colLast"><code><strong><a href="#withWriter(java.lang.String, groovy.lang.Closure)">withWriter</a></strong>(java.lang.String outputFileName, groovy.lang.Closure c)</code><br>Return a new SAMFileWriter configured with the same settings as this 
 SAM. </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.Object</code></td>
                            <td class="colLast"><code><strong><a href="#withWriter(java.lang.String, boolean, groovy.lang.Closure)">withWriter</a></strong>(java.lang.String outputFileName, boolean sorted, groovy.lang.Closure c)</code><br></td>
                        </tr>
                        
                    </table>
                   </ul>
              </li>
            
            <li class="blockList"><a name="method_summary"><!--   --></a>
                    <h3>Inherited Methods Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Inherited Methods Summary table">
                        <caption><span>Inherited Methods</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Methods inherited from class</th>
                            <th class="colLast" scope="col">Name</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code>class java.lang.Object</code></td>
                            <td class="colLast"><code>java.lang.Object#wait(long), java.lang.Object#wait(long, int), java.lang.Object#wait(), java.lang.Object#equals(java.lang.Object), java.lang.Object#toString(), java.lang.Object#hashCode(), java.lang.Object#getClass(), java.lang.Object#notify(), java.lang.Object#notifyAll()</code></td>
                        </tr>
                    </table>
                   </ul>
              </li>
                
        </ul>
        
    </li>
    </ul>
</div>

<div class="details">
    <ul class="blockList">
        <li class="blockList">
           

            

            
            <!-- =========== PROPERTY DETAIL =========== -->
            <ul class="blockList">
                <li class="blockList"><a name="prop_detail">
                    <!--   -->
                </a>
                    <h3>Property Detail</h3>
                    
                        <a name="indexFile"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.io.File <strong>indexFile</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="minMappingQuality"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>int <strong>minMappingQuality</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="progress"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>static&nbsp;boolean <strong>progress</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="samFile"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.io.File <strong>samFile</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="samFileReader"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>htsjdk.samtools.SamReader <strong>samFileReader</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="samStream"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.io.InputStream <strong>samStream</strong></h4>
                                <p> Only used when created from stream</p>
                            </li>
                        </ul>
                    
                        <a name="samples"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.String&gt; <strong>samples</strong></h4>
                                <p> List of samples in the BAM file. Note: this list can be overridden
 to associate different samples to the BAM files to those in the BAM 
 header.</p>
                            </li>
                        </ul>
                    
                        <a name="useMemoryMapping"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>boolean <strong>useMemoryMapping</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="verbose"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>boolean <strong>verbose</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                </li>
            </ul>
            

            

            
            <!-- =========== CONSTRUCTOR DETAIL =========== -->
            <ul class="blockList">
                <li class="blockList"><a name="constructor_detail">
                    <!--   -->
                </a>
                    <h3>Constructor Detail</h3>
                    
                        <a name="SAM()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><strong>SAM</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="SAM(java.io.InputStream)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><strong>SAM</strong>(java.io.InputStream ips)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="SAM(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><strong>SAM</strong>(java.lang.String fileName)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="SAM(java.io.File)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><strong>SAM</strong>(java.io.File file)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                </li>
            </ul>
            


            
            <!-- =========== METHOD DETAIL =========== -->
            <ul class="blockList">
                <li class="blockList"><a name="method_detail">
                    <!--   -->
                </a>
                    <h3>Method Detail</h3>
                    
                        <a name="asType(java.lang.Class)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../gngs/Regions.html'>Regions</a> <strong>asType</strong>(java.lang.Class clazz)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="basesAt(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.util.Map&lt;java.lang.String, java.lang.Integer&gt; <strong>basesAt</strong>(java.lang.String chr, int pos)</h4>
                                <p> Return a Map with a key for each base observed at the given position,
 with the value being the number of times that base was observed. Additionally,
 a keys for deletions ('deletion') and total bases ('total') are set.
 @return <DL><DT><B>Parameters:</B></DT><DD>chr</DD><DD>pos</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="close()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>close</strong>()</h4>
                                <p> Close the underlying SamReader</p>
                            </li>
                        </ul>
                    
                        <a name="countOf(java.lang.String, int, java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>countOf</strong>(java.lang.String chr, int pos, java.lang.String baseString)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="countOnTarget(gngs.Regions)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Map&lt;java.lang.String, java.lang.Long&gt; <strong>countOnTarget</strong>(<a href='../gngs/Regions.html'>Regions</a> targets)</h4>
                                <p> Return a set of read counts indicating the counts of reads in that overlap the target region.
 <p><DL><DT><B>todo:</B></DT><DD>this is pretty inefficient compared to how it could be. It makes 2 passes through the data
          where it could do 1.</DD></DL> <DL><DT><B>Parameters:</B></DT><DD>targets</DD></DL><DL><DT><B>Returns:</B></DT><DD>Map with keys on_target: reads overlapping target region, and bp_on_target: number of base pairs
         sequenced that overlap the target region, total: total count of reads.</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="countPairs(gngs.Region)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>int <strong>countPairs</strong>(gngs.Region r)</h4>
                                <p> Count the number of read pairs in the given region <DL><DT><B>Parameters:</B></DT><DD><code>r</code> -  region to count pairs over</DD></DL><DL><DT><B>Returns:</B></DT><DD>number of read pairs that overlap the region</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="countPairs(java.lang.String, int, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>countPairs</strong>(java.lang.String chr, int start, int end)</h4>
                                <p> Count the number of read pairs in the given region <DL><DT><B>Parameters:</B></DT><DD><code>chr</code> -    chromsome of region</DD><DD><code>start</code> -  start of region</DD><DD><code>end</code> -    end of region</DD></DL><DL><DT><B>Returns:</B></DT><DD>number of read pairs that overlap the region</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="coverage(java.lang.String, int, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>coverage</strong>(java.lang.String chr, int pos, groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="coverage(java.lang.String, int, int, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>coverage</strong>(java.lang.String chr, int pos, int end, groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="coverage(htsjdk.samtools.SamReader, java.lang.String, int, int, groovy.lang.Closure, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
static&nbsp;int <strong>coverage</strong>(htsjdk.samtools.SamReader r, java.lang.String chr, int pos, int end, groovy.lang.Closure c, int minMappingQuality)</h4>
                                <p> Return the number of mapped reads overlapping the given position <DL><DT><B>Parameters:</B></DT><DD><code>r</code> -      the SamReader (SAM / BAM file) containing reads</DD><DD><code>chr</code> -    the sequence name / chromosome to query</DD><DD><code>pos</code> -    the chromosomal position to query</DD><DD><code>c</code> -      a filter to include or reject reads</DD></DL><DL><DT><B>Returns:</B></DT><DD>the number of reads overlapping the position in the file</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="coverageAsync(gngs.Region, int, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/RegulatingActor.html'>RegulatingActor</a> <strong>coverageAsync</strong>(gngs.Region region, int minMAPQ, @groovy.transform.stc.ClosureParams(value: SimpleType, options: gngs.SampleReadCount) groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="coverageAsync(gngs.Regions, int, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/RegulatingActor.html'>RegulatingActor</a> <strong>coverageAsync</strong>(<a href='../gngs/Regions.html'>Regions</a> regions, int minMAPQ, @groovy.transform.stc.ClosureParams(value: SimpleType, options: gngs.SampleReadCount) groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="coverageAsync(gngs.Region, gngs.RegulatingActor, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>coverageAsync</strong>(gngs.Region region, <a href='../gngs/RegulatingActor.html'>RegulatingActor</a> sink, int minMAPQ)</h4>
                                <p> Stream coverage values as <a href='../gngs/coverage/SampleReadCount.html' title='SampleReadCount'>SampleReadCount</a> objects to the given actor 
 from the given region(s).</p>
                            </li>
                        </ul>
                    
                        <a name="coverageAsync(gngs.Regions, gngs.RegulatingActor, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>coverageAsync</strong>(<a href='../gngs/Regions.html'>Regions</a> regions, <a href='../gngs/RegulatingActor.html'>RegulatingActor</a> sink, int minMAPQ)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="coverageStatistics(gngs.IRegion)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/CoverageStats.html'>CoverageStats</a> <strong>coverageStatistics</strong>(<a href='../gngs/IRegion.html'>IRegion</a> r)</h4>
                                <p> Create a CoverageStats object for the depth of coverage over the given region</p>
                            </li>
                        </ul>
                    
                        <a name="coverageStatistics(java.lang.String, int, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/CoverageStats.html'>CoverageStats</a> <strong>coverageStatistics</strong>(java.lang.String chr, int pos, int end)</h4>
                                <p> Create a CoverageStats object for the depth of coverage over the given region</p>
                            </li>
                        </ul>
                    
                        <a name="coverageStatistics(gngs.Regions)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/CoverageStats.html'>CoverageStats</a> <strong>coverageStatistics</strong>(<a href='../gngs/Regions.html'>Regions</a> regions)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="eachPair(groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>eachPair</strong>(@groovy.transform.stc.ClosureParams(value: SimpleType, options: [htsjdk.samtools.SAMRecord, htsjdk.samtools.SAMRecord]) groovy.lang.Closure c)</h4>
                                <p> Call the given closure for every pair of reads in a BAM file containing paired end reads.
 <p>
 Note: the algorithm works by keeping a running buffer of reads, and iterating through
 the reads in order until each single read finds its mate. This means that 
 reads having no mate accumulate in the buffer without ever being removed. Thus 
 a large BAM file containing millions of unpaired reads could cause this method to use
 substantial ammounts of memory. <DL><DT><B>Parameters:</B></DT><DD><code>c</code> -         Closure to call</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="eachPair(java.util.Map, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>eachPair</strong>(java.util.Map options, @groovy.transform.stc.ClosureParams(value: SimpleType, options: [htsjdk.samtools.SAMRecord, htsjdk.samtools.SAMRecord]) groovy.lang.Closure c)</h4>
                                <p> Call the given closure for every pair of reads in a BAM file containing paired end reads.
 <p>
 Note: the algorithm works by keeping a running buffer of reads, and iterating through
 the reads in order until each single read finds its mate. This means that 
 reads having no mate accumulate in the buffer without ever being removed. Thus 
 a large BAM file containing millions of unpaired reads could cause this method to use
 substantial ammounts of memory. <DL><DT><B>Parameters:</B></DT><DD><code>c</code> -         Closure to call</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="eachPair(java.util.Map, htsjdk.samtools.SAMRecordIterator, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>eachPair</strong>(java.util.Map options, htsjdk.samtools.SAMRecordIterator iter, @groovy.transform.stc.ClosureParams(value: SimpleType, options: [htsjdk.samtools.SAMRecord, htsjdk.samtools.SAMRecord]) groovy.lang.Closure c)</h4>
                                <p> Call the given closure for every pair of reads in a BAM file containing paired end reads.
 <p>
 Note: the algorithm works by keeping a running buffer of reads, and iterating through
 the reads in order until each single read finds its mate. This means that 
 reads having no mate accumulate in the buffer without ever being removed. Thus 
 a large BAM file containing millions of unpaired reads could cause this method to use
 substantial ammounts of memory. <DL><DT><B>Parameters:</B></DT><DD><code>iter</code> -     iterator to consumer reads from</DD><DD><code>c</code> -         Closure to call</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="eachPair(gngs.Region, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>eachPair</strong>(gngs.Region r, groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="eachPair(java.lang.String, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>eachPair</strong>(java.lang.String chr, groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="eachPair(java.util.Map, java.lang.String, int, int, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>eachPair</strong>(java.util.Map options, java.lang.String chr, int start, int end, groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="eachRead(groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
static&nbsp;void <strong>eachRead</strong>(groovy.lang.Closure c)</h4>
                                <p> Read a BAM or SAM file from standard input and call the given closure for
 each read contained therein. <DL><DT><B>Parameters:</B></DT><DD>c</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="eachRecord(java.util.Map, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>eachRecord</strong>(java.util.Map options, @groovy.transform.stc.ClosureParams(value: SimpleType, options: [htsjdk.samtools.SAMRecord]) groovy.lang.Closure c)</h4>
                                <p> Iterate over each record in the same file in the order they are in the file <DL><DT><B>Parameters:</B></DT><DD>c</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="eachRecord(RegulatingActor)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>eachRecord</strong>(<a href='../gngs/RegulatingActor.html' title='RegulatingActor'>RegulatingActor</a>&lt;SAMRecord&gt; actor)</h4>
                                <p> Iterate over each record in the same file in the order they are in the file <DL><DT><B>Parameters:</B></DT><DD>c</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="eachRecord(gngs.Region, RegulatingActor)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>eachRecord</strong>(gngs.Region region, <a href='../gngs/RegulatingActor.html' title='RegulatingActor'>RegulatingActor</a>&lt;java.util.List&lt;SAMRecord&gt;&gt; actor)</h4>
                                <p> Iterate over each record in the same file in the order they are in the file <DL><DT><B>Parameters:</B></DT><DD>c</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="eachRecord(int, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>eachRecord</strong>(int threads, groovy.lang.Closure c)</h4>
                                <p> Call the given closure for each read in this alignment, using the
 given number of threads eg:
 
 <code>
 SAM sam = new SAM("test.bam")
 sam.eachRecord(5) { r -> println(r.readName) }
 </code> <DL><DT><B>Parameters:</B></DT><DD><code>threads</code> -    number of threads to use</DD><DD><code>c</code> -          closure to call</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="filter(groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>filter</strong>(groovy.lang.Closure c)</h4>
                                <p> Filter the SAM/BAM file to include only the reads for which the 
 given closure returns true. Output is written to stdout. <DL><DT><B>Parameters:</B></DT><DD><code>c</code> -              Closure that should return true for reads that will be 
                      included in the output BAM file</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="filter(java.util.Map, java.lang.String, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>filter</strong>(java.util.Map options, java.lang.String outputFile, @groovy.transform.stc.ClosureParams(value: SimpleType, options: [htsjdk.samtools.SAMRecord]) groovy.lang.Closure c)</h4>
                                <p> Filter the SAM/BAM file to include only the reads for which the given closure returns true <DL><DT><B>Parameters:</B></DT><DD><code>outputFile</code> -     the path to the output file</DD><DD><code>c</code> -              Closure that should return true for reads that will be 
                      included in the output BAM file</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="filterOrderedPairs(java.util.Map, java.lang.String, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>filterOrderedPairs</strong>(java.util.Map options, java.lang.String outputFileName, groovy.lang.Closure c)</h4>
                                <p> Execute the given closure with an actor (parallel thread) set to write ordered 
 pairs to the given output file, based on this BAM file.
 <p>
 If the closure returns a SAMRecordPair then the pair is written. If any other 
 value is returned, the value is evaluated as a boolean and the original read pair 
 is written if the boolean is true.</p>
                            </li>
                        </ul>
                    
                        <a name="genotype(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.lang.String <strong>genotype</strong>(java.lang.String chr, int pos)</h4>
                                <p> Use a simple thresholding approach to genotype SNPs at the given location
 @return <DL><DT><B>Parameters:</B></DT><DD>chr</DD><DD>pos</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getContigList()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.util.List&lt;java.lang.String&gt; <strong>getContigList</strong>()</h4>
                                <p> <DL><DT><B>Returns:</B></DT><DD>a list of contigs in the BAM file in sort order of the BAM file</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getContigRegions()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/Regions.html'>Regions</a> <strong>getContigRegions</strong>()</h4>
                                <p> Return the contigs of this BAM file as a set of regions
 
 @return</p>
                            </li>
                        </ul>
                    
                        <a name="getContigs()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.util.Map&lt;java.lang.String, java.lang.Integer&gt; <strong>getContigs</strong>()</h4>
                                <p> <DL><DT><B>Returns:</B></DT><DD>Map of contig name to size for all the contigs in this SAM</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getReadGroups()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.util.List&lt;SAMReadGroupRecord&gt; <strong>getReadGroups</strong>()</h4>
                                <p> <DL><DT><B>Returns:</B></DT><DD>Return the list of read groups present in the SAM file</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getSamples()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.String&gt; <strong>getSamples</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="index(java.io.File)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
static&nbsp;void <strong>index</strong>(java.io.File bamFile)</h4>
                                <p> Create an index for the given BAM file <DL><DT><B>Parameters:</B></DT><DD>bamFile</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="meanCoverage(java.lang.String, int, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
float <strong>meanCoverage</strong>(java.lang.String chr, int pos, int end)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="movingWindow(int, java.lang.String, groovy.lang.Closure, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>movingWindow</strong>(int windowSize, java.lang.String chr, groovy.lang.Closure c, groovy.lang.Closure filterFn)</h4>
                                <p> Call the given closure for each base position with a moving window of reads over that position
 <p>
 <b>NOTE:</b> Requires a BAM file sorted by position. Will not work with unsorted bam file. <DL><DT><B>Parameters:</B></DT><DD>windowSize</DD><DD>chr</DD><DD>c</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="movingWindow(int, java.lang.String, int, int, groovy.lang.Closure, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>movingWindow</strong>(int windowSize, java.lang.String chr, int start, int end, groovy.lang.Closure c, groovy.lang.Closure filterFn)</h4>
                                <p> Call the given closure for each base position with a moving window of reads over that position
 <p>
 <b>NOTE:</b> Requires a BAM file sorted by position. Will not work with unsorted bam file. <DL><DT><B>Parameters:</B></DT><DD><code>windowSize</code> -     the size of window to use in bp</DD><DD><code>start</code> -          start position</DD><DD><code>end</code> -            end position</DD><DD><code>c</code> -              callback function to invoke</DD><DD><code>filterFn</code> -       optional filter function to apply that filters out reads</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="newReader(java.util.Map)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
htsjdk.samtools.SamReader <strong>newReader</strong>(java.util.Map options)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="newWriter(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>htsjdk.samtools.SAMFileWriter <strong>newWriter</strong>(java.lang.String outputFileName)</h4>
                                <p> Return a new SAMFileWriter configured with the same settings as this 
 SAM. It is the caller's responsibility to close the writer. <DL><DT><B>Parameters:</B></DT><DD><code>outputFileName</code> -     Name of file to write to</DD></DL><DL><DT><B>Returns:</B></DT><DD>SAMFileWriter</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="pileup(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/PileupIterator.Pileup.html'>Pileup</a> <strong>pileup</strong>(java.lang.String chr, int pos)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="pileup(java.lang.String, int, int, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>pileup</strong>(java.lang.String chr, int start, int end, groovy.lang.Closure c)</h4>
                                <p> Call the given closure once for each base between the 
 start and end positions with a Pileup object representing
 the pileup state at that position. <DL><DT><B>Parameters:</B></DT><DD>chr</DD><DD>start</DD><DD>end</DD><DD>c</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="pileup(java.lang.String, int, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/PileupIterator.html'>PileupIterator</a> <strong>pileup</strong>(java.lang.String chr, int start, int end)</h4>
                                <p> Create and return an iterator that iterates over Pileup objects over
 the given range. <DL><DT><B>Parameters:</B></DT><DD><code>chr</code> -    Chromsome of range to iterate over</DD><DD><code>start</code> -  Start of range</DD><DD><code>end</code> -    End of range</DD></DL><DL><DT><B>Returns:</B></DT><DD>Iterator for iterating over Pileup state in range</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="pileup(htsjdk.samtools.SamReader, java.lang.String, int, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/PileupIterator.html'>PileupIterator</a> <strong>pileup</strong>(htsjdk.samtools.SamReader reader, java.lang.String chr, int start, int end)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="queryMate(htsjdk.samtools.SamReader, htsjdk.samtools.SAMRecord)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
htsjdk.samtools.SAMRecord <strong>queryMate</strong>(htsjdk.samtools.SamReader r, htsjdk.samtools.SAMRecord r1)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="size()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>size</strong>()</h4>
                                <p> Count the total number of reads in the SAM file <DL><DT><B>Returns:</B></DT><DD>total number of reads (not pairs, each pair is counted as 2 reads)</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="sniffGenomeBuild()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>sniffGenomeBuild</strong>()</h4>
                                <p> Probe the given BAM file to make a guess about what genome build it is 
 generated from.
 
 @return</p>
                            </li>
                        </ul>
                    
                        <a name="toPairRegions(java.lang.String, int, int, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/Regions.html'>Regions</a> <strong>toPairRegions</strong>(java.lang.String chr, int start, int end, int maxSize)</h4>
                                <p> Return read pairs from this SAM file that overlap the specified region
 as a Regions object - that is, as a set of genomic intervals. Each 
 region returned by this method spans from the start of the 5' read
 to the end of the 3' read.
 <p>
 Reads that are missing a start or end alignment position are 
 omitted. Unpaired reads are also omitted.
 <p>
 If called without passing start or end, the start / end
 are interpreted as the beginning / end of the chromosome
 / reference sequence respectively. <DL><DT><B>Returns:</B></DT><DD>Regions object containing intervals representing all
            the reads in this SAM object</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="toQueryIntervals(gngs.Regions)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.util.List&lt;QueryInterval&gt; <strong>toQueryIntervals</strong>(<a href='../gngs/Regions.html'>Regions</a> regions)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="toRegions(gngs.Region)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../gngs/Regions.html'>Regions</a> <strong>toRegions</strong>(gngs.Region overlapping)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="toRegions(java.lang.String, int, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/Regions.html'>Regions</a> <strong>toRegions</strong>(java.lang.String chr, int start, int end)</h4>
                                <p> Return reads from this SAM file that overlap the specified region
 as a Regions object - that is, as a set of genomic intervals.
 <p>
 Reads that are missing a start or end alignment position are 
 omitted.
 <p>
 If called without passing start or end, the start / end
 are interpreted as the beginning / end of the chromosome
 / reference sequence respectively. <DL><DT><B>Returns:</B></DT><DD>Regions object containing intervals representing all
            the reads in this SAM object</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="withIterator(gngs.Region, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.lang.Object <strong>withIterator</strong>(gngs.Region region, @groovy.transform.stc.ClosureParams(value: SimpleType, options: [htsjdk.samtools.SAMRecordIterator]) groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="withIterator(groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.lang.Object <strong>withIterator</strong>(@groovy.transform.stc.ClosureParams(value: SimpleType, options: [htsjdk.samtools.SAMRecordIterator]) groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="withOrderedPairActor(java.util.Map, java.lang.String, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.Object <strong>withOrderedPairActor</strong>(java.util.Map options, java.lang.String outputFileName, groovy.lang.Closure c)</h4>
                                <p> Execute the given closure with an actor set to write ordered pairs to the given
 output file, based on this BAM file.</p>
                            </li>
                        </ul>
                    
                        <a name="withOrderedPairWriter(java.util.Map, java.lang.String, boolean, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.Object <strong>withOrderedPairWriter</strong>(java.util.Map options, java.lang.String outputFileName, boolean sorted, groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="withReader(groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.lang.Object <strong>withReader</strong>(groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="withWriter(java.lang.String, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.Object <strong>withWriter</strong>(java.lang.String outputFileName, groovy.lang.Closure c)</h4>
                                <p> Return a new SAMFileWriter configured with the same settings as this 
 SAM. It is the caller's responsibility to close the writer.
 In this version of the method the files are assumed to be 
 <em>pre-sorted</em>*. <DL><DT><B>Parameters:</B></DT><DD><code>outputFileName</code> -     Name of file to write to</DD></DL><DL><DT><B>Returns:</B></DT><DD>SAMFileWriter</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="withWriter(java.lang.String, boolean, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.Object <strong>withWriter</strong>(java.lang.String outputFileName, boolean sorted, groovy.lang.Closure c)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                </li>
            </ul>
            
        </li>
    </ul>
</div>

<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
    <!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="../overview-summary.html">Overview</a></li>
        <li><a href="package-summary.html">Package</a></li>
        <li class="navBarCell1Rev">Class</li>
        <li><a href="../deprecated-list.html">Deprecated</a></li>
        <li><a href="../index-all.html">Index</a></li>
        <li><a href="../help-doc.html">Help</a></li>
    </ul>
</div>

<div class="subNav">
    <div>
        <ul class="navList">
            <li><a href="../index.html?gngs/SAM" target="_top">Frames</a></li>
            <li><a href="SAM.html" target="_top">No Frames</a></li>
        </ul>
    </div>
    <div>
        <ul class="subNavList">
            <li>Summary:&nbsp;</li>
            Nested&nbsp;&nbsp;&nbsp;Field&nbsp;&nbsp;&nbsp;<li><a href="#property_summary">Property</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#constructor_summary">Constructor</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#method_summary">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
        <ul class="subNavList">
            <li>&nbsp;|&nbsp;Detail:&nbsp;</li>
            Field&nbsp;&nbsp;&nbsp;<li><a href="#prop_detail">Property</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#constructor_detail">Constructor</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#method_detail">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
    </div>
    <p>Groovy Documentation</p>
    <a name="skip-navbar_bottom">
        <!--   -->
    </a>
    </div>
</div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
