<!--

     Licensed to the Apache Software Foundation (ASF) under one
     or more contributor license agreements.  See the NOTICE file
     distributed with this work for additional information
     regarding copyright ownership.  The ASF licenses this file
     to you under the Apache License, Version 2.0 (the
     "License"); you may not use this file except in compliance
     with the License.  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

     Unless required by applicable law or agreed to in writing,
     software distributed under the License is distributed on an
     "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
     KIND, either express or implied.  See the License for the
     specific language governing permissions and limitations
     under the License.

-->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- **************************************************************** -->
<!-- *  PLEASE KEEP COMPLICATED EXPRESSIONS OUT OF THESE TEMPLATES, * -->
<!-- *  i.e. only iterate & print data where possible. Thanks, Jez. * -->
<!-- **************************************************************** -->

<html>
<head>
    <!-- Generated by groovydoc (2.4.10) on Thu Mar 01 18:43:49 AEDT 2018 -->
    <title>Variant (groovy-ngs-utils API)</title>
    <meta name="date" content="2018-03-01">
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <link href="../groovy.ico" type="image/x-icon" rel="shortcut icon">
    <link href="../groovy.ico" type="image/x-icon" rel="icon">
    <link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">

<body class="center">
<script type="text/javascript"><!--
if (location.href.indexOf('is-external=true') == -1) {
    parent.document.title="Variant (groovy-ngs-utils API)";
}
//-->
</script>
<noscript>
    <div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
    <!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="../overview-summary.html">Overview</a></li>
        <li><a href="package-summary.html">Package</a></li>
        <li class="navBarCell1Rev">Class</li>
        <li><a href="../deprecated-list.html">Deprecated</a></li>
        <li><a href="../index-all.html">Index</a></li>
        <li><a href="../help-doc.html">Help</a></li>
    </ul>
</div>

<div class="subNav">
    <div>
        <ul class="navList">
            <li><a href="../index.html?gngs/Variant" target="_top">Frames</a></li>
            <li><a href="Variant.html" target="_top">No Frames</a></li>
        </ul>
    </div>
    <div>
        <ul class="subNavList">
            <li>Summary:&nbsp;</li>
            Nested&nbsp;&nbsp;&nbsp;Field&nbsp;&nbsp;&nbsp;<li><a href="#property_summary">Property</a></li>&nbsp;&nbsp;&nbsp;Constructor&nbsp;&nbsp;&nbsp;<li><a href="#method_summary">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
        <ul class="subNavList">
            <li>&nbsp;|&nbsp;Detail:&nbsp;</li>
            Field&nbsp;&nbsp;&nbsp;<li><a href="#prop_detail">Property</a></li>&nbsp;&nbsp;&nbsp;Constructor&nbsp;&nbsp;&nbsp;<li><a href="#method_detail">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
    </div>
    <a name="skip-navbar_top">
        <!--   -->
    </a></div>
<!-- ========= END OF TOP NAVBAR ========= -->

<!-- ======== START OF CLASS DATA ======== -->
<div class="header">

    <div class="subTitle">Package: <strong>gngs</strong></div>

    <h2 title="[Groovy] Class Variant" class="title">[Groovy] Class Variant</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><ul class="inheritance"></ul></li><li>gngs.Variant
</ul>
<div class="description">
    <ul class="blockList">
        <li class="blockList">

            <dl>
                <dt>All Implemented Interfaces and Traits:</dt>
                <dd><a href='../gngs/IRegion.html'>IRegion</a></dd>
            </dl>
            <!-- todo: direct known subclasses -->
            <hr>
            <br>
<pre>class Variant
extends java.lang.Object

</pre>



    <p> Represents the genetic state at a specific locus in a genome
 ie: captures the information at a single line of a VCF file.
 Note that this can include multiple alleles that may be present
 at the site, having different start and end position.
 <p>
 The <a href='../gngs/Variant.html' title='Variant'>Variant</a> class includes support for accessing the structured
 annotations and attributes included in the VCF format and popular
 annotation tools such as VEP, Annovar and SnpEFF. See <a href='#getSnpEffInfo()'>getSnpEffInfo()</a>
 and <a href='#getVepInfo()'>getVepInfo()</a> for accessors that return parsed annotation
 information.
 <p>
 Although <a href='../gngs/Variant.html' title='Variant'>Variant</a> instances can be constructed manually, typically
 you will query them from <a href='../gngs/VCF.html' title='VCF'>VCF</a> or <a href='../gngs/VCFIndex.html' title='VCFIndex'>VCFIndex</a> classes, and this
 is required for some operations because the VCF header is required to 
 parse and understand some of the fields (for example, sample names). Without
 a header many operations still work, but you may experience java.lang.IllegalStateException
 exceptions if you call methods that depend on the header.
 <p>
 One of the most useful operations is to determine the dosage (ie: heterozygosity, number of copies)
 of a variant in a particular sample. Eg:
 <pre>
 VCF vcf = new VCF.parse("test.vcf")
 
 // Find all the variants that affect sample MYSAMPLE in any way
 List<Variant> mySampleVariants = vcf.grep {  Variant v -> v.sampleDosage("MYSAMPLE")>0 }
 
 // Find all the homozygous variants (for autosomal + female X chromosomes) 
 List<Variant> mySampleVariants = vcf.grep {  Variant v -> v.sampleDosage("MYSAMPLE") == 2 }
 ...
 </pre>
 
 Some support is implemented for understanding pedigrees via the <a href='../gngs/Pedigree.html' title='Pedigree'>Pedigree</a> class.
 If a pedigree is set on the <a href='../gngs/VCF.html' title='VCF'>VCF</a> class from which the <a href='../gngs/Variant.html' title='Variant'>Variant</a> originates, 
 then one can query variants to find variants that segregate with families or conditions.
 <p>
 A Variant also implements the IRange interface and thus you can use
 them seamlessly with a RegionSource. For example, to check
 if a Variant falls within a BED file works simply:
 <pre>
 Variant v = ...
 BED bed = new BED("test.bed").load()
 if(v in bed) 
  println "Variant v falls in the ranges included in the BED file"
 </pre>
 
 The Variant class supports working with the full genotype data of each sample, and
 the genotype fields are parsed for you. For example, to check if every sample is below
 a particular genotype quality:
 <pre>
 if(v.genoTypes.every { it.GQ < 5.0f })
     println "Quality is low for every sample!"
 </pre>
 <DL><DT><B>Authors:</B></DT><DD>simon.sadedin@mcri.edu.au</DD></DL></p>

          </li>
    </ul>
</div>

<div class="summary">
    <ul class="blockList">
        <li class="blockList">
        <!-- =========== NESTED CLASS SUMMARY =========== -->
        

        <!-- =========== ENUM CONSTANT SUMMARY =========== -->
        
        <!-- =========== FIELD SUMMARY =========== -->
        

        <!-- =========== PROPERTY SUMMARY =========== -->
        
         <ul class="blockList">
                <li class="blockList"><a name="property_summary"><!--   --></a>
                    <h3>Properties Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Properties Summary table, listing nested classes, and an explanation">
                        <caption><span>Properties</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Type</th>
                            <th class="colLast" scope="col">Name and description</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>static&nbsp;java.util.regex.Pattern</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#AMPERSAND_SPLIT">AMPERSAND_SPLIT</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>static&nbsp;java.util.regex.Pattern</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#COLON_SPLIT">COLON_SPLIT</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>static&nbsp;java.util.regex.Pattern</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#COMMA_SPLIT">COMMA_SPLIT</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>static&nbsp;java.util.regex.Pattern</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#PIPE_OR_SLASH_SPLIT">PIPE_OR_SLASH_SPLIT</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>static&nbsp;java.util.regex.Pattern</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#PIPE_SPLIT">PIPE_SPLIT</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>static&nbsp;java.util.regex.Pattern</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#TAB_SPLIT">TAB_SPLIT</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#alt">alt</a></code><br>The sequence for the first alternate allele. </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>byte</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#altByte">altByte</a></code><br>The first character of the first alternate allele in byte form. </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.lang.String[]</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#alts">alts</a></code><br>List of alternate alleles represented by the variant.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong><a href='../gngs/Region.html'>Region</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#cachedRegion">cachedRegion</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#chr">chr</a></code><br>The chromosome on which the variant falls</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.util.List&lt;java.lang.Integer&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#dosages">dosages</a></code><br>Cached set of dosages for the first allele.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#filter">filter</a></code><br>The filter field as loaded from the VCF file.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.lang.String[]</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#genoTypeFields">genoTypeFields</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.util.List&lt;java.util.Map&lt;java.lang.String, java.lang.Object&gt;&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#genoTypes">genoTypes</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong><a href='../gngs/VCF.html'>VCF</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#header">header</a></code><br>The VCF from which header information will be extracted when required. </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#id">id</a></code><br>ID column as per VCF spec. </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#info">info</a></code><br>The whole INFO field as a raw string. </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.util.Map&lt;java.lang.String, java.lang.Object&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#infos">infos</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#line">line</a></code><br>The original line from which this variant was parsed</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.util.Set&lt;<a href='../gngs/Pedigree.html' title='Pedigree'>Pedigree</a>&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#pedigrees">pedigrees</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>int</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#pos">pos</a></code><br>The position at which the reference bases (<a href='#ref'>ref</a> attribute) starts.
 </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>float</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#qual">qual</a></code><br>The QUAL field as loaded from the VCF file</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#ref">ref</a></code><br>The reference allele at the position. </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>boolean</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#snpEffDirty">snpEffDirty</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.util.List&lt;<a href='../gngs/SnpEffInfo.html' title='SnpEffInfo'>SnpEffInfo</a>&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#snpEffInfo">snpEffInfo</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#type">type</a></code><br>One of SNP, INS, DEL or SV indicating the type of change represented by the first 
 alternate allele.</td>
                        </tr>
                        
                    </table>
                   </ul>
                </li>
            </ul>
        

        <!-- =========== ELEMENT SUMMARY =========== -->
        

        

        
        <!-- ========== METHOD SUMMARY =========== -->
        <ul class="blockList">
            
            <li class="blockList"><a name="method_summary"><!--   --></a>
                    <h3>Methods Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Methods Summary table">
                        <caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Type Params</th>
                            <th class="colLast" scope="col">Return Type</th>
                            <th class="colLast" scope="col">Name and description</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#applyDEL(java.lang.String, int)">applyDEL</a></strong>(java.lang.String sequence, int position)</code><br>Apply this deletion to the specified sequence at the specified position</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#applyINS(java.lang.String, int)">applyINS</a></strong>(java.lang.String sequence, int position)</code><br>Apply this deletion to the specified sequence at the specified position</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#applySNP(java.lang.String, int)">applySNP</a></strong>(java.lang.String sequence, int position)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#applyTo(java.lang.String, int)">applyTo</a></strong>(java.lang.String sequence, int position)</code><br>Apply this variant to mutate the given sequence at the specified position</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#convertType(java.lang.String, java.lang.String)">convertType</a></strong>(java.lang.String refSeq, java.lang.String altSeq)</code><br>Convert reference and alternate allele strings into a 
 mutation type, being one of "SNP","INS","DEL","GAIN"
 or "LOSS", with the latter two representing CNVs.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#displaySequenceDifference(java.lang.String, int)">displaySequenceDifference</a></strong>(java.lang.String sequence, int position)</code><br>A debug function, displays the sequence difference and returns the 
 new sequence</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#equalsAnnovar(java.lang.String, int, java.lang.String)">equalsAnnovar</a></strong>(java.lang.String chr, int pos, java.lang.String obs)</code><br>Return true if the given Annovar formattd position and observation
 match those of this variant</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#findAlleleIndex(gngs.Allele)">findAlleleIndex</a></strong>(<a href='../gngs/Allele.html'>Allele</a> other)</code><br>Return the index of the allele (if any) that matches the give other variant</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#fixGenoTypeOrder()">fixGenoTypeOrder</a></strong>()</code><br>VCF requires genotype to be the 1st field, but some tools (R, grrrr) write it in whatever order they 
 feel like. </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>float</code></td>
                            <td class="colLast"><code><strong><a href="#getAlleleBalance()">getAlleleBalance</a></strong>()</code><br>Return the distance of the balance of the default allele from 0.5
 for the default alternate allele and first sample</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>float</code></td>
                            <td class="colLast"><code><strong><a href="#getAlleleBalance(int, int, int)">getAlleleBalance</a></strong>(int allele1, int allele2, int sampleIndex = 0)</code><br>Return the distance of the balance of the default allele from 0.5
 for the specififed alternate allele (reference = 0)</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;java.lang.Integer&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getAlleleDepths(int)">getAlleleDepths</a></strong>(int alleleIndex)</code><br>Return the number of reads supporting the given allele as a list
 with one entry for each sample in the VCF</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;<a href='../gngs/Allele.html' title='Allele'>Allele</a>&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getAlleles()">getAlleles</a></strong>()</code><br>Return a list of Allele objects representing alleles present on this line of the VCF</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getAllelesAndTypes()">getAllelesAndTypes</a></strong>()</code><br>Return a list of each allele and its type.
 </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#getConsequence(int)">getConsequence</a></strong>(int alleleIndex)</code><br>Return the consequence of the specified allele, at the moment,
 from VEP annotations (later, from others). </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;java.lang.Integer&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getDosages()">getDosages</a></strong>()</code><br>Return list of dosages (number of copies of allele) for 
 each sample for the first alternate allele.
 </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;java.lang.Integer&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getDosages(int)">getDosages</a></strong>(int alleleIndex)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;java.lang.String&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getGenes(java.lang.String)">getGenes</a></strong>(java.lang.String minVEPCons)</code><br>Return the list of genes impacted by this variant, in a manner that is
 neutral to the annotator used (VEP and SnpEFF supported)</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#getGenotypeDepth(Map<String, Object>, int)">getGenotypeDepth</a></strong>(java.util.Map&lt;java.lang.String, java.lang.Object&gt; gt, int alleleIndex)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Map&lt;java.lang.String, java.lang.Object&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getInfo()">getInfo</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>float</code></td>
                            <td class="colLast"><code><strong><a href="#getMaxAlleleBalance()">getMaxAlleleBalance</a></strong>()</code><br>Return the maximum allele balance for any sample for the default
 alternate allele.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/SnpEffInfo.html'>SnpEffInfo</a></code></td>
                            <td class="colLast"><code><strong><a href="#getMaxEffect()">getMaxEffect</a></strong>()</code><br>Return the most impactful SnpEff effect for this variant</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Map&lt;java.lang.String, java.lang.Object&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getMaxVep()">getMaxVep</a></strong>()</code><br>Return the details of the most severe VEP consequence, as ranked by
 <a href='../gngs/VEPConsequences.html#RANKED_CONSEQUENCES' title='VEPConsequences.RANKED_CONSEQUENCES'>VEPConsequences.RANKED_CONSEQUENCES</a>.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#getMaxVepImpact()">getMaxVepImpact</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>float</code></td>
                            <td class="colLast"><code><strong><a href="#getMaxVepMaf()">getMaxVepMaf</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Set&lt;<a href='../gngs/Pedigree.html' title='Pedigree'>Pedigree</a>&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getPedigrees()">getPedigrees</a></strong>()</code><br>Return a list of all the pedigrees that contain this variant</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>groovy.lang.IntRange</code></td>
                            <td class="colLast"><code><strong><a href="#getRange()">getRange</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code><a href='../gngs/Region.html'>Region</a></code></td>
                            <td class="colLast"><code><strong><a href="#getRegion()">getRegion</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;<a href='../gngs/SnpEffInfo.html' title='SnpEffInfo'>SnpEffInfo</a>&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getSnpEffInfo()">getSnpEffInfo</a></strong>()</code><br>Return a list of SnpEffInfo objects, each describing
 a separate SnpEff effect caused by the variant
 @return</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.Integer</code></td>
                            <td class="colLast"><code><strong><a href="#getTotalDepth(java.lang.String)">getTotalDepth</a></strong>(java.lang.String sample)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.List&lt;java.util.Map&lt;java.lang.String, java.lang.Object&gt;&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#getVepInfo()">getVepInfo</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>boolean</code></td>
                            <td class="colLast"><code><strong><a href="#isCase(gngs.IRegion)">isCase</a></strong>(<a href='../gngs/IRegion.html'>IRegion</a> r)</code><br>Return true if this variant overlaps the given range</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>boolean</code></td>
                            <td class="colLast"><code><strong><a href="#isSV()">isSV</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>static&nbsp;<a href='../gngs/Variant.html'>Variant</a></code></td>
                            <td class="colLast"><code><strong><a href="#parse(java.lang.String)">parse</a></strong>(java.lang.String line)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>static&nbsp;<a href='../gngs/Variant.html'>Variant</a></code></td>
                            <td class="colLast"><code><strong><a href="#parse(java.lang.String, boolean)">parse</a></strong>(java.lang.String line, boolean ignoreNonRef)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Map&lt;java.lang.String, java.lang.Object&gt;</code></td>
                            <td class="colLast"><code><strong><a href="#parseGenoTypeFields(java.lang.String)">parseGenoTypeFields</a></strong>(java.lang.String gt)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#sampleDosage(java.lang.String)">sampleDosage</a></strong>(java.lang.String sampleName)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#sampleDosage(java.lang.String, int)">sampleDosage</a></strong>(java.lang.String sampleName, int alleleIndex)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Map</code></td>
                            <td class="colLast"><code><strong><a href="#sampleGenoType(java.lang.String)">sampleGenoType</a></strong>(java.lang.String sampleName)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>boolean</code></td>
                            <td class="colLast"><code><strong><a href="#segregatesWith(gngs.Pedigree)">segregatesWith</a></strong>(<a href='../gngs/Pedigree.html'>Pedigree</a> ped)</code><br>Return a Map of fields from the genotype column corresponding
 to the given sample. </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#setAlt(java.lang.String)">setAlt</a></strong>(java.lang.String alt)</code><br>Update the first alternate allele to the given value</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>int</code></td>
                            <td class="colLast"><code><strong><a href="#size()">size</a></strong>()</code><br>Returns the change in size of the genome caused by this variant's default allele.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.util.Map</code></td>
                            <td class="colLast"><code><strong><a href="#toAnnovar(int)">toAnnovar</a></strong>(int alleleIndex = 0)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#toJson(java.lang.String)">toJson</a></strong>(java.lang.String sample = null)</code><br>Return a JSON string representing the key details about this variant.
 </td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#toString()">toString</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#update(groovy.lang.Closure)">update</a></strong>(groovy.lang.Closure c)</code><br>Update using a default description.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#update(java.lang.String, groovy.lang.Closure)">update</a></strong>(java.lang.String desc, groovy.lang.Closure c)</code><br>Allows various fields to be updated and then synchronises the 
 rest of the data with those updated fields</td>
                        </tr>
                        
                    </table>
                   </ul>
              </li>
            
            <li class="blockList"><a name="method_summary"><!--   --></a>
                    <h3>Inherited Methods Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Inherited Methods Summary table">
                        <caption><span>Inherited Methods</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Methods inherited from class</th>
                            <th class="colLast" scope="col">Name</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code>class java.lang.Object</code></td>
                            <td class="colLast"><code>java.lang.Object#wait(long, int), java.lang.Object#wait(long), java.lang.Object#wait(), java.lang.Object#equals(java.lang.Object), java.lang.Object#toString(), java.lang.Object#hashCode(), java.lang.Object#getClass(), java.lang.Object#notify(), java.lang.Object#notifyAll()</code></td>
                        </tr>
                    </table>
                   </ul>
              </li>
                
        </ul>
        
    </li>
    </ul>
</div>

<div class="details">
    <ul class="blockList">
        <li class="blockList">
           

            

            
            <!-- =========== PROPERTY DETAIL =========== -->
            <ul class="blockList">
                <li class="blockList"><a name="prop_detail">
                    <!--   -->
                </a>
                    <h3>Property Detail</h3>
                    
                        <a name="AMPERSAND_SPLIT"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>static&nbsp;final&nbsp;java.util.regex.Pattern <strong>AMPERSAND_SPLIT</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="COLON_SPLIT"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>static&nbsp;final&nbsp;java.util.regex.Pattern <strong>COLON_SPLIT</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="COMMA_SPLIT"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>static&nbsp;final&nbsp;java.util.regex.Pattern <strong>COMMA_SPLIT</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="PIPE_OR_SLASH_SPLIT"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>static&nbsp;final&nbsp;java.util.regex.Pattern <strong>PIPE_OR_SLASH_SPLIT</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="PIPE_SPLIT"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>static&nbsp;final&nbsp;java.util.regex.Pattern <strong>PIPE_SPLIT</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="TAB_SPLIT"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>static&nbsp;final&nbsp;java.util.regex.Pattern <strong>TAB_SPLIT</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="alt"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>alt</strong></h4>
                                <p> The sequence for the first alternate allele. This is a convenience for the common case where 
 the first alternate allele is the only one of interest. In general you should be aware that
 more than one alternate Allele represented by a single Variant.
     </p>
                            </li>
                        </ul>
                    
                        <a name="altByte"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>byte <strong>altByte</strong></h4>
                                <p> The first character of the first alternate allele in byte form. This is a special case
 optimization to allow for fast access when interoperating with Picard (which wants to see
 bytes, not char or Strings).
     </p>
                            </li>
                        </ul>
                    
                        <a name="alts"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String[] <strong>alts</strong></h4>
                                <p> List of alternate alleles represented by the variant.
     </p>
                            </li>
                        </ul>
                    
                        <a name="cachedRegion"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../gngs/Region.html'>Region</a> <strong>cachedRegion</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="chr"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>chr</strong></h4>
                                <p> The chromosome on which the variant falls
     </p>
                            </li>
                        </ul>
                    
                        <a name="dosages"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.Integer&gt; <strong>dosages</strong></h4>
                                <p> Cached set of dosages for the first allele.
 <p>
 <li> 0 = hom ref
 <li> 1 = het
 <li> 2 = hom alt
 
 There is one entry in the list for each sample, in the same order
 as the samples in the VCF header.
     </p>
                            </li>
                        </ul>
                    
                        <a name="filter"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>filter</strong></h4>
                                <p> The filter field as loaded from the VCF file.
     </p>
                            </li>
                        </ul>
                    
                        <a name="genoTypeFields"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String[] <strong>genoTypeFields</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="genoTypes"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.util.Map&lt;java.lang.String, java.lang.Object&gt;&gt; <strong>genoTypes</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="header"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../gngs/VCF.html'>VCF</a> <strong>header</strong></h4>
                                <p> The VCF from which header information will be extracted when required. This enables the variant to 
 intelligently parse INFO fields and be aware of sample names. If the header is not provided,
 many functions still work, but some functions will be disabled.
     </p>
                            </li>
                        </ul>
                    
                        <a name="id"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>id</strong></h4>
                                <p> ID column as per VCF spec. Usually this contains the rsID (dbSNP) identifer
     </p>
                            </li>
                        </ul>
                    
                        <a name="info"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>info</strong></h4>
                                <p> The whole INFO field as a raw string. Users will generally call <a href='#getInfo()'>getInfo()</a> to get
 this field in parsed form rather than accessing this field directly.
     </p>
                            </li>
                        </ul>
                    
                        <a name="infos"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Map&lt;java.lang.String, java.lang.Object&gt; <strong>infos</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="line"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>line</strong></h4>
                                <p> The original line from which this variant was parsed
     </p>
                            </li>
                        </ul>
                    
                        <a name="pedigrees"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Set&lt;<a href='../gngs/Pedigree.html' title='Pedigree'>Pedigree</a>&gt; <strong>pedigrees</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="pos"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>int <strong>pos</strong></h4>
                                <p> The position at which the reference bases (<a href='#ref'>ref</a> attribute) starts.
 Note that this is <i>not</i> necessarily the start of the genetic change represented
 by the variant.
     </p>
                            </li>
                        </ul>
                    
                        <a name="qual"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>float <strong>qual</strong></h4>
                                <p> The QUAL field as loaded from the VCF file
     </p>
                            </li>
                        </ul>
                    
                        <a name="ref"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>ref</strong></h4>
                                <p> The reference allele at the position. Note that as per VCF spec, this is not necessarily
 the start of the change indicated by the Variant. For the start of actual changes,
 you need to query the <a href='../gngs/Allele.html' title='Allele'>Allele</a> objects and look at the <a href='../gngs/Allele.html#start' title='Allele.start'>Allele.start</a> and
 <a href='../gngs/Allele.html#end' title='Allele.end'>Allele.end</a> attributes.
     </p>
                            </li>
                        </ul>
                    
                        <a name="snpEffDirty"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>boolean <strong>snpEffDirty</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="snpEffInfo"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;<a href='../gngs/SnpEffInfo.html' title='SnpEffInfo'>SnpEffInfo</a>&gt; <strong>snpEffInfo</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="type"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>type</strong></h4>
                                <p> One of SNP, INS, DEL or SV indicating the type of change represented by the first 
 alternate allele. 
     </p>
                            </li>
                        </ul>
                    
                </li>
            </ul>
            

            

            


            
            <!-- =========== METHOD DETAIL =========== -->
            <ul class="blockList">
                <li class="blockList"><a name="method_detail">
                    <!--   -->
                </a>
                    <h3>Method Detail</h3>
                    
                        <a name="applyDEL(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>applyDEL</strong>(java.lang.String sequence, int position)</h4>
                                <p> Apply this deletion to the specified sequence at the specified position
 <p>NOTE: this variant MUST be a deletion
     <DL><DT><B>Returns:</B></DT><DD>given sequence mutated to include this deletion</DD></DL><DL><DT><B>Parameters:</B></DT><DD>sequence</DD><DD>position</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="applyINS(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>applyINS</strong>(java.lang.String sequence, int position)</h4>
                                <p> Apply this deletion to the specified sequence at the specified position
 <p>NOTE: this variant MUST be a deletion
     <DL><DT><B>Returns:</B></DT><DD>given sequence mutated to include this deletion</DD></DL><DL><DT><B>Parameters:</B></DT><DD>sequence</DD><DD>position</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="applySNP(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>applySNP</strong>(java.lang.String sequence, int position)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="applyTo(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>applyTo</strong>(java.lang.String sequence, int position)</h4>
                                <p> Apply this variant to mutate the given sequence at the specified position
     </p>
                            </li>
                        </ul>
                    
                        <a name="convertType(java.lang.String, java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.lang.String <strong>convertType</strong>(java.lang.String refSeq, java.lang.String altSeq)</h4>
                                <p> Convert reference and alternate allele strings into a 
 mutation type, being one of "SNP","INS","DEL","GAIN"
 or "LOSS", with the latter two representing CNVs. <DL><DT><B>Returns:</B></DT><DD></DD></DL><DL><DT><B>Parameters:</B></DT><DD>refSeq</DD><DD>altSeq</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="displaySequenceDifference(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>displaySequenceDifference</strong>(java.lang.String sequence, int position)</h4>
                                <p> A debug function, displays the sequence difference and returns the 
 new sequence
     </p>
                            </li>
                        </ul>
                    
                        <a name="equalsAnnovar(java.lang.String, int, java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>equalsAnnovar</strong>(java.lang.String chr, int pos, java.lang.String obs)</h4>
                                <p> Return true if the given Annovar formattd position and observation
 match those of this variant
 <p>
 Annovar outputs non-standard formatting that makes it difficult to trace
 an Annovar variant back to it's VCF source. This method implements
 the tricky logic to compare an Annovar variant to a VCF equivalent
 and say if they are the same.
     <DL><DT><B>Returns:</B></DT><DD>if the Annovar variant is not the same variant as this VCF variant,
         returns 0. If it is the same, returns the number of the alternate
         allele (if the variant has only one alternate, the return value will 
         be 1).</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="findAlleleIndex(gngs.Allele)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>int <strong>findAlleleIndex</strong>(<a href='../gngs/Allele.html'>Allele</a> other)</h4>
                                <p> Return the index of the allele (if any) that matches the give other variant <DL><DT><B>Returns:</B></DT><DD></DD></DL><DL><DT><B>Parameters:</B></DT><DD>other</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="fixGenoTypeOrder()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>fixGenoTypeOrder</strong>()</h4>
                                <p> VCF requires genotype to be the 1st field, but some tools (R, grrrr) write it in whatever order they 
 feel like. This function corrects it.
     </p>
                            </li>
                        </ul>
                    
                        <a name="getAlleleBalance()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>float <strong>getAlleleBalance</strong>()</h4>
                                <p> Return the distance of the balance of the default allele from 0.5
 for the default alternate allele and first sample <DL><DT><B>Returns:</B></DT><DD></DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getAlleleBalance(int, int, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>float <strong>getAlleleBalance</strong>(int allele1, int allele2, int sampleIndex = 0)</h4>
                                <p> Return the distance of the balance of the default allele from 0.5
 for the specififed alternate allele (reference = 0) <DL><DT><B>Returns:</B></DT><DD></DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getAlleleDepths(int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.Integer&gt; <strong>getAlleleDepths</strong>(int alleleIndex)</h4>
                                <p> Return the number of reads supporting the given allele as a list
 with one entry for each sample in the VCF
     <DL><DT><B>Parameters:</B></DT><DD><code>alleleIndex</code> -    index of allele, reference = 0</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getAlleles()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;<a href='../gngs/Allele.html' title='Allele'>Allele</a>&gt; <strong>getAlleles</strong>()</h4>
                                <p> Return a list of Allele objects representing alleles present on this line of the VCF
 <p>
 Note: currently this computes the results on the fly and they are not cached, so
       use with caution in computationaly intensive situations
     </p>
                            </li>
                        </ul>
                    
                        <a name="getAllelesAndTypes()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt; <strong>getAllelesAndTypes</strong>()</h4>
                                <p> Return a list of each allele and its type.
 Yes, VCF allows multiple types (INS,DEL) to be on the same line of 
 a VCF file.<DL><DT><B>deprecated:</B></DT><DD>use getAlleles() instead</DD></DL>
     </p>
                            </li>
                        </ul>
                    
                        <a name="getConsequence(int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>getConsequence</strong>(int alleleIndex)</h4>
                                <p> Return the consequence of the specified allele, at the moment,
 from VEP annotations (later, from others). If multiple consequences
 are present for the same allele, then the most severe consequence is 
 returned.
     </p>
                            </li>
                        </ul>
                    
                        <a name="getDosages()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.Integer&gt; <strong>getDosages</strong>()</h4>
                                <p> Return list of dosages (number of copies of allele) for 
 each sample for the first alternate allele. <DL><DT><B>Returns:</B></DT><DD></DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getDosages(int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.Integer&gt; <strong>getDosages</strong>(int alleleIndex)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="getGenes(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.String&gt; <strong>getGenes</strong>(java.lang.String minVEPCons)</h4>
                                <p> Return the list of genes impacted by this variant, in a manner that is
 neutral to the annotator used (VEP and SnpEFF supported)
     <DL><DT><B>Parameters:</B></DT><DD><code>minVEPCons</code> -   the VEP impact level above which genes should
                      be returned</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getGenotypeDepth(Map<String, Object>, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>getGenotypeDepth</strong>(java.util.Map&lt;java.lang.String, java.lang.Object&gt; gt, int alleleIndex)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="getInfo()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.util.Map&lt;java.lang.String, java.lang.Object&gt; <strong>getInfo</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="getMaxAlleleBalance()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>float <strong>getMaxAlleleBalance</strong>()</h4>
                                <p> Return the maximum allele balance for any sample for the default
 alternate allele. <DL><DT><B>Returns:</B></DT><DD></DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getMaxEffect()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../gngs/SnpEffInfo.html'>SnpEffInfo</a> <strong>getMaxEffect</strong>()</h4>
                                <p> Return the most impactful SnpEff effect for this variant
     </p>
                            </li>
                        </ul>
                    
                        <a name="getMaxVep()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Map&lt;java.lang.String, java.lang.Object&gt; <strong>getMaxVep</strong>()</h4>
                                <p> Return the details of the most severe VEP consequence, as ranked by
 <a href='../gngs/VEPConsequences.html#RANKED_CONSEQUENCES' title='VEPConsequences.RANKED_CONSEQUENCES'>VEPConsequences.RANKED_CONSEQUENCES</a>.
     <DL><DT><B>Returns:</B></DT><DD>A map with key value pairs of VEP annotation fields.</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getMaxVepImpact()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>getMaxVepImpact</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="getMaxVepMaf()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
float <strong>getMaxVepMaf</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="getPedigrees()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Set&lt;<a href='../gngs/Pedigree.html' title='Pedigree'>Pedigree</a>&gt; <strong>getPedigrees</strong>()</h4>
                                <p> Return a list of all the pedigrees that contain this variant
     </p>
                            </li>
                        </ul>
                    
                        <a name="getRange()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>groovy.lang.IntRange <strong>getRange</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="getRegion()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
<a href='../gngs/Region.html'>Region</a> <strong>getRegion</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="getSnpEffInfo()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;<a href='../gngs/SnpEffInfo.html' title='SnpEffInfo'>SnpEffInfo</a>&gt; <strong>getSnpEffInfo</strong>()</h4>
                                <p> Return a list of SnpEffInfo objects, each describing
 a separate SnpEff effect caused by the variant <DL><DT><B>Returns:</B></DT><DD></DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="getTotalDepth(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
java.lang.Integer <strong>getTotalDepth</strong>(java.lang.String sample)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="getVepInfo()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.util.Map&lt;java.lang.String, java.lang.Object&gt;&gt; <strong>getVepInfo</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="isCase(gngs.IRegion)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>boolean <strong>isCase</strong>(<a href='../gngs/IRegion.html'>IRegion</a> r)</h4>
                                <p> Return true if this variant overlaps the given range
     </p>
                            </li>
                        </ul>
                    
                        <a name="isSV()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
boolean <strong>isSV</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="parse(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
static&nbsp;<a href='../gngs/Variant.html'>Variant</a> <strong>parse</strong>(java.lang.String line)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="parse(java.lang.String, boolean)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
static&nbsp;<a href='../gngs/Variant.html'>Variant</a> <strong>parse</strong>(java.lang.String line, boolean ignoreNonRef)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="parseGenoTypeFields(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Map&lt;java.lang.String, java.lang.Object&gt; <strong>parseGenoTypeFields</strong>(java.lang.String gt)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="sampleDosage(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>sampleDosage</strong>(java.lang.String sampleName)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="sampleDosage(java.lang.String, int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>sampleDosage</strong>(java.lang.String sampleName, int alleleIndex)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="sampleGenoType(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic 
    /**
     * Return a Map of fields from the genotype column corresponding
     * to the given sample. Common fields include:
     * <li>AD - allele depth (list of numeric values)
     * <li>GQ - genotype qualities for each genotype (string value)
     * <li>GT - list of actual genotypes for sample for each allele (string)
     * <li>DP - total depth (integer)
     * @param sampleName
     */
java.util.Map <strong>sampleGenoType</strong>(java.lang.String sampleName)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="segregatesWith(gngs.Pedigree)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>boolean <strong>segregatesWith</strong>(<a href='../gngs/Pedigree.html'>Pedigree</a> ped)</h4>
                                <p> Return a Map of fields from the genotype column corresponding
 to the given sample. Common fields include:
 <li>AD - allele depth (list of numeric values)
 <li>GQ - genotype qualities for each genotype (string value)
 <li>GT - list of actual genotypes for sample for each allele (string)
 <li>DP - total depth (integer)
     <DL><DT><B>Parameters:</B></DT><DD>sampleName</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="setAlt(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>setAlt</strong>(java.lang.String alt)</h4>
                                <p> Update the first alternate allele to the given value
     <DL><DT><B>Parameters:</B></DT><DD><code>alt</code> -    Alternate sequence of bases. Note that this should conform to VCF 
              spec, eg: for deletions it is expected at least one base of context 
              should be provided, and this should be consistent with the <a href='#pos'>pos</a>
              field.</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="size()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
int <strong>size</strong>()</h4>
                                <p> Returns the change in size of the genome caused by this variant's default allele.
 <p>
 Note: since point mutations / SNVs don't change the size, they have size zero
     <DL><DT><B>Returns:</B></DT><DD>difference in genome size caused by this variant's default allele</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="toAnnovar(int)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Map <strong>toAnnovar</strong>(int alleleIndex = 0)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="toJson(java.lang.String)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>toJson</strong>(java.lang.String sample = null)</h4>
                                <p> Return a JSON string representing the key details about this variant.
 <b>Note:</b>
 <li>If no sample is provided then the first sample is assumed
 <li>The default information such as allele depths refers to the 
     first alternate allele.
     <DL><DT><B>Returns:</B></DT><DD>JSON string representing the variant</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="toString()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>toString</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="update(groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>update</strong>(groovy.lang.Closure c)</h4>
                                <p> Update using a default description. 
 <p>
 Call this method to prepare the Variant for updating and
 then make your updates within the Closure that you pass. After 
 the closure exits, other internal fields that are impacted by your changes
 will be synchronized with the changes you made.
 <p>
 Not all fields are supported! see <a href='#update(java.lang.String, groovy.lang.Closure)'>update(String, Closure)</a>
 <p>
 It's generally bad form to add INFO fields without annotating what you've done.
 Prefer to use <a href='#update(java.lang.String, groovy.lang.Closure)'>update(String, Closure)</a> and pass in a description
 over this method.
     <DL><DT><B>Parameters:</B></DT><DD><code>c</code> -      Closure within updates can be made.</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="update(java.lang.String, groovy.lang.Closure)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>update</strong>(java.lang.String desc, groovy.lang.Closure c)</h4>
                                <p> Allows various fields to be updated and then synchronises the 
 rest of the data with those updated fields
 <p>
 Call this method to prepare the Variant for updating and
 then make your updates within the Closure that you pass. After 
 the closure exits, other internal fields that are impacted by your changes
 will be synchronized with the changes you made. 
 <p>
 Not all fields are supported! see the ones that are set below.
 <p>
 The only update to snpEFF information is to remove individual 
 annotations.
     </p>
                            </li>
                        </ul>
                    
                </li>
            </ul>
            
        </li>
    </ul>
</div>

<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
    <!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="../overview-summary.html">Overview</a></li>
        <li><a href="package-summary.html">Package</a></li>
        <li class="navBarCell1Rev">Class</li>
        <li><a href="../deprecated-list.html">Deprecated</a></li>
        <li><a href="../index-all.html">Index</a></li>
        <li><a href="../help-doc.html">Help</a></li>
    </ul>
</div>

<div class="subNav">
    <div>
        <ul class="navList">
            <li><a href="../index.html?gngs/Variant" target="_top">Frames</a></li>
            <li><a href="Variant.html" target="_top">No Frames</a></li>
        </ul>
    </div>
    <div>
        <ul class="subNavList">
            <li>Summary:&nbsp;</li>
            Nested&nbsp;&nbsp;&nbsp;Field&nbsp;&nbsp;&nbsp;<li><a href="#property_summary">Property</a></li>&nbsp;&nbsp;&nbsp;Constructor&nbsp;&nbsp;&nbsp;<li><a href="#method_summary">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
        <ul class="subNavList">
            <li>&nbsp;|&nbsp;Detail:&nbsp;</li>
            Field&nbsp;&nbsp;&nbsp;<li><a href="#prop_detail">Property</a></li>&nbsp;&nbsp;&nbsp;Constructor&nbsp;&nbsp;&nbsp;<li><a href="#method_detail">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
    </div>
    <p>Groovy Documentation</p>
    <a name="skip-navbar_bottom">
        <!--   -->
    </a>
    </div>
</div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
