<!--

     Licensed to the Apache Software Foundation (ASF) under one
     or more contributor license agreements.  See the NOTICE file
     distributed with this work for additional information
     regarding copyright ownership.  The ASF licenses this file
     to you under the Apache License, Version 2.0 (the
     "License"); you may not use this file except in compliance
     with the License.  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

     Unless required by applicable law or agreed to in writing,
     software distributed under the License is distributed on an
     "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
     KIND, either express or implied.  See the License for the
     specific language governing permissions and limitations
     under the License.

-->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- **************************************************************** -->
<!-- *  PLEASE KEEP COMPLICATED EXPRESSIONS OUT OF THESE TEMPLATES, * -->
<!-- *  i.e. only iterate & print data where possible. Thanks, Jez. * -->
<!-- **************************************************************** -->

<html>
<head>
    <!-- Generated by groovydoc (2.5.6) on Sat Jan 25 15:54:01 AEDT 2020 -->
    <title>CoverageCalculatorActor (groovy-ngs-utils API)</title>
    <meta name="date" content="2020-01-25">
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <link href="../../groovy.ico" type="image/x-icon" rel="shortcut icon">
    <link href="../../groovy.ico" type="image/x-icon" rel="icon">
    <link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">

<body class="center">
<script type="text/javascript"><!--
if (location.href.indexOf('is-external=true') == -1) {
    parent.document.title="CoverageCalculatorActor (groovy-ngs-utils API)";
}
//-->
</script>
<noscript>
    <div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
    <!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="../../overview-summary.html">Overview</a></li>
        <li><a href="package-summary.html">Package</a></li>
        <li class="navBarCell1Rev">Class</li>
        <li><a href="../../deprecated-list.html">Deprecated</a></li>
        <li><a href="../../index-all.html">Index</a></li>
        <li><a href="../../help-doc.html">Help</a></li>
    </ul>
</div>

<div class="subNav">
    <div>
        <ul class="navList">
            <li><a href="../../index.html?gngs/coverage/CoverageCalculatorActor" target="_top">Frames</a></li>
            <li><a href="CoverageCalculatorActor.html" target="_top">No Frames</a></li>
        </ul>
    </div>
    <div>
        <ul class="subNavList">
            <li>Summary:&nbsp;</li>
            Nested&nbsp;&nbsp;&nbsp;Field&nbsp;&nbsp;&nbsp;<li><a href="#property_summary">Property</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#constructor_summary">Constructor</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#method_summary">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
        <ul class="subNavList">
            <li>&nbsp;|&nbsp;Detail:&nbsp;</li>
            Field&nbsp;&nbsp;&nbsp;<li><a href="#prop_detail">Property</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#constructor_detail">Constructor</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#method_detail">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
    </div>
    <a name="skip-navbar_top">
        <!--   -->
    </a></div>
<!-- ========= END OF TOP NAVBAR ========= -->

<!-- ======== START OF CLASS DATA ======== -->
<div class="header">

    <div class="subTitle">Package: <strong>gngs.coverage</strong></div>

    <h2 title="[Groovy] Class CoverageCalculatorActor" class="title">[Groovy] Class CoverageCalculatorActor</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><ul class="inheritance"></ul></li><li><ul class="inheritance"></ul></li><li><ul class="inheritance"></ul></li><li><ul class="inheritance"></ul></li><li><ul class="inheritance"></ul></li><li><ul class="inheritance"></ul></li><li><ul class="inheritance"></ul></li><li>gngs.coverage.CoverageCalculatorActor
</ul>
<div class="description">
    <ul class="blockList">
        <li class="blockList">



    <p> An actor that accepts coordinate ordered reads and transforms them into
 the coverage (number of overlapping reads at a position) over a set of 
 pre-defined regions. The coverage values are emitted as messages to 
 a downstream consumer as SampleReadCount objects, one for each position
 in the target range.
 <p>
 The logic used is that since the reads are coordinate sorted, coverage
 depth can be computed in a streaming manner by storing a buffer of reads 
 ordered by position.
 <p>
 When a new read is considered, all the positions from the previous written position
 to the position before the start of the read can be written out. As each such position
 is processed, all reads that end prior to the position should be dropped from 
 the buffer. The number of reads in the buffer then by definition all overlap the
 position being written.
 <p>
 The above is the conceptual algorithm, but several refinements are implemented to
 improve computational / memory efficiency, handling of paired reads etc.
 <DL><DT><B>Authors:</B></DT><DD>Simon Sadedin</DD></DL></p>

          </li>
    </ul>
</div>

<div class="summary">
    <ul class="blockList">
        <li class="blockList">
        <!-- =========== NESTED CLASS SUMMARY =========== -->
        

        <!-- =========== ENUM CONSTANT SUMMARY =========== -->
        
        <!-- =========== FIELD SUMMARY =========== -->
        
            <ul class="blockList">
                <li class="blockList"><a name="field_summary"><!--   --></a>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Inherited fields summary table">
                        <caption><span>Inherited fields</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Fields inherited from class</th>
                            <th class="colLast" scope="col">Fields</th>
                        </tr>
                                                <tr class="altColor">
                            <td class="colFirst"><strong><code>class groovyx.gpars.actor.DefaultActor</code></strong></td>
                            <td class="colLast"><code>CANNOT_SEND_REPLIES_NO_SENDER_HAS_BEEN_REGISTERED, TIMEOUT, serialHandle</code></td>
                        </tr>

                    </table>
                    </ul>
                </li>
                
            </ul>
        

        <!-- =========== PROPERTY SUMMARY =========== -->
        
         <ul class="blockList">
                <li class="blockList"><a name="property_summary"><!--   --></a>
                    <h3>Properties Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Properties Summary table, listing nested classes, and an explanation">
                        <caption><span>Properties</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Type</th>
                            <th class="colLast" scope="col">Name and description</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.util.List&lt;java.lang.String&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#bamContigs">bamContigs</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.util.Set&lt;java.lang.Integer&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#bedReferenceIndexes">bedReferenceIndexes</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>boolean</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#countFragments">countFragments</a></code><br>If true, fragments will be counted rather than reads</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#currentChr">currentChr</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong><a href='../../gngs/IntervalGCCache.html' title='IntervalGCCache'>IntervalGCCache</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#currentGcCache">currentGcCache</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong><a href='../../gngs/coverage/ReadRange.html'>ReadRange</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#currentRead">currentRead</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>int</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#currentReferenceIndex">currentReferenceIndex</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong><a href='../../gngs/Region.html'>Region</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#currentRegion">currentRegion</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>int</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#flushingReferenceIndex">flushingReferenceIndex</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.util.List&lt;java.lang.Double&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#gcProfile">gcProfile</a></code><br>List of gc content adjustment factors to apply to reads.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong><a href='../../gngs/FASTA.html'>FASTA</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#gcReference">gcReference</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.util.Map&lt;java.lang.String, <a href='../../gngs/IntervalGCCache.html' title='IntervalGCCache'>IntervalGCCache</a>&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#intervalGCCaches">intervalGCCaches</a></code><br>Caches of GC values for intervals to allow for efficient querying, keyed on chromosome</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.util.concurrent.atomic.AtomicInteger</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#pending">pending</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>int</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#pos">pos</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong><a href='../../gngs/ProgressCounter.html'>ProgressCounter</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#progress">progress</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong><a href='../../gngs/OverlapTracker.html'>OverlapTracker</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#reads">reads</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong>java.util.Iterator&lt;<a href='../../gngs/Region.html' title='Region'>Region</a>&gt;</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#regionIter">regionIter</a></code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code><strong>java.lang.String</strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#sample">sample</a></code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code><strong><a href='../../gngs/Regions.html'>Regions</a></strong></code>&nbsp;</td>
                            <td class="colLast"><code><a href="#targetRegions">targetRegions</a></code><br></td>
                        </tr>
                        
                    </table>
                   </ul>
                </li>
            </ul>
        

        <!-- =========== ELEMENT SUMMARY =========== -->
        

        
        <!-- ======== CONSTRUCTOR SUMMARY ======== -->
        <ul class="blockList">
                <li class="blockList"><a name="constructor_summary"><!--   --></a>
                    <h3>Constructor Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructors Summary table">
                        <caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Constructor and description</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst">
                                <code><a href="#CoverageCalculatorActor(gngs.SAM, gngs.Regions, groovyx.gpars.actor.Actor, java.lang.String, boolean)">CoverageCalculatorActor</a>
                                (<a href='../../gngs/SAM.html'>SAM</a> bam, <a href='../../gngs/Regions.html'>Regions</a> targetRegions, groovyx.gpars.actor.Actor combiner, java.lang.String sample, boolean countFragments = true)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst">
                                <code><a href="#CoverageCalculatorActor(List<String>, gngs.Regions, groovyx.gpars.actor.Actor, java.lang.String, boolean)">CoverageCalculatorActor</a>
                                (java.util.List&lt;java.lang.String&gt; allContigs, <a href='../../gngs/Regions.html'>Regions</a> targetRegions, groovyx.gpars.actor.Actor combiner, java.lang.String sample, boolean countFragments)</code><br></td>
                        </tr>
                        
                    </table>
                   </ul>
              </li>
        </ul>
        

        
        <!-- ========== METHOD SUMMARY =========== -->
        <ul class="blockList">
            
            <li class="blockList"><a name="method_summary"><!--   --></a>
                    <h3>Methods Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Methods Summary table">
                        <caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Type Params</th>
                            <th class="colLast" scope="col">Return Type</th>
                            <th class="colLast" scope="col">Name and description</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>static&nbsp;graxxia.Matrix</code></td>
                            <td class="colLast"><code><strong><a href="#calculateGCAdjustmentFactors(java.util.List)">calculateGCAdjustmentFactors</a></strong>(java.util.List profiles)</code><br>Given the list of GC bins with relative coverage (vs mean) for each bin, calculate a matrix
 that gives, for each sample and each bin, the multiplier that would equalise the coverage in the
 bin to the mean of other samples for that bin.</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#dropNonOverlapping()">dropNonOverlapping</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#enableGCNormalisation(List<Double>, gngs.FASTA, Map<String, IntervalGCCache>)">enableGCNormalisation</a></strong>(java.util.List&lt;java.lang.Double&gt; gcProfile, <a href='../../gngs/FASTA.html'>FASTA</a> fasta, java.util.Map&lt;java.lang.String, <a href='../../gngs/IntervalGCCache.html' title='IntervalGCCache'>IntervalGCCache</a>&gt; caches)</code><br>Enables GC normalisation based on pre-computed weightings for the given GC bins
 (which are assumed to span 0-100% GC content). </td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#iteratorRemove(java.util.Iterator)">iteratorRemove</a></strong>(java.util.Iterator i)</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>boolean</code></td>
                            <td class="colLast"><code><strong><a href="#nextRegion()">nextRegion</a></strong>()</code><br>Transition to the next target region, taking into account that it might 
 also cause transition to next reference sequence (chromosome).</td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#onEnd()">onEnd</a></strong>()</code><br></td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>void</code></td>
                            <td class="colLast"><code><strong><a href="#process(gngs.coverage.ReadRange)">process</a></strong>(<a href='../../gngs/coverage/ReadRange.html'>ReadRange</a> r)</code><br></td>
                        </tr>
                        
                        <tr class="rowColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>static&nbsp;void</code></td>
                            <td class="colLast"><code><strong><a href="#processBAM(gngs.SAM, gngs.Regions, gngs.RegulatingActor, int, java.lang.String, boolean, graxxia.Matrix)">processBAM</a></strong>(<a href='../../gngs/SAM.html'>SAM</a> bam, <a href='../../gngs/Regions.html'>Regions</a> scanRegions, <a href='../../gngs/RegulatingActor.html'>RegulatingActor</a> downstream, int minMQ, java.lang.String sample = null, boolean countFragments = true, graxxia.Matrix gcAdjustments = null)</code><br>Create a CoverageCalculatorActor and send all the reads from the given alignment that
 overlap the given regions to it.</td>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code></code></td>
                            <td class="colLast"><code>java.lang.String</code></td>
                            <td class="colLast"><code><strong><a href="#toString()">toString</a></strong>()</code><br></td>
                        </tr>
                        
                    </table>
                   </ul>
              </li>
            
            <li class="blockList"><a name="method_summary"><!--   --></a>
                    <h3>Inherited Methods Summary</h3>
                    <ul class="blockList">
                    <table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Inherited Methods Summary table">
                        <caption><span>Inherited Methods</span><span class="tabEnd">&nbsp;</span></caption>
                        <tr>
                            <th class="colFirst" scope="col">Methods inherited from class</th>
                            <th class="colLast" scope="col">Name</th>
                        </tr>
                        
                        <tr class="altColor">
                            <td class="colFirst"><code>class <a href='../../gngs/RegulatingActor.html'>RegulatingActor</a></code></td>
                            <td class="colLast"><code><a href='../../gngs/RegulatingActor.html#act()'>act</a>, <a href='../../gngs/RegulatingActor.html#actor(Closure)'>actor</a>, <a href='../../gngs/RegulatingActor.html#doTerminate()'>doTerminate</a>, <a href='../../gngs/RegulatingActor.html#onEnd()'>onEnd</a>, <a href='../../gngs/RegulatingActor.html#process(T)'>process</a>, <a href='../../gngs/RegulatingActor.html#run()'>run</a>, <a href='../../gngs/RegulatingActor.html#send(gngs.AcknowledgeableMessage)'>send</a>, <a href='../../gngs/RegulatingActor.html#sendDownstream(java.lang.Object)'>sendDownstream</a>, <a href='../../gngs/RegulatingActor.html#sendLimited(gngs.AcknowledgeableMessage, boolean)'>sendLimited</a>, <a href='../../gngs/RegulatingActor.html#sendStop()'>sendStop</a>, <a href='../../gngs/RegulatingActor.html#sendTo(T)'>sendTo</a></code></td>
                        </tr>
                        <tr class="altColor">
                            <td class="colFirst"><code>class groovyx.gpars.actor.DefaultActor</code></td>
                            <td class="colLast"><code>groovyx.gpars.actor.DefaultActor#start(), groovyx.gpars.actor.DefaultActor#react(groovy.lang.Closure), groovyx.gpars.actor.DefaultActor#silentStart(), groovyx.gpars.actor.DefaultActor#loop(java.lang.Runnable), groovyx.gpars.actor.DefaultActor#stop(), groovyx.gpars.actor.DefaultActor#send(java.lang.Object), groovyx.gpars.actor.DefaultActor#makeFair(), groovyx.gpars.actor.DefaultActor#terminate(), groovyx.gpars.actor.DefaultActor#setParallelGroup(groovyx.gpars.group.PGroup), groovyx.gpars.actor.DefaultActor#getSender(), groovyx.gpars.actor.DefaultActor#replyIfExists(java.lang.Object), groovyx.gpars.actor.DefaultActor#isActive(), groovyx.gpars.actor.DefaultActor#reply(java.lang.Object), groovyx.gpars.actor.DefaultActor#isFair(), groovyx.gpars.actor.DefaultActor#join(), groovyx.gpars.actor.DefaultActor#join(long, java.util.concurrent.TimeUnit), groovyx.gpars.actor.DefaultActor#join(groovy.time.BaseDuration), groovyx.gpars.actor.DefaultActor#join(groovyx.gpars.actor.impl.MessageStream), groovyx.gpars.actor.DefaultActor#isActorThread(), groovyx.gpars.actor.DefaultActor#getJoinLatch(), groovyx.gpars.actor.DefaultActor#getParallelGroup(), groovyx.gpars.actor.DefaultActor#sendAndContinue(java.lang.Object, groovy.lang.Closure), groovyx.gpars.actor.DefaultActor#sendAndPromise(java.lang.Object), groovyx.gpars.actor.DefaultActor#threadBoundActor(), groovyx.gpars.actor.DefaultActor#onStop(groovy.lang.Closure), groovyx.gpars.actor.DefaultActor#send(java.lang.Object, groovyx.gpars.actor.impl.MessageStream), groovyx.gpars.actor.DefaultActor#send(), groovyx.gpars.actor.DefaultActor#sendAndWait(java.lang.Object), groovyx.gpars.actor.DefaultActor#sendAndWait(java.lang.Object, groovy.time.Duration), groovyx.gpars.actor.DefaultActor#sendAndWait(java.lang.Object, long, java.util.concurrent.TimeUnit), groovyx.gpars.actor.DefaultActor#getRemoteClass(), groovyx.gpars.actor.DefaultActor#call(java.lang.Object), groovyx.gpars.actor.DefaultActor#leftShift(java.lang.Object), groovyx.gpars.actor.DefaultActor#getOrCreateSerialHandle(), groovyx.gpars.actor.DefaultActor#wait(long, int), groovyx.gpars.actor.DefaultActor#wait(long), groovyx.gpars.actor.DefaultActor#wait(), groovyx.gpars.actor.DefaultActor#equals(java.lang.Object), groovyx.gpars.actor.DefaultActor#toString(), groovyx.gpars.actor.DefaultActor#hashCode(), groovyx.gpars.actor.DefaultActor#getClass(), groovyx.gpars.actor.DefaultActor#notify(), groovyx.gpars.actor.DefaultActor#notifyAll()</code></td>
                        </tr>
                    </table>
                   </ul>
              </li>
                
        </ul>
        
    </li>
    </ul>
</div>

<div class="details">
    <ul class="blockList">
        <li class="blockList">
           

            

            
            <!-- =========== PROPERTY DETAIL =========== -->
            <ul class="blockList">
                <li class="blockList"><a name="prop_detail">
                    <!--   -->
                </a>
                    <h3>Property Detail</h3>
                    
                        <a name="bamContigs"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.String&gt; <strong>bamContigs</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="bedReferenceIndexes"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Set&lt;java.lang.Integer&gt; <strong>bedReferenceIndexes</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="countFragments"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>boolean <strong>countFragments</strong></h4>
                                <p> If true, fragments will be counted rather than reads
 <p>
 Counting reads over-estimates the biological sampling, but is concordant with 
 other tools such as BEDTools etc.
     </p>
                            </li>
                        </ul>
                    
                        <a name="currentChr"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>currentChr</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="currentGcCache"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../../gngs/IntervalGCCache.html' title='IntervalGCCache'>IntervalGCCache</a> <strong>currentGcCache</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="currentRead"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../../gngs/coverage/ReadRange.html'>ReadRange</a> <strong>currentRead</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="currentReferenceIndex"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>int <strong>currentReferenceIndex</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="currentRegion"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../../gngs/Region.html'>Region</a> <strong>currentRegion</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="flushingReferenceIndex"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>int <strong>flushingReferenceIndex</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="gcProfile"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.List&lt;java.lang.Double&gt; <strong>gcProfile</strong></h4>
                                <p> List of gc content adjustment factors to apply to reads.
 <p>
 The list is interpreted as fully spanning the range 0 - 1.0 with equal size bins.
 The coverage attributed to each read is weighted according to the GC bin within which
 the read falls.
     </p>
                            </li>
                        </ul>
                    
                        <a name="gcReference"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../../gngs/FASTA.html'>FASTA</a> <strong>gcReference</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="intervalGCCaches"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Map&lt;java.lang.String, <a href='../../gngs/IntervalGCCache.html' title='IntervalGCCache'>IntervalGCCache</a>&gt; <strong>intervalGCCaches</strong></h4>
                                <p> Caches of GC values for intervals to allow for efficient querying, keyed on chromosome
     </p>
                            </li>
                        </ul>
                    
                        <a name="pending"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.concurrent.atomic.AtomicInteger <strong>pending</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="pos"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>int <strong>pos</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="progress"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../../gngs/ProgressCounter.html'>ProgressCounter</a> <strong>progress</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="reads"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>final&nbsp;<a href='../../gngs/OverlapTracker.html'>OverlapTracker</a> <strong>reads</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="regionIter"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.util.Iterator&lt;<a href='../../gngs/Region.html' title='Region'>Region</a>&gt; <strong>regionIter</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="sample"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>sample</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="targetRegions"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><a href='../../gngs/Regions.html'>Regions</a> <strong>targetRegions</strong></h4>
                                <p></p>
                            </li>
                        </ul>
                    
                </li>
            </ul>
            

            

            
            <!-- =========== CONSTRUCTOR DETAIL =========== -->
            <ul class="blockList">
                <li class="blockList"><a name="constructor_detail">
                    <!--   -->
                </a>
                    <h3>Constructor Detail</h3>
                    
                        <a name="CoverageCalculatorActor(gngs.SAM, gngs.Regions, groovyx.gpars.actor.Actor, java.lang.String, boolean)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><strong>CoverageCalculatorActor</strong>(<a href='../../gngs/SAM.html'>SAM</a> bam, <a href='../../gngs/Regions.html'>Regions</a> targetRegions, groovyx.gpars.actor.Actor combiner, java.lang.String sample, boolean countFragments = true)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="CoverageCalculatorActor(List<String>, gngs.Regions, groovyx.gpars.actor.Actor, java.lang.String, boolean)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4><strong>CoverageCalculatorActor</strong>(java.util.List&lt;java.lang.String&gt; allContigs, <a href='../../gngs/Regions.html'>Regions</a> targetRegions, groovyx.gpars.actor.Actor combiner, java.lang.String sample, boolean countFragments)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                </li>
            </ul>
            


            
            <!-- =========== METHOD DETAIL =========== -->
            <ul class="blockList">
                <li class="blockList"><a name="method_detail">
                    <!--   -->
                </a>
                    <h3>Method Detail</h3>
                    
                        <a name="calculateGCAdjustmentFactors(java.util.List)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>static&nbsp;graxxia.Matrix <strong>calculateGCAdjustmentFactors</strong>(java.util.List profiles)</h4>
                                <p> Given the list of GC bins with relative coverage (vs mean) for each bin, calculate a matrix
 that gives, for each sample and each bin, the multiplier that would equalise the coverage in the
 bin to the mean of other samples for that bin.
 <p>
 The intent of this is that multiplying the coverage by this factor will compensate for GC
 bias.
 <p>
 Each profile in the input list is expected to be of the form:
 <pre>
 {
     sample: <sample name>,
     gc: [
        { bin: [0,0.05], mean: 0.45 },
        { bin: [0,0.10], mean: 0.50 },
        ...
     ]
 }
 </pre>
     <DL><DT><B>Parameters:</B></DT><DD>profiles</DD></DL><DL><DT><B>Returns:</B></DT><DD>Matrix with the adjustment factor for each sample / bin in the rows/columns respectively</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="dropNonOverlapping()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>dropNonOverlapping</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="enableGCNormalisation(List<Double>, gngs.FASTA, Map<String, IntervalGCCache>)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>enableGCNormalisation</strong>(java.util.List&lt;java.lang.Double&gt; gcProfile, <a href='../../gngs/FASTA.html'>FASTA</a> fasta, java.util.Map&lt;java.lang.String, <a href='../../gngs/IntervalGCCache.html' title='IntervalGCCache'>IntervalGCCache</a>&gt; caches)</h4>
                                <p> Enables GC normalisation based on pre-computed weightings for the given GC bins
 (which are assumed to span 0-100% GC content). GC profile is computed
 per-fragment based on the given reference FASTA.
 <p>
 The caches will be utilised and populated with new IntervalGCCache entries when one does
 not already exist for a chromosome.
     <DL><DT><B>Parameters:</B></DT><DD><code>gcProfile</code> -  coverage depth weighting for bins which are assumed to span 0-100% GC content.</DD><DD><code>fasta</code> -      reference FASTA for computing GC content of fragments</DD><DD><code>caches</code> -     set of GC content caches, keyed on chromosome / contig</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="iteratorRemove(java.util.Iterator)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>iteratorRemove</strong>(java.util.Iterator i)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="nextRegion()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
boolean <strong>nextRegion</strong>()</h4>
                                <p> Transition to the next target region, taking into account that it might 
 also cause transition to next reference sequence (chromosome).
     <DL><DT><B>Returns:</B></DT><DD>true iff a new region was set, false if regions are exhausted</DD></DL></p>
                            </li>
                        </ul>
                    
                        <a name="onEnd()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>void <strong>onEnd</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="process(gngs.coverage.ReadRange)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
void <strong>process</strong>(<a href='../../gngs/coverage/ReadRange.html'>ReadRange</a> r)</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                        <a name="processBAM(gngs.SAM, gngs.Regions, gngs.RegulatingActor, int, java.lang.String, boolean, graxxia.Matrix)"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>@groovy.transform.CompileStatic
static&nbsp;void <strong>processBAM</strong>(<a href='../../gngs/SAM.html'>SAM</a> bam, <a href='../../gngs/Regions.html'>Regions</a> scanRegions, <a href='../../gngs/RegulatingActor.html'>RegulatingActor</a> downstream, int minMQ, java.lang.String sample = null, boolean countFragments = true, graxxia.Matrix gcAdjustments = null)</h4>
                                <p> Create a CoverageCalculatorActor and send all the reads from the given alignment that
 overlap the given regions to it.
     </p>
                            </li>
                        </ul>
                    
                        <a name="toString()"><!-- --></a>
                        <ul class="blockListLast">
                            <li class="blockList">
                                <h4>java.lang.String <strong>toString</strong>()</h4>
                                <p></p>
                            </li>
                        </ul>
                    
                </li>
            </ul>
            
        </li>
    </ul>
</div>

<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
    <!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="../../overview-summary.html">Overview</a></li>
        <li><a href="package-summary.html">Package</a></li>
        <li class="navBarCell1Rev">Class</li>
        <li><a href="../../deprecated-list.html">Deprecated</a></li>
        <li><a href="../../index-all.html">Index</a></li>
        <li><a href="../../help-doc.html">Help</a></li>
    </ul>
</div>

<div class="subNav">
    <div>
        <ul class="navList">
            <li><a href="../../index.html?gngs/coverage/CoverageCalculatorActor" target="_top">Frames</a></li>
            <li><a href="CoverageCalculatorActor.html" target="_top">No Frames</a></li>
        </ul>
    </div>
    <div>
        <ul class="subNavList">
            <li>Summary:&nbsp;</li>
            Nested&nbsp;&nbsp;&nbsp;Field&nbsp;&nbsp;&nbsp;<li><a href="#property_summary">Property</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#constructor_summary">Constructor</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#method_summary">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
        <ul class="subNavList">
            <li>&nbsp;|&nbsp;Detail:&nbsp;</li>
            Field&nbsp;&nbsp;&nbsp;<li><a href="#prop_detail">Property</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#constructor_detail">Constructor</a></li>&nbsp;&nbsp;&nbsp;<li><a href="#method_detail">Method</a></li>&nbsp;&nbsp;&nbsp;
        </ul>
    </div>
    <p>Groovy Documentation</p>
    <a name="skip-navbar_bottom">
        <!--   -->
    </a>
    </div>
</div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
