<!--

     Licensed to the Apache Software Foundation (ASF) under one
     or more contributor license agreements.  See the NOTICE file
     distributed with this work for additional information
     regarding copyright ownership.  The ASF licenses this file
     to you under the Apache License, Version 2.0 (the
     "License"); you may not use this file except in compliance
     with the License.  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

     Unless required by applicable law or agreed to in writing,
     software distributed under the License is distributed on an
     "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
     KIND, either express or implied.  See the License for the
     specific language governing permissions and limitations
     under the License.

-->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">

<html>
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <!-- Generated by groovydoc (2.4.10) on Mon May 07 11:56:19 AEST 2018 -->
        <title>Index (groovy-ngs-utils API)</title>
        <meta name="date" content="2018-05-07">
        <link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
    </head>
<body class="center">
<script type="text/javascript"><!--
if (location.href.indexOf('is-external=true') == -1) {
    parent.document.title="Index (groovy-ngs-utils API)";
}
//-->
</script>
<noscript>
    <div>JavaScript is disabled on your browser.</div>
</noscript>

<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
    <!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="overview-summary.html">Overview</a></li>
        <li>Package</li>
        <li>Class</li>
        <li><a href="deprecated-list.html">Deprecated</a></li>
        <li class="navBarCell1Rev">Index</li>
        <li><a href="help-doc.html">Help</a></li>
    </ul>
</div>
<div class="subNav">
    <ul class="navList">
        <li><a href="./index.html?index-all.html" target="_top">Frames</a></li>
        <li><a href="index-all.html" target="_top">No Frames</a></li>
    </ul>
    <a name="skip-navbar_top">
        <!--   -->
    </a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="contentContainer">

<a href="#_A_">A</a>

<a href="#_B_">B</a>

<a href="#_C_">C</a>

<a href="#_D_">D</a>

<a href="#_E_">E</a>

<a href="#_F_">F</a>

<a href="#_G_">G</a>

<a href="#_H_">H</a>

<a href="#_I_">I</a>

<a href="#_J_">J</a>

<a href="#_K_">K</a>

<a href="#_L_">L</a>

<a href="#_M_">M</a>

<a href="#_N_">N</a>

<a href="#_O_">O</a>

<a href="#_P_">P</a>

<a href="#_Q_">Q</a>

<a href="#_R_">R</a>

<a href="#_S_">S</a>

<a href="#_T_">T</a>

<a href="#_U_">U</a>

<a href="#_V_">V</a>

<a href="#_W_">W</a>

<a href="#_X_">X</a>

<a href="#_Y_">Y</a>

<a href="#_Z_">Z</a>

<a href="#___">_</a>



<a name="_A_"><!-- --></a>
<h2 class="title">A</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#A" title="Field in FASTA">A</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Abort.html" title="Class in gngs">Abort</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Convenience for aborting loops invoked via closures</div></dd>
<dt><span class="strong"><a href="gngs/Abort.html#Abort(java.lang.String, java.lang.Throwable)" title="Constructor in Abort">Abort(String, Throwable)</a></span> - Constructor in <a href="gngs/Abort.html">Abort</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairActor.html#act()" title="Method in OrderedPairActor">act()</a></span> - Method in <a href="gngs/OrderedPairActor.html">OrderedPairActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFilter.html#act()" title="Method in PairFilter">act()</a></span> - Method in <a href="gngs/pair/PairFilter.html">PairFilter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFormatter.html#act()" title="Method in PairFormatter">act()</a></span> - Method in <a href="gngs/pair/PairFormatter.html">PairFormatter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairLocator.html#act()" title="Method in PairLocator">act()</a></span> - Method in <a href="gngs/pair/PairLocator.html">PairLocator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairWriter.html#act()" title="Method in PairWriter">act()</a></span> - Method in <a href="gngs/pair/PairWriter.html">PairWriter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadMateLookupActor.html#act()" title="Method in ReadMateLookupActor">act()</a></span> - Method in <a href="gngs/ReadMateLookupActor.html">ReadMateLookupActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ReadStatsActor.html#act()" title="Method in ReadStatsActor">act()</a></span> - Method in <a href="gngs/tools/ReadStatsActor.html">ReadStatsActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BED.html#add(java.lang.String, int, int, java.lang.Object)" title="Method in BED">add(String, int, int, Object)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block">Add the specified range to this BED file.</div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#add(gngs.SAMRecordPair)" title="Method in MissingMateIndex">add(SAMRecordPair)</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block">Index this mate</div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#add(gngs.Pedigree)" title="Method in Pedigrees">add(Pedigree)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block">Add the pedigree to this Pedigrees object</div></dd>
<dt><span class="strong"><a href="gngs/PrefixTrie.html#add(java.lang.String, T, int)" title="Method in PrefixTrie">add(String, T, int)</a></span> - Method in <a href="gngs/PrefixTrie.html">PrefixTrie</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#add(groovy.lang.IntRange)" title="Method in RangeIndex">add(IntRange)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieNode.html#add(java.lang.String, T)" title="Method in TrieNode">add(String, T)</a></span> - Method in <a href="trie/TrieNode.html">TrieNode</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#add(java.lang.String, gngs.Pedigrees, gngs.Variant, gngs.Allele, java.lang.String, java.lang.String, def)" title="Method in VariantDB">add(String, Pedigrees, Variant, Allele, String, String, def)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Add the given variant to the database. </div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#add(gngs.Variant)" title="Method in VCF">add(Variant)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Add the given variant to this VCF</div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairActor.html#addAlignmentPair(gngs.SAMRecordPair)" title="Method in OrderedPairActor">addAlignmentPair(SAMRecordPair)</a></span> - Method in <a href="gngs/OrderedPairActor.html">OrderedPairActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairWriter.html#addAlignmentPair(gngs.SAMRecordPair)" title="Method in OrderedPairWriter">addAlignmentPair(SAMRecordPair)</a></span> - Method in <a href="gngs/OrderedPairWriter.html">OrderedPairWriter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#addCachedSampleInfo(List<String>, java.lang.String, java.lang.String, gngs.Pedigrees)" title="Method in VariantDB">addCachedSampleInfo(List&lt;String&gt;, String, String, Pedigrees)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Returns cached rows from the sample table for each sample. </div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#addInfoHeader(java.lang.String, java.lang.String, java.lang.Object)" title="Method in VCF">addInfoHeader(String, String, Object)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Add a header for describing an info value to be added to a 
 VCF.</div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#addPreprocessedRead(gngs.SAMRecordPair)" title="Method in OrderedPairReader">addPreprocessedRead(SAMRecordPair)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#addRegion(gngs.Region)" title="Method in Regions">addRegion(Region)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Adds the region to this Regions object in such a way that the same Region
 object is returned in iteration, preserving any expando properties set on
 the region.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#addSample(java.lang.String, gngs.Pedigrees, java.lang.String, java.lang.String)" title="Method in VariantDB">addSample(String, Pedigrees, String, String)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Add information about the given sample to the database</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#addSampleBatch(java.lang.Long, java.lang.String, java.lang.String)" title="Method in VariantDB">addSampleBatch(Long, String, String)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadMateLookupActor.html#addStats(long, int, int, gngs.Region)" title="Method in ReadMateLookupActor">addStats(long, int, int, Region)</a></span> - Method in <a href="gngs/ReadMateLookupActor.html">ReadMateLookupActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/AdjustZygosity.html#adjustDeletion(gngs.Variant)" title="Method in AdjustZygosity">adjustDeletion(Variant)</a></span> - Method in <a href="gngs/tools/AdjustZygosity.html">AdjustZygosity</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/AdjustZygosity.html#adjustVariant(gngs.Variant)" title="Method in AdjustZygosity">adjustVariant(Variant)</a></span> - Method in <a href="gngs/tools/AdjustZygosity.html">AdjustZygosity</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/AdjustZygosity.html" title="Class in gngs.tools">AdjustZygosity</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block">Compares calls in a VCF file to the reads in a BAM file and adjusts
 variants that are called heterozygous to homozygous when the fraction
 of reads for the reference is implausibly low for a homozygous variant.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/Align.html" title="Class in Align">Align</a></span> - Class in <a href="./Align/package-summary.html">Align</a>
</dt><dd><div class="block">Utility for helping to perform alignments with BioJava</div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#align(java.lang.String, Closure)" title="Method in PDF">align(String, Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block">Set alignment within table cells</div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#alignments" title="Field in Pileup">alignments</a></span> - Field in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#alignments" title="Field in PileupIterator">alignments</a></span> - Field in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Allele.html" title="Class in gngs">Allele</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A specific allele in the context of a Variant in a VCF file</div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#allocateScannableRegion(gngs.SAMRecordPair, int, int)" title="Method in MissingMateIndex">allocateScannableRegion(SAMRecordPair, int, int)</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TargetedCNVAnnotator.html#annotate(gngs.IRegion, java.lang.String)" title="Method in TargetedCNVAnnotator">annotate(IRegion, String)</a></span> - Method in <a href="DefaultPackage/TargetedCNVAnnotator.html">TargetedCNVAnnotator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Table.html#annotateGenes(List<Map>)" title="Method in Table">annotateGenes(List&lt;Map&gt;)</a></span> - Method in <a href="gngs/tools/Table.html">Table</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TargetedCNVAnnotator.html#annotateSize(gngs.IRegion)" title="Method in TargetedCNVAnnotator">annotateSize(IRegion)</a></span> - Method in <a href="DefaultPackage/TargetedCNVAnnotator.html">TargetedCNVAnnotator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#appendTo(java.lang.StringBuilder, htsjdk.samtools.SAMRecord, boolean)" title="Method in CompactReadPair">appendTo(StringBuilder, SAMRecord, boolean)</a></span> - Method in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#appendTo(java.lang.StringBuilder, boolean)" title="Method in SAMRecordPair">appendTo(StringBuilder, boolean)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#applyCellProperties(com.lowagie.text.pdf.PdfPCell)" title="Method in PDF">applyCellProperties(PdfPCell)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#applyDEL(java.lang.String, int)" title="Method in Variant">applyDEL(String, int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Apply this deletion to the specified sequence at the specified position</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#applyINS(java.lang.String, int)" title="Method in Variant">applyINS(String, int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Apply this deletion to the specified sequence at the specified position</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#applySNP(java.lang.String, int)" title="Method in Variant">applySNP(String, int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#applyTo(java.lang.String, int)" title="Method in Variant">applyTo(String, int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Apply this variant to mutate the given sequence at the specified position</div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#array(int)" title="Method in Utils">array(int)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Assembler.html#assemble(List<String>)" title="Method in Assembler">assemble(List&lt;String&gt;)</a></span> - Method in <a href="DefaultPackage/Assembler.html">Assembler</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/AssembledContig.html" title="Class in AssembledContig">AssembledContig</a></span> - Class in <a href="./AssembledContig/package-summary.html">AssembledContig</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Assembler.html" title="Class in Assembler">Assembler</a></span> - Class in <a href="./Assembler/package-summary.html">Assembler</a>
</dt><dd><div class="block">A simplistic local assembler which can create contigs out of
 overlapping reads.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/Assembler.html#Assembler()" title="Constructor in Assembler">Assembler()</a></span> - Constructor in <a href="DefaultPackage/Assembler.html">Assembler</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageBlock.html#asType(java.lang.Class)" title="Method in CoverageBlock">asType(Class)</a></span> - Method in <a href="gngs/CoverageBlock.html">CoverageBlock</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageGaps.html#asType(java.lang.Class)" title="Method in CoverageGaps">asType(Class)</a></span> - Method in <a href="gngs/CoverageGaps.html">CoverageGaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#asType(java.lang.Class)" title="Method in SAM">asType(Class)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordCategory.html#asType(htsjdk.samtools.BAMRecord, java.lang.Class)" title="Method in SAMRecordCategory">asType(BAMRecord, Class)</a></span> - Method in <a href="gngs/SAMRecordCategory.html">SAMRecordCategory</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#asType(java.lang.Class)" title="Method in VCF">asType(Class)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/AsyncMateResolveWorker.html" title="Class in gngs">AsyncMateResolveWorker</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A background task that can read from a queue of unpaired reads and resolve their
 pairs in the background, as well as resolve other reads that are in the same
 region as the pair that is resolved.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/AutoRecTrio.html" title="Class in AutoRecTrio">AutoRecTrio</a></span> - Class in <a href="./AutoRecTrio/package-summary.html">AutoRecTrio</a>
</dt><dd><div class="block"></div></dd>
</dl>

    
<a name="_B_"><!-- --></a>
<h2 class="title">B</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#B" title="Field in FASTA">B</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#background(java.lang.String, Closure)" title="Method in PDF">background(String, Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block">Set background color of table cells</div></dd>
<dt><span class="strong"><a href="gngs/tools/BAM2Ped.html" title="Class in gngs.tools">BAM2Ped</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/BAM2Ped.html#BAM2Ped(gngs.VCF, List<SAM>)" title="Constructor in BAM2Ped">BAM2Ped(VCF, List&lt;SAM&gt;)</a></span> - Constructor in <a href="gngs/tools/BAM2Ped.html">BAM2Ped</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Cli.html#banner(java.lang.String)" title="Method in Cli">banner(String)</a></span> - Method in <a href="gngs/Cli.html">Cli</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Drawing.html#bar(gngs.IRegion, def, java.lang.String, java.lang.String)" title="Method in Drawing">bar(IRegion, def, String, String)</a></span> - Method in <a href="DefaultPackage/Drawing.html">Drawing</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html#base" title="Field in PileupState">base</a></span> - Field in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTA.html#baseBytesAt(java.lang.String, long, long)" title="Method in FASTA">baseBytesAt(String, long, long)</a></span> - Method in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block">Returns the sequence of bases over the given range
 <b>NOTE:</b>The range is <i>inclusive</i>.</div></dd>
<dt><span class="strong"><a href="gngs/BaseCounts.html" title="Class in gngs">BaseCounts</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Stores a set of base counts for a base read at a given position,
 and supports scoring metrics.</div></dd>
<dt><span class="strong"><a href="gngs/BaseCounts.html#BaseCounts(List<String>, int)" title="Constructor in BaseCounts">BaseCounts(List&lt;String&gt;, int)</a></span> - Constructor in <a href="gngs/BaseCounts.html">BaseCounts</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#basesAndQuals" title="Field in CompactReadPair">basesAndQuals</a></span> - Field in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTA.html#basesAt(java.lang.String, long, long)" title="Method in FASTA">basesAt(String, long, long)</a></span> - Method in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block">Returns the sequence of bases over the given range
 <b>NOTE:</b>The range is <i>inclusive</i>.</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#basesAt(java.lang.String, int)" title="Method in SAM">basesAt(String, int)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Return a Map with a key for each base observed at the given position,
 with the value being the number of times that base was observed. </div></dd>
<dt><span class="strong"><a href="gngs/BED.html" title="Class in gngs">BED</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Support for parsing, filtering, and manipuulating BED files.</div></dd>
<dt><span class="strong"><a href="gngs/BED.html#BED(java.util.Map, Iterable<Region>)" title="Constructor in BED">BED(Map, Iterable&lt;Region&gt;)</a></span> - Constructor in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block">Create a BED from a list of regions</div></dd>
<dt><span class="strong"><a href="DefaultPackage/BEDToolsCoverage.html" title="Class in BEDToolsCoverage">BEDToolsCoverage</a></span> - Class in <a href="./BEDToolsCoverage/package-summary.html">BEDToolsCoverage</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/BEDToolsCoverage.html#BEDToolsCoverage(java.lang.String, gngs.Regions)" title="Constructor in BEDToolsCoverage">BEDToolsCoverage(String, Regions)</a></span> - Constructor in <a href="DefaultPackage/BEDToolsCoverage.html">BEDToolsCoverage</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BaseCounts.html#bestScore()" title="Method in BaseCounts">bestScore()</a></span> - Method in <a href="gngs/BaseCounts.html">BaseCounts</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#bg(java.lang.String, Closure)" title="Method in PDF">bg(String, Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#bkr()" title="Method in Regions">bkr()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#bold(Closure)" title="Method in PDF">bold(Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#br(int)" title="Method in PDF">br(int)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTA.html#bracket(java.lang.String, int, int)" title="Method in FASTA">bracket(String, int, int)</a></span> - Method in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block">Display given sequence with given bases bracketed</div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#BUFFER_SIZE" title="Field in VCFIndex">BUFFER_SIZE</a></span> - Field in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block">A 256 kb memory buffer that will try to minimize operations
 on the indexed file</div></dd>
<dt><span class="strong"><a href="gngs/Consensus.html#build()" title="Method in Consensus">build()</a></span> - Method in <a href="gngs/Consensus.html">Consensus</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#buildGtFieldValue(java.lang.String, List<String>, List<String>)" title="Method in VCF">buildGtFieldValue(String, List&lt;String&gt;, List&lt;String&gt;)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_C_"><!-- --></a>
<h2 class="title">C</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#C" title="Field in FASTA">C</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ConcordanceScore.html#calculate()" title="Method in ConcordanceScore">calculate()</a></span> - Method in <a href="gngs/tools/ConcordanceScore.html">ConcordanceScore</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageGaps.html#calculate()" title="Method in CoverageGaps">calculate()</a></span> - Method in <a href="gngs/CoverageGaps.html">CoverageGaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageGaps.html#calculateFromStream(java.io.InputStream)" title="Method in CoverageGaps">calculateFromStream(InputStream)</a></span> - Method in <a href="gngs/CoverageGaps.html">CoverageGaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#calculateGaps(java.lang.String)" title="Method in Gaps">calculateGaps(String)</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#calculateROC()" title="Method in VCFtoHTML">calculateROC()</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DepthOfCoverage.html#calculateStatistics(gngs.SAM, gngs.Region)" title="Method in DepthOfCoverage">calculateStatistics(SAM, Region)</a></span> - Method in <a href="gngs/tools/DepthOfCoverage.html">DepthOfCoverage</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#cell(java.lang.Object)" title="Method in PDF">cell(Object)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#cells(java.lang.Object)" title="Method in PDF">cells(Object)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Cli.html#check(java.lang.Object, java.util.List)" title="Method in Cli">check(Object, List)</a></span> - Method in <a href="gngs/Cli.html">Cli</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#checkAsyncResolve(gngs.SAMRecordPair)" title="Method in OrderedPairReader">checkAsyncResolve(SAMRecordPair)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#checkRanges(int)" title="Method in RangeIndex">checkRanges(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="db/Schema.html#checkSchema(groovy.sql.Sql)" title="Method in Schema">checkSchema(Sql)</a></span> - Method in <a href="db/Schema.html">Schema</a>
</dt><dd> <div class="block">Iterate over the schema versions and check if they exist</div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#chr" title="Field in Pileup">chr</a></span> - Field in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#chr" title="Field in PileupIterator">chr</a></span> - Field in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/LXPos.html#chrIndex(java.lang.String)" title="Method in LXPos">chrIndex(String)</a></span> - Method in <a href="gngs/LXPos.html">LXPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/LXPos.html#chrToInt(java.lang.String)" title="Method in LXPos">chrToInt(String)</a></span> - Method in <a href="gngs/LXPos.html">LXPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/XPos.html#chrToInt(java.lang.String)" title="Method in XPos">chrToInt(String)</a></span> - Method in <a href="gngs/XPos.html">XPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html#cigar" title="Field in PileupState">cigar</a></span> - Field in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html#cigarElement" title="Field in PileupState">cigarElement</a></span> - Field in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html#cigarIterator" title="Field in PileupState">cigarIterator</a></span> - Field in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html#cigarPos" title="Field in PileupState">cigarPos</a></span> - Field in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block">Offset within the current cigar element</div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#clear()" title="Method in MissingMateIndex">clear()</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Cli.html" title="Class in gngs">Cli</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">This class is a simple utility that configures a CliBuilder how I like it and adds a few
 simple options that I like.</div></dd>
<dt><span class="strong"><a href="gngs/Cli.html#Cli()" title="Constructor in Cli">Cli()</a></span> - Constructor in <a href="gngs/Cli.html">Cli</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ToolBase.html#cli(java.lang.String, java.lang.String, groovy.lang.Closure)" title="Method in ToolBase">cli(String, String, Closure)</a></span> - Method in <a href="gngs/ToolBase.html">ToolBase</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/ClinVar.html" title="Class in ClinVar">ClinVar</a></span> - Class in <a href="./ClinVar/package-summary.html">ClinVar</a>
</dt><dd><div class="block">A thin wrapper around a RangedData object to add some utility functions
 for accessing clinvar data.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/ClinVar.html#ClinVar(gngs.RangedData)" title="Constructor in ClinVar">ClinVar(RangedData)</a></span> - Constructor in <a href="DefaultPackage/ClinVar.html">ClinVar</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#close()" title="Method in OrderedPairReader">close()</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairWriter.html#close()" title="Method in OrderedPairWriter">close()</a></span> - Method in <a href="gngs/OrderedPairWriter.html">OrderedPairWriter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#close()" title="Method in Pileup">close()</a></span> - Method in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#close()" title="Method in PileupIterator">close()</a></span> - Method in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadMateLookupActor.html#close()" title="Method in ReadMateLookupActor">close()</a></span> - Method in <a href="gngs/ReadMateLookupActor.html">ReadMateLookupActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#close()" title="Method in SAM">close()</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Close the underlying SAMFileReader</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#close()" title="Method in VariantDB">close()</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#closeQuietly(def)" title="Method in Utils">closeQuietly(def)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CNVDatabase.html" title="Class in gngs">CNVDatabase</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Abstract interface that can be implemented by different
 CNV sources.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#color(java.lang.String, Closure)" title="Method in PDF">color(String, Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html" title="Class in gngs">CompactReadPair</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#CompactReadPair(htsjdk.samtools.SAMRecord)" title="Constructor in CompactReadPair">CompactReadPair(SAMRecord)</a></span> - Constructor in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMatePriority.html#compare(gngs.SAMRecordPair, gngs.SAMRecordPair)" title="Method in MissingMatePriority">compare(SAMRecordPair, SAMRecordPair)</a></span> - Method in <a href="gngs/MissingMatePriority.html">MissingMatePriority</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/NumericRegionComparator.html#compare(gngs.Region, gngs.Region)" title="Method in NumericRegionComparator">compare(Region, Region)</a></span> - Method in <a href="gngs/NumericRegionComparator.html">NumericRegionComparator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RegionComparator.html#compare(gngs.Region, gngs.Region)" title="Method in RegionComparator">compare(Region, Region)</a></span> - Method in <a href="gngs/RegionComparator.html">RegionComparator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#compareTo(java.lang.Object)" title="Method in SAMRecordPair">compareTo(Object)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block">Compare this read pair to another read pair by genomic position.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFSimilarity.html#compute()" title="Method in VCFSimilarity">compute()</a></span> - Method in <a href="DefaultPackage/VCFSimilarity.html">VCFSimilarity</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#computeAllGaps()" title="Method in Gaps">computeAllGaps()</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PairReaderStats.html#computeBlockId(long, int)" title="Method in PairReaderStats">computeBlockId(long, int)</a></span> - Method in <a href="gngs/PairReaderStats.html">PairReaderStats</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ConcordanceScore.html#computeChildHetConcordance(java.lang.String)" title="Method in ConcordanceScore">computeChildHetConcordance(String)</a></span> - Method in <a href="gngs/tools/ConcordanceScore.html">ConcordanceScore</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ConcordanceScore.html#computeChildHomConcordance(java.lang.String)" title="Method in ConcordanceScore">computeChildHomConcordance(String)</a></span> - Method in <a href="gngs/tools/ConcordanceScore.html">ConcordanceScore</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#computeDiffGaps()" title="Method in Gaps">computeDiffGaps()</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#computeGapUnion(List<Regions>)" title="Method in Gaps">computeGapUnion(List&lt;Regions&gt;)</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/LXPos.html#computeId(int, int)" title="Method in LXPos">computeId(int, int)</a></span> - Method in <a href="gngs/LXPos.html">LXPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/XPos.html#computeId(int, int)" title="Method in XPos">computeId(int, int)</a></span> - Method in <a href="gngs/XPos.html">XPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TargetedCNVAnnotator.html#computeMaxRange(gngs.IRegion)" title="Method in TargetedCNVAnnotator">computeMaxRange(IRegion)</a></span> - Method in <a href="DefaultPackage/TargetedCNVAnnotator.html">TargetedCNVAnnotator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ConcordanceScore.html#computeParentHomConcordance(java.lang.String)" title="Method in ConcordanceScore">computeParentHomConcordance(String)</a></span> - Method in <a href="gngs/tools/ConcordanceScore.html">ConcordanceScore</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/LXPos.html#computePos(java.lang.String, int)" title="Method in LXPos">computePos(String, int)</a></span> - Method in <a href="gngs/LXPos.html">LXPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/XPos.html#computePos(java.lang.String, int)" title="Method in XPos">computePos(String, int)</a></span> - Method in <a href="gngs/XPos.html">XPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ReadStatsActor.html#computeStats()" title="Method in ReadStatsActor">computeStats()</a></span> - Method in <a href="gngs/tools/ReadStatsActor.html">ReadStatsActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ConcordanceScore.html" title="Class in gngs.tools">ConcordanceScore</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ConcordanceScore.html#ConcordanceScore(java.lang.String, java.lang.String, java.lang.String, Map<String, Map<String, Integer>>)" title="Constructor in ConcordanceScore">ConcordanceScore(String, String, String, Map&lt;String, Map&lt;String, Integer&gt;&gt;)</a></span> - Constructor in <a href="gngs/tools/ConcordanceScore.html">ConcordanceScore</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#configureSimpleLogging(java.util.logging.Level)" title="Method in Utils">configureSimpleLogging(Level)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block">Set up Java logging with a nice, simple, reasonable format</div></dd>
<dt><span class="strong"><a href="gngs/Consensus.html" title="Class in gngs">Consensus</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Support to determine a consensus sequence from a set of sequences</div></dd>
<dt><span class="strong"><a href="gngs/Consensus.html#Consensus(List<String>, int)" title="Constructor in Consensus">Consensus(List&lt;String&gt;, int)</a></span> - Constructor in <a href="gngs/Consensus.html">Consensus</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Constraint.html" title="Interface in gngs">Constraint</a></span> - Interface in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Represents a constraint in a genetic model</div></dd>
<dt><span class="strong"><a href="gngs/FASTQ.html#consumeRead(java.io.Reader)" title="Method in FASTQ">consumeRead(Reader)</a></span> - Method in <a href="gngs/FASTQ.html">FASTQ</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#contains(java.lang.String, int)" title="Method in Regions">contains(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Return true iff the given chromosome and position fall into
 a range covered by this BED file</div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#contains(gngs.Variant)" title="Method in VCFIndex">contains(Variant)</a></span> - Method in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block">Returns true if this VCF contains the specified variant</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#convertType(java.lang.String, java.lang.String)" title="Method in Variant">convertType(String, String)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Convert reference and alternate allele strings into a 
 mutation type, being one of "SNP","INS","DEL","GAIN"
 or "LOSS", with the latter two representing CNVs.</div></dd>
<dt><span class="strong"><a href="gngs/Region.html#copy()" title="Method in Region">copy()</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#copySubject(java.lang.String, java.lang.String)" title="Method in Pedigree">copySubject(String, String)</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieQuery.html#cost(trie.TrieMismatchCosts)" title="Method in TrieQuery">cost(TrieMismatchCosts)</a></span> - Method in <a href="trie/TrieQuery.html">TrieQuery</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ProgressCounter.html#count(groovy.lang.Closure)" title="Method in ProgressCounter">count(Closure)</a></span> - Method in <a href="gngs/ProgressCounter.html">ProgressCounter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#countObservations(gngs.Variant, gngs.Allele, java.lang.String)" title="Method in VariantDB">countObservations(Variant, Allele, String)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Return a counts of the number of observations of the given variant.
 </div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#countOf(byte)" title="Method in Pileup">countOf(byte)</a></span> - Method in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block">Return the count of reads supporting the specified base at the current
 position.</div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.Pileup.html#countOf(Closure<Object>)" title="Method in PileupIterator.Pileup">countOf(Closure&lt;Object&gt;)</a></span> - Method in <a href="gngs/PileupIterator.Pileup.html">PileupIterator.Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#countOf(def, def, java.lang.String)" title="Method in SAM">countOf(def, def, String)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#countOnTarget(gngs.Regions)" title="Method in SAM">countOnTarget(Regions)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Return a set of read counts indicating the counts of reads in that overlap the target region.</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#countPairs(java.lang.String, int, int)" title="Method in SAM">countPairs(String, int, int)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Count the number of read pairs in the given region</div></dd>
<dt><span class="strong"><a href="DefaultPackage/BEDToolsCoverage.html#coverage(gngs.Region)" title="Method in BEDToolsCoverage">coverage(Region)</a></span> - Method in <a href="DefaultPackage/BEDToolsCoverage.html">BEDToolsCoverage</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#coverage()" title="Method in RangeIndex">coverage()</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Return a list of ranges representing "coverage" blocks within this
 index. </div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#coverage()" title="Method in Regions">coverage()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Return a new regions object that has each distinct region of this object
 with a "coverage" value assigned.</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#coverage(htsjdk.samtools.SAMFileReader, java.lang.String, int, int, groovy.lang.Closure, int)" title="Method in SAM">coverage(SAMFileReader, String, int, int, Closure, int)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Return the number of mapped reads overlapping the given position</div></dd>
<dt><span class="strong"><a href="gngs/CoverageBlock.html" title="Class in gngs">CoverageBlock</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A run of contiguous bases within a band of coverage depth
 (eg: a run of low coverage bases).</div></dd>
<dt><span class="strong"><a href="gngs/CoverageGaps.html" title="Class in gngs">CoverageGaps</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Discovers blocks of low coverage in a file output in BEDTools Coverage format.</div></dd>
<dt><span class="strong"><a href="gngs/CoverageGaps.html#CoverageGaps(java.lang.String)" title="Constructor in CoverageGaps">CoverageGaps(String)</a></span> - Constructor in <a href="gngs/CoverageGaps.html">CoverageGaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#coverageStatistics(gngs.Regions)" title="Method in SAM">coverageStatistics(Regions)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageStats.html" title="Class in gngs">CoverageStats</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">An efficient method to calculate percentiles of coverage values
 that doesn't require holding them all in memory or sorting them.</div></dd>
<dt><span class="strong"><a href="gngs/CoverageStats.html#CoverageStats(int)" title="Constructor in CoverageStats">CoverageStats(int)</a></span> - Constructor in <a href="gngs/CoverageStats.html">CoverageStats</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/CpipeRunSet.html" title="Class in CpipeRunSet">CpipeRunSet</a></span> - Class in <a href="./CpipeRunSet/package-summary.html">CpipeRunSet</a>
</dt><dd><div class="block">Simple utilities that understand the structure of Cpipe runs</div></dd>
<dt><span class="strong"><a href="DefaultPackage/CpipeRunSet.html#CpipeRunSet(java.lang.String, List<String>)" title="Constructor in CpipeRunSet">CpipeRunSet(String, List&lt;String&gt;)</a></span> - Constructor in <a href="DefaultPackage/CpipeRunSet.html">CpipeRunSet</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TestTrio.html#createFamilyId(java.lang.String, java.lang.String, java.lang.String)" title="Method in TestTrio">createFamilyId(String, String, String)</a></span> - Method in <a href="DefaultPackage/TestTrio.html">TestTrio</a>
</dt><dd> <div class="block">Attempt to create a human interpretable family id for the family based on
 the ids of the mother / father / child.</div></dd>
<dt><span class="strong"><a href="gngs/Subject.html#createFather(java.lang.String)" title="Method in Subject">createFather(String)</a></span> - Method in <a href="gngs/Subject.html">Subject</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairScanner.html#createLocator()" title="Method in PairScanner">createLocator()</a></span> - Method in <a href="gngs/pair/PairScanner.html">PairScanner</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Subject.html#createMother(java.lang.String)" title="Method in Subject">createMother(String)</a></span> - Method in <a href="gngs/Subject.html">Subject</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/CSV.html" title="Class in CSV">CSV</a></span> - Class in <a href="./CSV/package-summary.html">CSV</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/CSV.html#CSV(Map, Reader)" title="Constructor in CSV">CSV(Map, Reader)</a></span> - Constructor in <a href="DefaultPackage/CSV.html">CSV</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/RestrictionEnzyme.html#cutsSequenceBefore(java.lang.String, int)" title="Method in RestrictionEnzyme">cutsSequenceBefore(String, int)</a></span> - Method in <a href="DefaultPackage/RestrictionEnzyme.html">RestrictionEnzyme</a>
</dt><dd> <div class="block">Returns true if this restriction enzyme would cut the given sequence
 in between the given position and the previous position.</div></dd>
</dl>

    
<a name="_D_"><!-- --></a>
<h2 class="title">D</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#D" title="Field in FASTA">D</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#debugdump()" title="Method in RangeIndex">debugdump()</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/DecipherCNVs.html" title="Class in gngs">DecipherCNVs</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A <a href='../gngs/CNVDatabase.html' title='CNVDatabase'>CNVDatabase</a> implementation for CNVs from Decipher 
 Developmental Delay project.</div></dd>
<dt><span class="strong"><a href="gngs/DecipherCNVs.html#DecipherCNVs(gngs.RangedData)" title="Constructor in DecipherCNVs">DecipherCNVs(RangedData)</a></span> - Constructor in <a href="gngs/DecipherCNVs.html">DecipherCNVs</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Sex.html#decode(def)" title="Method in Sex">decode(def)</a></span> - Method in <a href="gngs/Sex.html">Sex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SamplesToPed.html#decodeRelationship(java.lang.String, gngs.Subject)" title="Method in SamplesToPed">decodeRelationship(String, Subject)</a></span> - Method in <a href="DefaultPackage/SamplesToPed.html">SamplesToPed</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieNode.html#DEFAULT_COSTS" title="Field in TrieNode">DEFAULT_COSTS</a></span> - Field in <a href="trie/TrieNode.html">TrieNode</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SamplesToPed.html#defaultRelationship(java.lang.String)" title="Method in SamplesToPed">defaultRelationship(String)</a></span> - Method in <a href="DefaultPackage/SamplesToPed.html">SamplesToPed</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieQuery.html#deletion(java.lang.String)" title="Method in TrieQuery">deletion(String)</a></span> - Method in <a href="trie/TrieQuery.html">TrieQuery</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/DenovoRate.html" title="Class in DenovoRate">DenovoRate</a></span> - Class in <a href="./DenovoRate/package-summary.html">DenovoRate</a>
</dt><dd><div class="block">Determines the rate of denovos for a trio</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#denovoRate(gngs.VCF, gngs.VCF)" title="Method in VCF">denovoRate(VCF, VCF)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DepthOfCoverage.html" title="Class in gngs.tools">DepthOfCoverage</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/DGV.html" title="Class in gngs">DGV</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A <a href='../gngs/CNVDatabase.html' title='CNVDatabase'>CNVDatabase</a> implementation for CNVs from the Database of
 Genomic Variants (sourced via UCSC).</div></dd>
<dt><span class="strong"><a href="gngs/DGV.html#DGV(gngs.RangedData)" title="Constructor in DGV">DGV(RangedData)</a></span> - Constructor in <a href="gngs/DGV.html">DGV</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#displaySequenceDifference(java.lang.String, int)" title="Method in Variant">displaySequenceDifference(String, int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">A debug function, displays the sequence difference and returns the 
 new sequence</div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#document(java.lang.String, Closure)" title="Method in PDF">document(String, Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#download(java.lang.String)" title="Method in RefGenes">download(String)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DownsampleFastq.html" title="Class in gngs.tools">DownsampleFastq</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Drawing.html" title="Class in Drawing">Drawing</a></span> - Class in <a href="./Drawing/package-summary.html">Drawing</a>
</dt><dd><div class="block">Drawing is a very simple graphics class that enables simple line and text drawings on a 
 virtualised canvas that automatically maps mathematical coordinates to pixel based
 coordinates.</div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#dump()" title="Method in RangeIndex">dump()</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Time.html#dump()" title="Method in Time">dump()</a></span> - Method in <a href="DefaultPackage/Time.html">Time</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#dumpProfile(java.io.Writer)" title="Method in OrderedPairReader">dumpProfile(Writer)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_E_"><!-- --></a>
<h2 class="title">E</h2>
<dl>

<dt><span class="strong"><a href="gngs/VCF.html#each(groovy.lang.Closure)" title="Method in VCF">each(Closure)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BED.html#eachInputRange(groovy.lang.Closure)" title="Method in BED">eachInputRange(Closure)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BED.html#eachLoadedRange(java.util.Map, groovy.lang.Closure)" title="Method in BED">eachLoadedRange(Map, Closure)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BED.html#eachOverlap(java.lang.String, int, groovy.lang.Closure)" title="Method in BED">eachOverlap(String, int, Closure)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block">Call closure c for each range that overlaps the given position</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#eachOverlap(java.lang.String, int, groovy.lang.Closure)" title="Method in Regions">eachOverlap(String, int, Closure)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Call closure c for each range that overlaps the given position</div></dd>
<dt><span class="strong"><a href="gngs/FASTQ.html#eachPair(java.lang.String, java.lang.String, groovy.lang.Closure)" title="Method in FASTQ">eachPair(String, String, Closure)</a></span> - Method in <a href="gngs/FASTQ.html">FASTQ</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#eachPair(gngs.Regions, groovy.lang.Closure)" title="Method in OrderedPairReader">eachPair(Regions, Closure)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#eachPair(java.util.Map, java.lang.String, int, int, groovy.lang.Closure)" title="Method in SAM">eachPair(Map, String, int, int, Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#eachPairImpl(htsjdk.samtools.SAMRecordIterator, groovy.lang.Closure)" title="Method in OrderedPairReader">eachPairImpl(SAMRecordIterator, Closure)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BED.html#eachPosition(java.lang.String, groovy.lang.Closure)" title="Method in BED">eachPosition(String, Closure)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block">Iterate every single base in the regions of the specified bed file</div></dd>
<dt><span class="strong"><a href="gngs/BED.html#eachRange(java.util.Map, groovy.lang.Closure)" title="Method in BED">eachRange(Map, Closure)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block">Iterate over each line in the BED file and invoke the specified
 closure with bed file information for the line.</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#eachRange(java.util.Map, groovy.lang.Closure)" title="Method in Regions">eachRange(Map, Closure)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Iterate over each region and invoke the specified
 closure with bed file information for the line.</div></dd>
<dt><span class="strong"><a href="gngs/FASTQ.html#eachRead(java.lang.String, groovy.lang.Closure)" title="Method in FASTQ">eachRead(String, Closure)</a></span> - Method in <a href="gngs/FASTQ.html">FASTQ</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#eachRead(groovy.lang.Closure)" title="Method in SAM">eachRead(Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Read a BAM or SAM file from standard input and call the given closure for
 each read contained therein.</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#eachRecord(int, groovy.lang.Closure)" title="Method in SAM">eachRecord(int, Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Call the given closure for each read in this alignment, using the
 given number of threads eg:</div></dd>
<dt><span class="strong"><a href="gngs/FASTA.html#eachSequence(gngs.Regions, groovy.lang.Closure)" title="Method in FASTA">eachSequence(Regions, Closure)</a></span> - Method in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block">Iterates over the ranges in the BED file and extracts the bases for the ranges,
 passing the bases of the sequence
 as a byte array to the given closure for each range.</div></dd>
<dt><span class="strong"><a href="gngs/Sex.html#encode()" title="Method in Sex">encode()</a></span> - Method in <a href="gngs/Sex.html">Sex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PairReaderStats.html#end()" title="Method in PairReaderStats">end()</a></span> - Method in <a href="gngs/PairReaderStats.html">PairReaderStats</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#end" title="Field in Pileup">end</a></span> - Field in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#end" title="Field in PileupIterator">end</a></span> - Field in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ProgressCounter.html#end()" title="Method in ProgressCounter">end()</a></span> - Method in <a href="gngs/ProgressCounter.html">ProgressCounter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ReadStatsActor.html#end()" title="Method in ReadStatsActor">end()</a></span> - Method in <a href="gngs/tools/ReadStatsActor.html">ReadStatsActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#endingAt(int)" title="Method in RangeIndex">endingAt(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#endingAt(java.lang.String, int)" title="Method in Regions">endingAt(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Return a list of ranges that end exactly at the specified position.</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#enhance()" title="Method in Regions">enhance()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#equalsAnnovar(java.lang.String, int, java.lang.String)" title="Method in Variant">equalsAnnovar(String, int, String)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return true if the given Annovar formattd position and observation
 match those of this variant</div></dd>
<dt><span class="strong"><a href="gngs/ToolBase.html#error(java.lang.String)" title="Method in ToolBase">error(String)</a></span> - Method in <a href="gngs/ToolBase.html">ToolBase</a>
</dt><dd> <div class="block">Print a formatted error message and exit</div></dd>
<dt><span class="strong"><a href="gngs/tools/MeanCoverageEstimator.html#estimate()" title="Method in MeanCoverageEstimator">estimate()</a></span> - Method in <a href="gngs/tools/MeanCoverageEstimator.html">MeanCoverageEstimator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/ExonCov.html" title="Class in ExonCov">ExonCov</a></span> - Class in <a href="./ExonCov/package-summary.html">ExonCov</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/ExonCoverageStats.html" title="Script in ExonCoverageStats">ExonCoverageStats</a></span> - Script in <a href="./ExonCoverageStats/package-summary.html">ExonCoverageStats</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ExtractFASTQ.html" title="Class in gngs.tools">ExtractFASTQ</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block">Extract reads as pairs from a sorted BAM and writes them out to standard 
 output in interleaved format.</div></dd>
<dt><span class="strong"><a href="gngs/tools/ExtractFASTQ.html#ExtractFASTQ(gngs.SAM, gngs.Regions)" title="Constructor in ExtractFASTQ">ExtractFASTQ(SAM, Regions)</a></span> - Constructor in <a href="gngs/tools/ExtractFASTQ.html">ExtractFASTQ</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#extractPairsToScan(gngs.RegionScan)" title="Method in MissingMateIndex">extractPairsToScan(RegionScan)</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block">Remove all read pairs from the index overlapping the given region
 and add them to the RegionScan object.</div></dd>
</dl>

    
<a name="_F_"><!-- --></a>
<h2 class="title">F</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html" title="Class in gngs">FASTA</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Simple utilities for manipulating FASTA files.</div></dd>
<dt><span class="strong"><a href="gngs/FASTA.html#FASTA(java.io.File)" title="Constructor in FASTA">FASTA(File)</a></span> - Constructor in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block">Create a FASTA object for the given FASTA file, which must be indexed.</div></dd>
<dt><span class="strong"><a href="gngs/FASTAIndex.html" title="Class in gngs">FASTAIndex</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A very simplistic FASTA index that allows lookup of sequence names by their
 sequence content using a fixed length prefix seed. </div></dd>
<dt><span class="strong"><a href="gngs/FASTAIndex.html#FASTAIndex(gngs.FASTA, groovy.lang.IntRange, int, int, gngs.BED)" title="Constructor in FASTAIndex">FASTAIndex(FASTA, IntRange, int, int, BED)</a></span> - Constructor in <a href="gngs/FASTAIndex.html">FASTAIndex</a>
</dt><dd> <div class="block">Create an index from the given fasta, where each fasta sequence corresponds
 to a single amplicon</div></dd>
<dt><span class="strong"><a href="gngs/FASTQ.html" title="Class in gngs">FASTQ</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Simple support for reading fastq files.</div></dd>
<dt><span class="strong"><a href="gngs/FASTQRead.html" title="Class in gngs">FASTQRead</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Models a single FASTQ read, including the header (with name), bases and 
 quality information.</div></dd>
<dt><span class="strong"><a href="gngs/FASTQRead.html#FASTQRead(java.lang.String, java.lang.String, java.lang.String)" title="Constructor in FASTQRead">FASTQRead(String, String, String)</a></span> - Constructor in <a href="gngs/FASTQRead.html">FASTQRead</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#fatherOf(java.lang.String)" title="Method in Pedigree">fatherOf(String)</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#fatherOf(java.lang.String)" title="Method in Pedigrees">fatherOf(String)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SamplesToPed.html#fillRelationships(gngs.Pedigree)" title="Method in SamplesToPed">fillRelationships(Pedigree)</a></span> - Method in <a href="DefaultPackage/SamplesToPed.html">SamplesToPed</a>
</dt><dd> <div class="block">Fill in all the relationship ids on the assumption that they are 
 relative to a single affected child.</div></dd>
<dt><span class="strong"><a href="gngs/FASTQ.html#filter(java.lang.String, java.lang.String, java.io.Writer, java.io.Writer, groovy.lang.Closure)" title="Method in FASTQ">filter(String, String, Writer, Writer, Closure)</a></span> - Method in <a href="gngs/FASTQ.html">FASTQ</a>
</dt><dd> <div class="block">Filter paired reads from fileName1 and fileName2 and write them to 
 uncompressed output files with extensions .filter.fastq based on the 
 input file names, only where true is returned from the given closure</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#filter(java.lang.String, groovy.lang.Closure)" title="Method in SAM">filter(String, Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Filter the SAM/BAM file to include only the reads for which the given closure returns true</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#filter(java.util.Map, java.io.InputStream, List<Pedigree>, groovy.lang.Closure)" title="Method in VCF">filter(Map, InputStream, List&lt;Pedigree&gt;, Closure)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TargetedCNVAnnotator.html#filterByType(java.lang.String, List<Region>)" title="Method in TargetedCNVAnnotator">filterByType(String, List&lt;Region&gt;)</a></span> - Method in <a href="DefaultPackage/TargetedCNVAnnotator.html">TargetedCNVAnnotator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#filterOrderedPairs(java.util.Map, java.lang.String, groovy.lang.Closure)" title="Method in SAM">filterOrderedPairs(Map, String, Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Execute the given closure with an actor (parallel thread) set to write ordered 
 pairs to the given output file, based on this BAM file.</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#filterSamples(java.io.PrintStream, List<String>)" title="Method in VCF">filterSamples(PrintStream, List&lt;String&gt;)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Print out a version of this VCF with only the given samples included</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#find(gngs.Variant)" title="Method in VCF">find(Variant)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Attempt to locate a variant having the same change as the given variant
 inside this VCF.</div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#find(java.lang.String, int, int, groovy.lang.Closure)" title="Method in VCFIndex">find(String, int, int, Closure)</a></span> - Method in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block">Find the first variant in an interval that returns true from the
 given closure</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#findAlleleIndex(gngs.Allele)" title="Method in Variant">findAlleleIndex(Allele)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return the index of the allele (if any) that matches the give other variant</div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#findAnnovarVariant(java.lang.String, def, def, java.lang.String, int)" title="Method in VCFIndex">findAnnovarVariant(String, def, def, String, int)</a></span> - Method in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block">Attempts to locate the given Annovar variant in this VCF file.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/TargetedCNVAnnotator.html#findCompatibleRanges(gngs.IRegion)" title="Method in TargetedCNVAnnotator">findCompatibleRanges(IRegion)</a></span> - Method in <a href="DefaultPackage/TargetedCNVAnnotator.html">TargetedCNVAnnotator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/FindCompoundHets.html" title="Script in FindCompoundHets">FindCompoundHets</a></span> - Script in <a href="./FindCompoundHets/package-summary.html">FindCompoundHets</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VEPDiff.html#findDifferences()" title="Method in VEPDiff">findDifferences()</a></span> - Method in <a href="DefaultPackage/VEPDiff.html">VEPDiff</a>
</dt><dd> <div class="block">Return a list of maps containing identical variants having different VEP consequences or features
 between the two VCFs</div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#findMaximalUnrelatedSet()" title="Method in Pedigree">findMaximalUnrelatedSet()</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block">Compute the largest set of samples within the pedigree who are unrelated. 
 </div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#findMaximalUnrelatedSet(java.lang.String)" title="Method in Pedigrees">findMaximalUnrelatedSet(String)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block">Find the largest possible set of samples that are unrelated to 
 the given sample. </div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#findMaxMaf(def)" title="Method in VCFtoHTML">findMaxMaf(def)</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/BAM2Ped.html#findParents(java.lang.String)" title="Method in BAM2Ped">findParents(String)</a></span> - Method in <a href="gngs/tools/BAM2Ped.html">BAM2Ped</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#findPedigreeBySampleIndex(int)" title="Method in VCF">findPedigreeBySampleIndex(int)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DepthOfCoverage.html#findReadRegion(htsjdk.samtools.SAMRecord)" title="Method in DepthOfCoverage">findReadRegion(SAMRecord)</a></span> - Method in <a href="gngs/tools/DepthOfCoverage.html">DepthOfCoverage</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#findSample(java.lang.String)" title="Method in VariantDB">findSample(String)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Return row of database representing sample, or null if it does not
 exist.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#findVariant(gngs.Variant, gngs.Allele)" title="Method in VariantDB">findVariant(Variant, Allele)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Find a variant in the database by its start position</div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#first()" title="Method in RangeIndex">first()</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#fixGenoTypeOrder()" title="Method in Variant">fixGenoTypeOrder()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">VCF requires genotype to be the 1st field, but some tools (R, grrrr) write it in whatever order they 
 feel like. </div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairWriter.html#flushBuffer()" title="Method in OrderedPairWriter">flushBuffer()</a></span> - Method in <a href="gngs/OrderedPairWriter.html">OrderedPairWriter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFormatter.html#flushBuffer()" title="Method in PairFormatter">flushBuffer()</a></span> - Method in <a href="gngs/pair/PairFormatter.html">PairFormatter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#flushSpooledReads(groovy.lang.Closure)" title="Method in OrderedPairReader">flushSpooledReads(Closure)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#font(java.awt.Color, int, Closure)" title="Method in PDF">font(Color, int, Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#fontSize(float, Closure)" title="Method in PDF">fontSize(float, Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#forceQueryOverflowRead(int)" title="Method in OrderedPairReader">forceQueryOverflowRead(int)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTA.html#format(java.lang.String, java.lang.String)" title="Method in FASTA">format(String, String)</a></span> - Method in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SimpleLogFormatter.html#format(java.util.logging.LogRecord)" title="Method in SimpleLogFormatter">format(LogRecord)</a></span> - Method in <a href="gngs/SimpleLogFormatter.html">SimpleLogFormatter</a>
</dt><dd> <div class="block">A Custom format implementation that is designed for brevity.</div></dd>
<dt><span class="strong"><a href="gngs/FormatMetaData.html" title="Class in gngs">FormatMetaData</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageStats.html#fractionAbove(int)" title="Method in CoverageStats">fractionAbove(int)</a></span> - Method in <a href="gngs/CoverageStats.html">CoverageStats</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#fromSingletons(List<String>)" title="Method in Pedigrees">fromSingletons(List&lt;String&gt;)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block">A convenience method that creates a set of pedigrees from set of 
 unrelated singletons. </div></dd>
</dl>

    
<a name="_G_"><!-- --></a>
<h2 class="title">G</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#G" title="Field in FASTA">G</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html" title="Class in gngs.tools">Gaps</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block">A tool that computes gaps in coverage based on BEDTools format output.</div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#Gaps(groovy.util.OptionAccessor)" title="Constructor in Gaps">Gaps(OptionAccessor)</a></span> - Constructor in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/GATKIntervalSummary.html" title="Class in GATKIntervalSummary">GATKIntervalSummary</a></span> - Class in <a href="./GATKIntervalSummary/package-summary.html">GATKIntervalSummary</a>
</dt><dd><div class="block">Reads output of GATK DepthOfCoverage interval summary and
 makes it available as a Regions object</div></dd>
<dt><span class="strong"><a href="DefaultPackage/GATKIntervalSummary.html#GATKIntervalSummary(java.lang.String)" title="Constructor in GATKIntervalSummary">GATKIntervalSummary(String)</a></span> - Constructor in <a href="DefaultPackage/GATKIntervalSummary.html">GATKIntervalSummary</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/OMIM.html#GENEMAP2_FIELDS" title="Field in OMIM">GENEMAP2_FIELDS</a></span> - Field in <a href="DefaultPackage/OMIM.html">OMIM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#generateTestTrios()" title="Method in VCFtoPedigree">generateTestTrios()</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/BAM2Ped.html#genotypeBAM(gngs.SAM)" title="Method in BAM2Ped">genotypeBAM(SAM)</a></span> - Method in <a href="gngs/tools/BAM2Ped.html">BAM2Ped</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/BAM2Ped.html#genotypeVariant(gngs.SAM, gngs.Variant)" title="Method in BAM2Ped">genotypeVariant(SAM, Variant)</a></span> - Method in <a href="gngs/tools/BAM2Ped.html">BAM2Ped</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ProgressCounter.html#getAbort()" title="Method in ProgressCounter">getAbort()</a></span> - Method in <a href="gngs/ProgressCounter.html">ProgressCounter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#getAffected()" title="Method in Pedigree">getAffected()</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#getAffected()" title="Method in Pedigrees">getAffected()</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.Pileup.html#getAlignments()" title="Method in PileupIterator.Pileup">getAlignments()</a></span> - Method in <a href="gngs/PileupIterator.Pileup.html">PileupIterator.Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getAlleleBalance(int, int, int)" title="Method in Variant">getAlleleBalance(int, int, int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return the distance of the balance of the default allele from 0.5
 for the specififed alternate allele (reference = 0)</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getAlleleDepths(int)" title="Method in Variant">getAlleleDepths(int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return the number of reads supporting the given allele as a list
 with one entry for each sample in the VCF</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getAlleles()" title="Method in Variant">getAlleles()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return a list of Allele objects representing alleles present on this line of the VCF</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getAllelesAndTypes()" title="Method in Variant">getAllelesAndTypes()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return a list of each allele and its type.
 </div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#getAllSubjects()" title="Method in Pedigrees">getAllSubjects()</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieNode.html#getAllValues(int)" title="Method in TrieNode">getAllValues(int)</a></span> - Method in <a href="trie/TrieNode.html">TrieNode</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordCategory.html#getAlternateAlignments(htsjdk.samtools.BAMRecord)" title="Method in SAMRecordCategory">getAlternateAlignments(BAMRecord)</a></span> - Method in <a href="gngs/SAMRecordCategory.html">SAMRecordCategory</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#getAnnotation(java.lang.String, def)" title="Method in VariantDB">getAnnotation(String, def)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Search for the given value in the annotations as a frequency
 value.</div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#getAt(java.lang.String)" title="Method in Pedigree">getAt(String)</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#getAt(java.lang.String)" title="Method in Pedigrees">getAt(String)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block">Get Subject by their id</div></dd>
<dt><span class="strong"><a href="gngs/PrefixTrie.html#getAt(java.lang.String)" title="Method in PrefixTrie">getAt(String)</a></span> - Method in <a href="gngs/PrefixTrie.html">PrefixTrie</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#getAt(java.lang.Object)" title="Method in Regions">getAt(Object)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieNode.html#getAt(java.lang.String)" title="Method in TrieNode">getAt(String)</a></span> - Method in <a href="trie/TrieNode.html">TrieNode</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#getBoundaries()" title="Method in RangeIndex">getBoundaries()</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Returns a range for each boundary within the index.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/CpipeRunSet.html#getCaptureTarget(java.lang.String)" title="Method in CpipeRunSet">getCaptureTarget(String)</a></span> - Method in <a href="DefaultPackage/CpipeRunSet.html">CpipeRunSet</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/CpipeRunSet.html#getCaptureTargetFile(java.lang.String)" title="Method in CpipeRunSet">getCaptureTargetFile(String)</a></span> - Method in <a href="DefaultPackage/CpipeRunSet.html">CpipeRunSet</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/IRegion.html#getChr()" title="Method in IRegion">getChr()</a></span> - Method in <a href="gngs/IRegion.html">IRegion</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ConcordanceScore.html#getConcordance()" title="Method in ConcordanceScore">getConcordance()</a></span> - Method in <a href="gngs/tools/ConcordanceScore.html">ConcordanceScore</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BaseCounts.html#getConsensus()" title="Method in BaseCounts">getConsensus()</a></span> - Method in <a href="gngs/BaseCounts.html">BaseCounts</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SnpEffInfo.html#getConsequence()" title="Method in SnpEffInfo">getConsequence()</a></span> - Method in <a href="gngs/SnpEffInfo.html">SnpEffInfo</a>
</dt><dd> <div class="block">This is an alias for the SNPEFF 'type' which is useful when trying to make
 code use both SNPEFF and VEP annotations</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getConsequence(int)" title="Method in Variant">getConsequence(int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return the consequence of the specified allele, at the moment,
 from VEP annotations (later, from others). </div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#getConsequenceCounts()" title="Method in VCF">getConsequenceCounts()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Calculate the number of variants for each different VEP consequence, returning
 a map of consequence => count.</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#getContigs()" title="Method in SAM">getContigs()</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getDosages(int)" title="Method in Variant">getDosages(int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#getExons(java.lang.String, boolean)" title="Method in RefGenes">getExons(String, boolean)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block">Return a regions object containing a region for each unique exon in the transcript.
 </div></dd>
<dt><span class="strong"><a href="gngs/Region.html#getExtra()" title="Method in Region">getExtra()</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#getExtrasAtPosition(java.lang.String, int)" title="Method in Regions">getExtrasAtPosition(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#getFemales()" title="Method in Pedigrees">getFemales()</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#getFormatMetaData()" title="Method in VCF">getFormatMetaData()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#getFreq(java.lang.String, def)" title="Method in VariantDB">getFreq(String, def)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Search for the given value in the annotations as a frequency
 value.</div></dd>
<dt><span class="strong"><a href="gngs/Region.html#getFrom()" title="Method in Region">getFrom()</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#getGeneRegion(java.lang.String)" title="Method in RefGenes">getGeneRegion(String)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block">Look up the entire span of a gene from its HGNC symbol</div></dd>
<dt><span class="strong"><a href="gngs/tools/ExtractFASTQ.html#getGeneRegions(java.lang.String, List<String>)" title="Method in ExtractFASTQ">getGeneRegions(String, List&lt;String&gt;)</a></span> - Method in <a href="gngs/tools/ExtractFASTQ.html">ExtractFASTQ</a>
</dt><dd> <div class="block">Attempt to interpret the given list of genes as either a list of gene symbols
 or a text file specifying the genes to read.</div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#getGenes(gngs.Region)" title="Method in RefGenes">getGenes(Region)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getGenes(java.lang.String)" title="Method in Variant">getGenes(String)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return the list of genes impacted by this variant, in a manner that is
 neutral to the annotator used (VEP and SnpEFF supported)</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#getGenes()" title="Method in VCF">getGenes()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getGenotypeDepth(Map<String, Object>, int)" title="Method in Variant">getGenotypeDepth(Map&lt;String, Object&gt;, int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#getHighQualityHets(gngs.VCF)" title="Method in VCF">getHighQualityHets(VCF)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Finds a set of high quality SNPs that distinguish this VCF from the
 other VCF provided.</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#getIndex()" title="Method in VCF">getIndex()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getInfo()" title="Method in Variant">getInfo()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#getInfoMetaData(java.lang.String)" title="Method in VCF">getInfoMetaData(String)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getLeftPosition()" title="Method in SAMRecordPair">getLeftPosition()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#getMales()" title="Method in Pedigrees">getMales()</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getMaxAlleleBalance()" title="Method in Variant">getMaxAlleleBalance()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return the maximum allele balance for any sample for the default
 alternate allele.</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getMaxEffect()" title="Method in Variant">getMaxEffect()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return the most impactful SnpEff effect for this variant</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getMaxVep()" title="Method in Variant">getMaxVep()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return the details of the most severe VEP consequence, as ranked by
 <a href='../gngs/VEPConsequences.html#RANKED_CONSEQUENCES' title='VEPConsequences.RANKED_CONSEQUENCES'>VEPConsequences.RANKED_CONSEQUENCES</a>.</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getMaxVepImpact()" title="Method in Variant">getMaxVepImpact()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getMaxVepMaf()" title="Method in Variant">getMaxVepMaf()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#getMinMappingQuality()" title="Method in PileupIterator">getMinMappingQuality()</a></span> - Method in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getMissingReadPosition()" title="Method in SAMRecordPair">getMissingReadPosition()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block">If one of the pair is missing, return
 @return</div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getMissingReadReferenceName()" title="Method in SAMRecordPair">getMissingReadReferenceName()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block">If one of the pair is missing, return
 @return</div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getNonMissingRead()" title="Method in SAMRecordPair">getNonMissingRead()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#getNumberOfOpenFiles()" title="Method in Utils">getNumberOfOpenFiles()</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#getNumberOfRanges()" title="Method in Regions">getNumberOfRanges()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#getOverlaps(int, int, boolean)" title="Method in RangeIndex">getOverlaps(int, int, boolean)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Return a list of the ranges in this index that overlap the given range.
 </div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#getOverlaps(java.lang.String, int, int)" title="Method in Regions">getOverlaps(String, int, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Return a list of ranges that overlap the specified range. 
 </div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#getPair(java.lang.String)" title="Method in MissingMateIndex">getPair(String)</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Subject.html#getParentIds()" title="Method in Subject">getParentIds()</a></span> - Method in <a href="gngs/Subject.html">Subject</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieQuery.html#getPath()" title="Method in TrieQuery">getPath()</a></span> - Method in <a href="trie/TrieQuery.html">TrieQuery</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#getPedData(gngs.Subject)" title="Method in Pedigree">getPedData(Subject)</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getPedigrees()" title="Method in Variant">getPedigrees()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return a list of all the pedigrees that contain this variant</div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getR1Pos()" title="Method in SAMRecordPair">getR1Pos()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getR1Position()" title="Method in SAMRecordPair">getR1Position()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getR1ReferenceIndex()" title="Method in SAMRecordPair">getR1ReferenceIndex()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getR1ReferenceName()" title="Method in SAMRecordPair">getR1ReferenceName()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getR2Pos()" title="Method in SAMRecordPair">getR2Pos()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getR2Position()" title="Method in SAMRecordPair">getR2Position()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getR2ReferenceIndex()" title="Method in SAMRecordPair">getR2ReferenceIndex()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getR2ReferenceName()" title="Method in SAMRecordPair">getR2ReferenceName()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageBlock.html#getRange()" title="Method in CoverageBlock">getRange()</a></span> - Method in <a href="gngs/CoverageBlock.html">CoverageBlock</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/IRegion.html#getRange()" title="Method in IRegion">getRange()</a></span> - Method in <a href="gngs/IRegion.html">IRegion</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getRange()" title="Method in Variant">getRange()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/InheritanceConcordance.html#getRate()" title="Method in InheritanceConcordance">getRate()</a></span> - Method in <a href="gngs/tools/InheritanceConcordance.html">InheritanceConcordance</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangedData.html#getReader(java.lang.String)" title="Method in RangedData">getReader(String)</a></span> - Method in <a href="gngs/RangedData.html">RangedData</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#getReadGroups()" title="Method in SAM">getReadGroups()</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">@return Return the list of read groups present in the SAM file</div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#getReadIterator()" title="Method in PileupIterator">getReadIterator()</a></span> - Method in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#getReadLength()" title="Method in CompactReadPair">getReadLength()</a></span> - Method in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getReadLength()" title="Method in SAMRecordPair">getReadLength()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getReadName()" title="Method in SAMRecordPair">getReadName()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getRegion()" title="Method in Variant">getRegion()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Pairs.html#getRegions()" title="Method in Pairs">getRegions()</a></span> - Method in <a href="gngs/tools/Pairs.html">Pairs</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#getRegions(java.lang.String, int, int)" title="Method in SAM">getRegions(String, int, int)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#getResolveQueueSize()" title="Method in MissingMateIndex">getResolveQueueSize()</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getRightPosition()" title="Method in SAMRecordPair">getRightPosition()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#getSamples()" title="Method in Pedigree">getSamples()</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#getSamples()" title="Method in SAM">getSamples()</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BaseCounts.html#getScore()" title="Method in BaseCounts">getScore()</a></span> - Method in <a href="gngs/BaseCounts.html">BaseCounts</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#getSize()" title="Method in VCF">getSize()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getSnpEffInfo()" title="Method in Variant">getSnpEffInfo()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return a list of SnpEffInfo objects, each describing
 a separate SnpEff effect caused by the variant
 @return</div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#getSpliceSites(java.lang.String)" title="Method in RefGenes">getSpliceSites(String)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block">Return a regions object containing a region for each unique exon in the transcript.
 </div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#getSubjectsBySex(gngs.Sex)" title="Method in Pedigrees">getSubjectsBySex(Sex)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#getSummary()" title="Method in Pileup">getSummary()</a></span> - Method in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block">Returns a summary of the coverage at the location:</div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.Pileup.html#getSummary()" title="Method in PileupIterator.Pileup">getSummary()</a></span> - Method in <a href="gngs/PileupIterator.Pileup.html">PileupIterator.Pileup</a>
</dt><dd> <div class="block">Returns a summary of the coverage at the location:</div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#getSummaryAsMap()" title="Method in Pileup">getSummaryAsMap()</a></span> - Method in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block">Convenience method to return counts of each base as a map indexed
 by the base as a string. </div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.Pileup.html#getSummaryAsMap()" title="Method in PileupIterator.Pileup">getSummaryAsMap()</a></span> - Method in <a href="gngs/PileupIterator.Pileup.html">PileupIterator.Pileup</a>
</dt><dd> <div class="block">Convenience method to return counts of each base as a map indexed
 by the base as a string. </div></dd>
<dt><span class="strong"><a href="gngs/Region.html#getTo()" title="Method in Region">getTo()</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getTotalDepth(java.lang.String)" title="Method in Variant">getTotalDepth(String)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#getTranscriptExons(java.lang.String)" title="Method in RefGenes">getTranscriptExons(String)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#getTranscriptIds(java.lang.String)" title="Method in RefGenes">getTranscriptIds(String)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block">Return the names of all the transcripts (NM_... identifiers) for the
 given HGNC symbol</div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#getTranscripts(java.lang.String)" title="Method in RefGenes">getTranscripts(String)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block">Return a list of transcript objects as Regions having all the 
 full properties of a transcript (see COLUMN_NAMES)</div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#getUnaffected()" title="Method in Pedigree">getUnaffected()</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#getUnaffected()" title="Method in Pedigrees">getUnaffected()</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#getUnmapped()" title="Method in CompactReadPair">getUnmapped()</a></span> - Method in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadPair.html#getUnmapped()" title="Method in ReadPair">getUnmapped()</a></span> - Method in <a href="gngs/ReadPair.html">ReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#getUnmapped()" title="Method in SAMRecordPair">getUnmapped()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block">@return true if either r1 or r2 is unmapped</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFSimilarity.html#getVariantSet(gngs.VCF, java.lang.String)" title="Method in VCFSimilarity">getVariantSet(VCF, String)</a></span> - Method in <a href="DefaultPackage/VCFSimilarity.html">VCFSimilarity</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/CpipeRunSet.html#getVCFPaths(java.lang.String)" title="Method in CpipeRunSet">getVCFPaths(String)</a></span> - Method in <a href="DefaultPackage/CpipeRunSet.html">CpipeRunSet</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#getVepColumns(java.lang.String)" title="Method in VCF">getVepColumns(String)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Dedicated support for dynamically returning the VEP columns present in this VCF file</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#getVepInfo()" title="Method in Variant">getVepInfo()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Align.html#global(Map, java.lang.String, java.lang.String)" title="Method in Align">global(Map, String, String)</a></span> - Method in <a href="DefaultPackage/Align.html">Align</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/GRange.html" title="Class in gngs">GRange</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Range over an interval</div></dd>
<dt><span class="strong"><a href="gngs/GRange.html#GRange(int, int, java.lang.Object)" title="Constructor in GRange">GRange(int, int, Object)</a></span> - Constructor in <a href="gngs/GRange.html">GRange</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/GRegion.html" title="Class in gngs">GRegion</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/GRegion.html#GRegion(java.lang.String, groovy.lang.Range)" title="Constructor in GRegion">GRegion(String, Range)</a></span> - Constructor in <a href="gngs/GRegion.html">GRegion</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#guessSex(int)" title="Method in VCF">guessSex(int)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Attempt to guess the sex of a human VCF file by sampling high quality 
 variants on the X chromosome.</div></dd>
</dl>

    
<a name="_H_"><!-- --></a>
<h2 class="title">H</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#H" title="Field in FASTA">H</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#hasBothReads()" title="Method in SAMRecordPair">hasBothReads()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Hash.html" title="Class in gngs">Hash</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Hash.html#Hash()" title="Constructor in Hash">Hash()</a></span> - Constructor in <a href="gngs/Hash.html">Hash</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#hasInfo(java.lang.String)" title="Method in VCF">hasInfo(String)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Return true if this VCF file contains the specified INFO tags</div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#hasNext()" title="Method in Pileup">hasNext()</a></span> - Method in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#hasNext()" title="Method in PileupIterator">hasNext()</a></span> - Method in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RegionIterator.html#hasNext()" title="Method in RegionIterator">hasNext()</a></span> - Method in <a href="gngs/RegionIterator.html">RegionIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#head(Closure)" title="Method in PDF">head(Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Histogram.html" title="Class in Histogram">Histogram</a></span> - Class in <a href="./Histogram/package-summary.html">Histogram</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Histogram.html#Histogram(java.lang.Iterable)" title="Constructor in Histogram">Histogram(Iterable)</a></span> - Constructor in <a href="DefaultPackage/Histogram.html">Histogram</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#human(java.lang.Number)" title="Method in Utils">human(Number)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#humanBp(java.lang.Number, def)" title="Method in Utils">humanBp(Number, def)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_I_"><!-- --></a>
<h2 class="title">I</h2>
<dl>

<dt><span class="strong"><a href="DefaultPackage/IlluminaFileNameParser.html" title="Class in IlluminaFileNameParser">IlluminaFileNameParser</a></span> - Class in <a href="./IlluminaFileNameParser/package-summary.html">IlluminaFileNameParser</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/IlluminaFileNameParser.DIALECT.html" title="Enum in IlluminaFileNameParser">IlluminaFileNameParser.DIALECT</a></span> - Enum in <a href="./IlluminaFileNameParser/DIALECT/package-summary.html">IlluminaFileNameParser.DIALECT</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#img(java.lang.String)" title="Method in PDF">img(String)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PairReaderStats.html#inBlock(htsjdk.samtools.SAMRecord, int)" title="Method in PairReaderStats">inBlock(SAMRecord, int)</a></span> - Method in <a href="gngs/PairReaderStats.html">PairReaderStats</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#includeVariant(gngs.Variant)" title="Method in VCFtoPedigree">includeVariant(Variant)</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTAIndex.html#index(gngs.FASTA, gngs.Regions)" title="Method in FASTAIndex">index(FASTA, Regions)</a></span> - Method in <a href="gngs/FASTAIndex.html">FASTAIndex</a>
</dt><dd> <div class="block">Index the given fasta, masked using the given BED file</div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#index()" title="Method in RefGenes">index()</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#index(java.io.File)" title="Method in SAM">index(File)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Create an index for the given BAM file</div></dd>
<dt><span class="strong"><a href="gngs/tools/InheritanceConcordance.html" title="Class in gngs.tools">InheritanceConcordance</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#init()" title="Method in VariantDB">init()</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Check the database exists and upgrade it if necessary</div></dd>
<dt><span class="strong"><a href="gngs/pair/PairScanner.html#initLocators()" title="Method in PairScanner">initLocators()</a></span> - Method in <a href="gngs/pair/PairScanner.html">PairScanner</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieQuery.html#insertion(java.lang.String)" title="Method in TrieQuery">insertion(String)</a></span> - Method in <a href="trie/TrieQuery.html">TrieQuery</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/GRange.html#intersect(groovy.lang.IntRange)" title="Method in GRange">intersect(IntRange)</a></span> - Method in <a href="gngs/GRange.html">GRange</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#intersect(int, int)" title="Method in RangeIndex">intersect(int, int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Return a list of ranges that intersect the given start and end points.</div></dd>
<dt><span class="strong"><a href="gngs/Region.html#intersect(gngs.IRegion)" title="Method in Region">intersect(IRegion)</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#intersect(gngs.Regions)" title="Method in Regions">intersect(Regions)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Returns a set of regions representing each region in this Regions
 intersected with the Regions in the other regions.</div></dd>
<dt><span class="strong"><a href="gngs/LXPos.html#intToChr(int)" title="Method in LXPos">intToChr(int)</a></span> - Method in <a href="gngs/LXPos.html">LXPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/XPos.html#intToChr(int)" title="Method in XPos">intToChr(int)</a></span> - Method in <a href="gngs/XPos.html">XPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/IRegion.html" title="Interface in gngs">IRegion</a></span> - Interface in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Subject.html#isAffected()" title="Method in Subject">isAffected()</a></span> - Method in <a href="gngs/Subject.html">Subject</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BED.html#isCase(gngs.IRegion)" title="Method in BED">isCase(IRegion)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#isCase(int)" title="Method in RangeIndex">isCase(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Support for 'in' operator</div></dd>
<dt><span class="strong"><a href="gngs/Region.html#isCase(gngs.IRegion)" title="Method in Region">isCase(IRegion)</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block">Return true of this region overlaps the other.</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#isCase(gngs.IRegion)" title="Method in Variant">isCase(IRegion)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return true if this variant overlaps the given range</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#isCase(java.lang.Object)" title="Method in VCF">isCase(Object)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#isCase(gngs.Variant)" title="Method in VCFIndex">isCase(Variant)</a></span> - Method in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RelationshipType.html#isChild()" title="Method in RelationshipType">isChild()</a></span> - Method in <a href="gngs/RelationshipType.html">RelationshipType</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Subject.html#isChild()" title="Method in Subject">isChild()</a></span> - Method in <a href="gngs/Subject.html">Subject</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#isChimeric()" title="Method in CompactReadPair">isChimeric()</a></span> - Method in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadPair.html#isChimeric()" title="Method in ReadPair">isChimeric()</a></span> - Method in <a href="gngs/ReadPair.html">ReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#isChimeric()" title="Method in SAMRecordPair">isChimeric()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#isEmpty()" title="Method in Regions">isEmpty()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Region.html#isMinorContig(java.lang.String)" title="Method in Region">isMinorContig(String)</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#isNewChromosome(htsjdk.samtools.SAMRecord)" title="Method in OrderedPairReader">isNewChromosome(SAMRecord)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#isProcessableRead(htsjdk.samtools.SAMRecord)" title="Method in OrderedPairReader">isProcessableRead(SAMRecord)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#isSV()" title="Method in Variant">isSV()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TestTrio.html#isTrio()" title="Method in TestTrio">isTrio()</a></span> - Method in <a href="DefaultPackage/TestTrio.html">TestTrio</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SnpEffInfo.html#isTruncating()" title="Method in SnpEffInfo">isTruncating()</a></span> - Method in <a href="gngs/SnpEffInfo.html">SnpEffInfo</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/GATKIntervalSummary.html#iterator()" title="Method in GATKIntervalSummary">iterator()</a></span> - Method in <a href="DefaultPackage/GATKIntervalSummary.html">GATKIntervalSummary</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#iterator()" title="Method in RangeIndex">iterator()</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#iterator()" title="Method in Regions">iterator()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#iterator()" title="Method in VCF">iterator()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#iterator(java.lang.String, int, int)" title="Method in VCFIndex">iterator(String, int, int)</a></span> - Method in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#iteratorAt(int)" title="Method in RangeIndex">iteratorAt(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Return an iterator that will return each range in the index
 in genomic order by starting position, starting from the given 
 position.</div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#iteratorPosition" title="Field in PileupIterator">iteratorPosition</a></span> - Field in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_J_"><!-- --></a>
<h2 class="title">J</h2>
<dl>

<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#json(def)" title="Method in VCFtoHTML">json(def)</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_K_"><!-- --></a>
<h2 class="title">K</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#K" title="Field in FASTA">K</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_L_"><!-- --></a>
<h2 class="title">L</h2>
<dl>

<dt><span class="strong"><a href="gngs/RangeIndex.html#last()" title="Method in RangeIndex">last()</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SamplesToPed.html#lcs(List<String>)" title="Method in SamplesToPed">lcs(List&lt;String&gt;)</a></span> - Method in <a href="DefaultPackage/SamplesToPed.html">SamplesToPed</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/LinePlot.html" title="Class in LinePlot">LinePlot</a></span> - Class in <a href="./LinePlot/package-summary.html">LinePlot</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/LinePlot.html#LinePlot(java.lang.Iterable)" title="Constructor in LinePlot">LinePlot(Iterable)</a></span> - Constructor in <a href="DefaultPackage/LinePlot.html">LinePlot</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SamplesToPed.html#linkParentRelationship(gngs.Pedigree, gngs.Subject, gngs.RelationshipType)" title="Method in SamplesToPed">linkParentRelationship(Pedigree, Subject, RelationshipType)</a></span> - Method in <a href="DefaultPackage/SamplesToPed.html">SamplesToPed</a>
</dt><dd> <div class="block">Find the other individual</div></dd>
<dt><span class="strong"><a href="gngs/BED.html#load(java.util.Map)" title="Method in BED">load(Map)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block">Load the data for this BED file into memory.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/GATKIntervalSummary.html#load()" title="Method in GATKIntervalSummary">load()</a></span> - Method in <a href="DefaultPackage/GATKIntervalSummary.html">GATKIntervalSummary</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangedData.html#load(java.util.Map, groovy.lang.Closure)" title="Method in RangedData">load(Map, Closure)</a></span> - Method in <a href="gngs/RangedData.html">RangedData</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#load(java.io.Reader)" title="Method in RefGenes">load(Reader)</a></span> - Method in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/RestrictionEnzyme.html#load(InputStream)" title="Method in RestrictionEnzyme">load(InputStream)</a></span> - Method in <a href="DefaultPackage/RestrictionEnzyme.html">RestrictionEnzyme</a>
</dt><dd> <div class="block">Convenience method to load enzymes from UCSC's cutter table format</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#load(groovy.lang.Closure)" title="Method in VCF">load(Closure)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/CpipeRunSet.html#loadGaps()" title="Method in CpipeRunSet">loadGaps()</a></span> - Method in <a href="DefaultPackage/CpipeRunSet.html">CpipeRunSet</a>
</dt><dd> <div class="block">Return a Regions representing gaps for each sample in each batch,
 indexed first by batch, then by sample (batch.sample.gaps)</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#loadVcf(java.lang.String)" title="Method in VCFtoPedigree">loadVcf(String)</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#loadVCFs()" title="Method in VCFtoHTML">loadVCFs()</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#loadVCFs()" title="Method in VCFtoPedigree">loadVCFs()</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block">Load all the VCFs into the chosen sampling regions</div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#logGapInfo(java.lang.String, gngs.CoverageGaps)" title="Method in Gaps">logGapInfo(String, CoverageGaps)</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/LXPos.html" title="Class in gngs">LXPos</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
</dl>

    
<a name="_M_"><!-- --></a>
<h2 class="title">M</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#M" title="Field in FASTA">M</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/AdjustZygosity.html#main(def)" title="Method in AdjustZygosity">main(def)</a></span> - Method in <a href="gngs/tools/AdjustZygosity.html">AdjustZygosity</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/BAM2Ped.html#main(java.lang.String)" title="Method in BAM2Ped">main(String)</a></span> - Method in <a href="gngs/tools/BAM2Ped.html">BAM2Ped</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DepthOfCoverage.html#main(java.lang.String)" title="Method in DepthOfCoverage">main(String)</a></span> - Method in <a href="gngs/tools/DepthOfCoverage.html">DepthOfCoverage</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DownsampleFastq.html#main(java.lang.String)" title="Method in DownsampleFastq">main(String)</a></span> - Method in <a href="gngs/tools/DownsampleFastq.html">DownsampleFastq</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/ExonCoverageStats.html#main(java.lang.String)" title="Method in ExonCoverageStats">main(String)</a></span> - Method in <a href="DefaultPackage/ExonCoverageStats.html">ExonCoverageStats</a>
</dt><dd> <div class="block">Implicit main method for Groovy Scripts</div></dd>
<dt><span class="strong"><a href="gngs/tools/ExtractFASTQ.html#main(java.lang.String)" title="Method in ExtractFASTQ">main(String)</a></span> - Method in <a href="gngs/tools/ExtractFASTQ.html">ExtractFASTQ</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/FindCompoundHets.html#main(java.lang.String)" title="Method in FindCompoundHets">main(String)</a></span> - Method in <a href="DefaultPackage/FindCompoundHets.html">FindCompoundHets</a>
</dt><dd> <div class="block">Implicit main method for Groovy Scripts</div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#main(java.lang.String)" title="Method in Gaps">main(String)</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/MeanCoverageEstimator.html#main(java.lang.String)" title="Method in MeanCoverageEstimator">main(String)</a></span> - Method in <a href="gngs/tools/MeanCoverageEstimator.html">MeanCoverageEstimator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Pairs.html#main(def)" title="Method in Pairs">main(def)</a></span> - Method in <a href="gngs/tools/Pairs.html">Pairs</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Plot.html#main(java.lang.String)" title="Method in Plot">main(String)</a></span> - Method in <a href="DefaultPackage/Plot.html">Plot</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SamplesToPed.html#main(java.lang.String)" title="Method in SamplesToPed">main(String)</a></span> - Method in <a href="DefaultPackage/SamplesToPed.html">SamplesToPed</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Sex.html#main(def)" title="Method in Sex">main(def)</a></span> - Method in <a href="gngs/tools/Sex.html">Sex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SexKaryotyper.html#main(java.lang.String)" title="Method in SexKaryotyper">main(String)</a></span> - Method in <a href="DefaultPackage/SexKaryotyper.html">SexKaryotyper</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/SplitCommonRecords.html#main(def)" title="Method in SplitCommonRecords">main(def)</a></span> - Method in <a href="gngs/tools/SplitCommonRecords.html">SplitCommonRecords</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Table.html#main(java.lang.String)" title="Method in Table">main(String)</a></span> - Method in <a href="gngs/tools/Table.html">Table</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TargetedCNVAnnotator.html#main(java.lang.String)" title="Method in TargetedCNVAnnotator">main(String)</a></span> - Method in <a href="DefaultPackage/TargetedCNVAnnotator.html">TargetedCNVAnnotator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TestUpdate.html#main(def)" title="Method in TestUpdate">main(def)</a></span> - Method in <a href="DefaultPackage/TestUpdate.html">TestUpdate</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#main(java.lang.String)" title="Method in VariantDB">main(String)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Simple test program - all it does is creates the database</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFSimilarity.html#main(java.lang.String)" title="Method in VCFSimilarity">main(String)</a></span> - Method in <a href="DefaultPackage/VCFSimilarity.html">VCFSimilarity</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoDB.html#main(java.lang.String)" title="Method in VCFtoDB">main(String)</a></span> - Method in <a href="DefaultPackage/VCFtoDB.html">VCFtoDB</a>
</dt><dd> <div class="block">Implicit main method for Groovy Scripts</div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#main(java.lang.String)" title="Method in VCFtoHTML">main(String)</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#main(java.lang.String)" title="Method in VCFtoPedigree">main(String)</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VEPDiff.html#main(java.lang.String)" title="Method in VEPDiff">main(String)</a></span> - Method in <a href="DefaultPackage/VEPDiff.html">VEPDiff</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VInfo.html#main(java.lang.String)" title="Method in VInfo">main(String)</a></span> - Method in <a href="gngs/tools/VInfo.html">VInfo</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RegionScan.html#markUnpaired()" title="Method in RegionScan">markUnpaired()</a></span> - Method in <a href="gngs/RegionScan.html">RegionScan</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#max(int[])" title="Method in Utils">max(int[])</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CNVDatabase.html#maxFreq(java.util.Map, gngs.Region)" title="Method in CNVDatabase">maxFreq(Map, Region)</a></span> - Method in <a href="gngs/CNVDatabase.html">CNVDatabase</a>
</dt><dd> <div class="block">Find the maximum frequency of this CNV within any study within the database
 where the study has more than a minimum threshold size (default: 10 people).</div></dd>
<dt><span class="strong"><a href="gngs/DecipherCNVs.html#maxFreq(java.util.Map, gngs.Region)" title="Method in DecipherCNVs">maxFreq(Map, Region)</a></span> - Method in <a href="gngs/DecipherCNVs.html">DecipherCNVs</a>
</dt><dd> <div class="block">Find the maximum frequency of this CNV within any study within DDD where the 
 study has more than a minimum threshold size (default: 10 people).</div></dd>
<dt><span class="strong"><a href="gngs/DGV.html#maxFreq(java.util.Map, gngs.Region)" title="Method in DGV">maxFreq(Map, Region)</a></span> - Method in <a href="gngs/DGV.html">DGV</a>
</dt><dd> <div class="block">Find the maximum frequency of this CNV within any study within DGV where the 
 study has more than a minimum threshold size (default: 10 people).</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#meanCoverage(java.lang.String, int, int)" title="Method in SAM">meanCoverage(String, int, int)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/MeanCoverageEstimator.html" title="Class in gngs.tools">MeanCoverageEstimator</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block">Estimates the mean coverage of a BAM file quickly through random sampling.</div></dd>
<dt><span class="strong"><a href="gngs/tools/MeanCoverageEstimator.html#MeanCoverageEstimator(gngs.SAM, gngs.Regions)" title="Constructor in MeanCoverageEstimator">MeanCoverageEstimator(SAM, Regions)</a></span> - Constructor in <a href="gngs/tools/MeanCoverageEstimator.html">MeanCoverageEstimator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#merge(gngs.VCF)" title="Method in VCF">merge(VCF)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Perform a simplistic (emphasis on simplistic) merge between VCFs.</div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#min(int[])" title="Method in Utils">min(int[])</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#minPos()" title="Method in SAMRecordPair">minPos()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieQuery.html#mismatch(java.lang.String, int)" title="Method in TrieQuery">mismatch(String, int)</a></span> - Method in <a href="trie/TrieQuery.html">TrieQuery</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html" title="Class in gngs">MissingMateIndex</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#MissingMateIndex(int)" title="Constructor in MissingMateIndex">MissingMateIndex(int)</a></span> - Constructor in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMatePriority.html" title="Class in gngs">MissingMatePriority</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#motherOf(java.lang.String)" title="Method in Pedigree">motherOf(String)</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#motherOf(java.lang.String)" title="Method in Pedigrees">motherOf(String)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#movingWindow(int, java.lang.String, int, int, groovy.lang.Closure, groovy.lang.Closure)" title="Method in SAM">movingWindow(int, String, int, int, Closure, Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Call the given closure for each base position with a moving window of reads over that position</div></dd>
</dl>

    
<a name="_N_"><!-- --></a>
<h2 class="title">N</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#N" title="Field in FASTA">N</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#nearest(int)" title="Method in RangeIndex">nearest(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#nearest(java.lang.String, int)" title="Method in Regions">nearest(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Returns the range that is "closest" to the given
 position.</div></dd>
<dt><span class="strong"><a href="gngs/AsyncMateResolveWorker.html#needsResolution(gngs.SAMRecordPair)" title="Method in AsyncMateResolveWorker">needsResolution(SAMRecordPair)</a></span> - Method in <a href="gngs/AsyncMateResolveWorker.html">AsyncMateResolveWorker</a>
</dt><dd> <div class="block">Check if a read pair is still in a state where the mate needs to be resolved</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#newReader()" title="Method in SAM">newReader()</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#newWriter(java.lang.String)" title="Method in SAM">newWriter(String)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Return a new SAMFileWriter configured with the same settings as this 
 SAM. </div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#next()" title="Method in Pileup">next()</a></span> - Method in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#next()" title="Method in PileupIterator">next()</a></span> - Method in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html#next()" title="Method in PileupState">next()</a></span> - Method in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block">Move forward one reference position.
 </div></dd>
<dt><span class="strong"><a href="gngs/RegionIterator.html#next()" title="Method in RegionIterator">next()</a></span> - Method in <a href="gngs/RegionIterator.html">RegionIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#nextRange(int)" title="Method in RangeIndex">nextRange(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#nextRange(java.lang.String, int)" title="Method in Regions">nextRange(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Returns the next range that has its beginning
 closest to the given position.</div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html#nextReadPos" title="Field in PileupState">nextReadPos</a></span> - Field in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block">Offset from start of read for the next</div></dd>
<dt><span class="strong"><a href="gngs/pair/PairLocator.html#notInRegion(PairType, boolean)" title="Method in PairLocator">notInRegion(PairType, boolean)</a></span> - Method in <a href="gngs/pair/PairLocator.html">PairLocator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#notInRegions(gngs.Regions)" title="Method in CompactReadPair">notInRegions(Regions)</a></span> - Method in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadPair.html#notInRegions(gngs.Regions)" title="Method in ReadPair">notInRegions(Regions)</a></span> - Method in <a href="gngs/ReadPair.html">ReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#notInRegions(gngs.Regions)" title="Method in SAMRecordPair">notInRegions(Regions)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/NumericRegionComparator.html" title="Class in gngs">NumericRegionComparator</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
</dl>

    
<a name="_O_"><!-- --></a>
<h2 class="title">O</h2>
<dl>

<dt><span class="strong"><a href="DefaultPackage/OMIM.html" title="Class in OMIM">OMIM</a></span> - Class in <a href="./OMIM/package-summary.html">OMIM</a>
</dt><dd><div class="block">Parser for OMIM genemap files</div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#ONE_GIG" title="Field in VCFIndex">ONE_GIG</a></span> - Field in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTQ.html#openStream(java.lang.String)" title="Method in FASTQ">openStream(String)</a></span> - Method in <a href="gngs/FASTQ.html">FASTQ</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairActor.html" title="Class in gngs">OrderedPairActor</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">An actor that adds asynchronous writing to an OrderedPairWriter</div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairActor.html#OrderedPairActor(gngs.OrderedPairWriter)" title="Constructor in OrderedPairActor">OrderedPairActor(OrderedPairWriter)</a></span> - Constructor in <a href="gngs/OrderedPairActor.html">OrderedPairActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html" title="Class in gngs">OrderedPairReader</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A utility class for efficiently iterating all the read pairs in a BAM file.</div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#OrderedPairReader(java.util.Map, gngs.SAM)" title="Constructor in OrderedPairReader">OrderedPairReader(Map, SAM)</a></span> - Constructor in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairWriter.html" title="Class in gngs">OrderedPairWriter</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">When processing reads as pairs it can be tricky to write them out again while maintaining the
 order expected for a "sorted" BAM file. </div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairWriter.html#OrderedPairWriter(htsjdk.samtools.SAMFileWriter)" title="Constructor in OrderedPairWriter">OrderedPairWriter(SAMFileWriter)</a></span> - Constructor in <a href="gngs/OrderedPairWriter.html">OrderedPairWriter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageGaps.html#outputBlock(int)" title="Method in CoverageGaps">outputBlock(int)</a></span> - Method in <a href="gngs/CoverageGaps.html">CoverageGaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Overlap.html" title="Class in Overlap">Overlap</a></span> - Class in <a href="./Overlap/package-summary.html">Overlap</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/GRange.html#overlaps(groovy.lang.IntRange, groovy.lang.IntRange)" title="Method in GRange">overlaps(IntRange, IntRange)</a></span> - Method in <a href="gngs/GRange.html">GRange</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#overlaps(int, int)" title="Method in RangeIndex">overlaps(int, int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Return true if the given range overlaps at least one range in this
 RangeIndex.</div></dd>
<dt><span class="strong"><a href="gngs/Region.html#overlaps(gngs.IRegion)" title="Method in Region">overlaps(IRegion)</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block">@return true iff the specified region partially or fully overlaps this one</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#overlaps(gngs.Regions)" title="Method in Regions">overlaps(Regions)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Returns true if at least one region in r overlaps at least one
 region in this Regions object</div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#overlaps(gngs.Region)" title="Method in SAMRecordPair">overlaps(Region)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_P_"><!-- --></a>
<h2 class="title">P</h2>
<dl>

<dt><span class="strong"><a href="DefaultPackage/PDF.html#p(java.lang.String)" title="Method in PDF">p(String)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairBuffer.html" title="Class in gngs.pair">PairBuffer</a></span> - Class in <a href="./gngs/pair/package-summary.html">gngs.pair</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFilter.html" title="Class in gngs.pair">PairFilter</a></span> - Class in <a href="./gngs/pair/package-summary.html">gngs.pair</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFilter.html#PairFilter(gngs.pair.PairFormatter, java.lang.String)" title="Constructor in PairFilter">PairFilter(PairFormatter, String)</a></span> - Constructor in <a href="gngs/pair/PairFilter.html">PairFilter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFormatter.html" title="Class in gngs.pair">PairFormatter</a></span> - Class in <a href="./gngs/pair/package-summary.html">gngs.pair</a>
</dt><dd><div class="block">Converts reads to FASTQ format, and sends them in batches to a writer.</div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFormatter.html#PairFormatter(int, gngs.pair.PairWriter)" title="Constructor in PairFormatter">PairFormatter(int, PairWriter)</a></span> - Constructor in <a href="gngs/pair/PairFormatter.html">PairFormatter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairLocator.html" title="Class in gngs.pair">PairLocator</a></span> - Class in <a href="./gngs/pair/package-summary.html">gngs.pair</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairLocator.html#PairLocator(groovyx.gpars.actor.Actor)" title="Constructor in PairLocator">PairLocator(Actor)</a></span> - Constructor in <a href="gngs/pair/PairLocator.html">PairLocator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PairReaderStats.html" title="Class in gngs">PairReaderStats</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PairReaderStats.html#PairReaderStats(long, int)" title="Constructor in PairReaderStats">PairReaderStats(long, int)</a></span> - Constructor in <a href="gngs/PairReaderStats.html">PairReaderStats</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Pairs.html" title="Class in gngs.tools">Pairs</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block">Extracts read pairs from BAM or CRAM files sorted in coordinate order.</div></dd>
<dt><span class="strong"><a href="gngs/pair/PairScanner.html" title="Class in gngs.pair">PairScanner</a></span> - Class in <a href="./gngs/pair/package-summary.html">gngs.pair</a>
</dt><dd><div class="block">Scans a BAM file and feeds reads encountered to a pool of <a href='../../gngs/pair/PairLocator.html' title='PairLocator'>PairLocator</a>
 instances for matching to pairs. </div></dd>
<dt><span class="strong"><a href="gngs/pair/PairScanner.html#PairScanner(java.io.Writer, int, gngs.Regions, java.lang.String)" title="Constructor in PairScanner">PairScanner(Writer, int, Regions, String)</a></span> - Constructor in <a href="gngs/pair/PairScanner.html">PairScanner</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairWriter.html" title="Class in gngs.pair">PairWriter</a></span> - Class in <a href="./gngs/pair/package-summary.html">gngs.pair</a>
</dt><dd><div class="block">A simple class that whose purpose is purely to paralleise
 writing output to a Writer.</div></dd>
<dt><span class="strong"><a href="gngs/pair/PairWriter.html#PairWriter(java.io.Writer)" title="Constructor in PairWriter">PairWriter(Writer)</a></span> - Constructor in <a href="gngs/pair/PairWriter.html">PairWriter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BED.html#parse(java.lang.String)" title="Method in BED">parse(String)</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/ClinVar.html#parse()" title="Method in ClinVar">parse()</a></span> - Method in <a href="DefaultPackage/ClinVar.html">ClinVar</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/DecipherCNVs.html#parse()" title="Method in DecipherCNVs">parse()</a></span> - Method in <a href="gngs/DecipherCNVs.html">DecipherCNVs</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/DGV.html#parse()" title="Method in DGV">parse()</a></span> - Method in <a href="gngs/DGV.html">DGV</a>
</dt><dd> <div class="block">Parse the data source set for this database</div></dd>
<dt><span class="strong"><a href="DefaultPackage/IlluminaFileNameParser.html#parse(java.lang.String)" title="Method in IlluminaFileNameParser">parse(String)</a></span> - Method in <a href="DefaultPackage/IlluminaFileNameParser.html">IlluminaFileNameParser</a>
</dt><dd> <div class="block">Examples of the various formats we try to parse:</div></dd>
<dt><span class="strong"><a href="DefaultPackage/OMIM.html#parse(java.lang.String)" title="Method in OMIM">parse(String)</a></span> - Method in <a href="DefaultPackage/OMIM.html">OMIM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#parse(java.io.Reader, groovy.lang.Closure)" title="Method in Pedigrees">parse(Reader, Closure)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block">Parse a PED file from the given reader, with optional filtering via the provided closure</div></dd>
<dt><span class="strong"><a href="DefaultPackage/PicardMetrics.html#parse(java.lang.String)" title="Method in PicardMetrics">parse(String)</a></span> - Method in <a href="DefaultPackage/PicardMetrics.html">PicardMetrics</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#parse(java.lang.String, boolean)" title="Method in Variant">parse(String, boolean)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#parse(java.util.Map, java.io.InputStream, List<Pedigree>, groovy.lang.Closure)" title="Method in VCF">parse(Map, InputStream, List&lt;Pedigree&gt;, Closure)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#parseFormatMetaDataLine(java.lang.String)" title="Method in VCF">parseFormatMetaDataLine(String)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#parseGenoTypeFields(java.lang.String)" title="Method in Variant">parseGenoTypeFields(String)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#parseInfoMetaData(java.lang.String)" title="Method in VCF">parseInfoMetaData(String)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Parse a VCF INFO meta data line and return the values as a Map</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#parseLastHeaderLine()" title="Method in VCF">parseLastHeaderLine()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Extract sample names and other info from the final header line in the VCF file</div></dd>
<dt><span class="strong"><a href="gngs/LXPos.html#parsePos(long)" title="Method in LXPos">parsePos(long)</a></span> - Method in <a href="gngs/LXPos.html">LXPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/XPos.html#parsePos(long)" title="Method in XPos">parsePos(long)</a></span> - Method in <a href="gngs/XPos.html">XPos</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangedData.html#parseRegion(com.xlson.groovycsv.PropertyMapper)" title="Method in RangedData">parseRegion(PropertyMapper)</a></span> - Method in <a href="gngs/RangedData.html">RangedData</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Region.html#parseRegion(java.lang.String)" title="Method in Region">parseRegion(String)</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#parseVepColumns()" title="Method in VCF">parseVepColumns()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html" title="Class in PDF">PDF</a></span> - Class in <a href="./PDF/package-summary.html">PDF</a>
</dt><dd><div class="block">A light weight PDF Builder for creating PDFs to show results.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#PDF()" title="Constructor in PDF">PDF()</a></span> - Constructor in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html" title="Class in gngs">Pedigree</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Represents a set of relationships between individuals in a family</div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html" title="Class in gngs">Pedigrees</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairWriter.html#pending" title="Field in PairWriter">pending</a></span> - Field in <a href="gngs/pair/PairWriter.html">PairWriter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#perc(double)" title="Method in Utils">perc(double)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Phenotype.html" title="Class in Phenotype">Phenotype</a></span> - Class in <a href="./Phenotype/package-summary.html">Phenotype</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PicardMetrics.html" title="Class in PicardMetrics">PicardMetrics</a></span> - Class in <a href="./PicardMetrics/package-summary.html">PicardMetrics</a>
</dt><dd><div class="block">A simple class for parsing information out of the metrics file written by Picard</div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html" title="Class in gngs">Pileup</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Overall class containing pileup state</div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#Pileup(htsjdk.samtools.SAMFileReader, java.lang.String, int, int)" title="Constructor in Pileup">Pileup(SAMFileReader, String, int, int)</a></span> - Constructor in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#pileup(htsjdk.samtools.SAMFileReader, java.lang.String, int, int)" title="Method in SAM">pileup(SAMFileReader, String, int, int)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html" title="Class in gngs">PileupIterator</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Overall class containing pileup state</div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.Pileup.html" title="Class in gngsIterator.Pileup">PileupIterator.Pileup</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Inner class that captures pileup state at a specific position</div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#PileupIterator(htsjdk.samtools.SAMFileReader, java.lang.String, int, int)" title="Method in PileupIterator">PileupIterator(SAMFileReader, String, int, int)</a></span> - Method in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html" title="Class in gngs">PileupState</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Additional state about a read that is part of a pileup</div></dd>
<dt><span class="strong"><a href="DefaultPackage/Plot.html" title="Class in Plot">Plot</a></span> - Class in <a href="./Plot/package-summary.html">Plot</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Plot.html#Plot()" title="Constructor in Plot">Plot()</a></span> - Constructor in <a href="DefaultPackage/Plot.html">Plot</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/InheritanceConcordance.html#plus(gngs.tools.InheritanceConcordance)" title="Method in InheritanceConcordance">plus(InheritanceConcordance)</a></span> - Method in <a href="gngs/tools/InheritanceConcordance.html">InheritanceConcordance</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#plus(gngs.Regions)" title="Method in Regions">plus(Regions)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#position" title="Field in Pileup">position</a></span> - Field in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.Pileup.html#position" title="Field in PileupIterator.Pileup">position</a></span> - Field in <a href="gngs/PileupIterator.Pileup.html">PileupIterator.Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PrefixTrie.html" title="Class in gngs">PrefixTrie</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PrefixTrie.html#PrefixTrie()" title="Constructor in PrefixTrie">PrefixTrie()</a></span> - Constructor in <a href="gngs/PrefixTrie.html">PrefixTrie</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#previousRange(int)" title="Method in RangeIndex">previousRange(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#previousRange(java.lang.String, int)" title="Method in Regions">previousRange(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Returns the prior range that has its end closest
 to the given position.</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#print(java.io.PrintStream)" title="Method in VCF">print(PrintStream)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#printHeader()" title="Method in VCF">printHeader()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#printROC()" title="Method in VCFtoHTML">printROC()</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block">Calculate</div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFilter.html#process(gngs.SAMRecordPair, htsjdk.samtools.SAMRecord)" title="Method in PairFilter">process(SAMRecordPair, SAMRecord)</a></span> - Method in <a href="gngs/pair/PairFilter.html">PairFilter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairFormatter.html#process(gngs.ReadPair, htsjdk.samtools.SAMRecord)" title="Method in PairFormatter">process(ReadPair, SAMRecord)</a></span> - Method in <a href="gngs/pair/PairFormatter.html">PairFormatter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ReadStatsActor.html#process(java.lang.String)" title="Method in ReadStatsActor">process(String)</a></span> - Method in <a href="gngs/tools/ReadStatsActor.html">ReadStatsActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#processPartneredReads(groovy.lang.Closure)" title="Method in OrderedPairReader">processPartneredReads(Closure)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairLocator.html#processRead(htsjdk.samtools.SAMRecord)" title="Method in PairLocator">processRead(SAMRecord)</a></span> - Method in <a href="gngs/pair/PairLocator.html">PairLocator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#processVariant(gngs.Variant, java.io.Writer)" title="Method in VCFtoHTML">processVariant(Variant, Writer)</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ProgressCounter.html" title="Class in gngs">ProgressCounter</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Simple utility for displaying progress</div></dd>
</dl>

    
<a name="_Q_"><!-- --></a>
<h2 class="title">Q</h2>
<dl>

<dt><span class="strong"><a href="gngs/PileupState.html#qual" title="Field in PileupState">qual</a></span> - Field in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PrefixTrie.html#query(java.lang.String, int, int, int, int, int)" title="Method in PrefixTrie">query(String, int, int, int, int, int)</a></span> - Method in <a href="gngs/PrefixTrie.html">PrefixTrie</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#query(java.lang.String, int, int, groovy.lang.Closure)" title="Method in VCFIndex">query(String, int, int, Closure)</a></span> - Method in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#queryIdx(java.lang.String, int, int, groovy.lang.Closure)" title="Method in VCFIndex">queryIdx(String, int, int, Closure)</a></span> - Method in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block">Query the current VCF file for all variants in the specified
 region</div></dd>
<dt><span class="strong"><a href="gngs/ReadMateLookupActor.html#queryMate(htsjdk.samtools.SAMRecord)" title="Method in ReadMateLookupActor">queryMate(SAMRecord)</a></span> - Method in <a href="gngs/ReadMateLookupActor.html">ReadMateLookupActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#queryMate(htsjdk.samtools.SAMFileReader, htsjdk.samtools.SAMRecord)" title="Method in SAM">queryMate(SAMFileReader, SAMRecord)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#queryMateByDirectLookup(htsjdk.samtools.SAMRecord)" title="Method in OrderedPairReader">queryMateByDirectLookup(SAMRecord)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadMateLookupActor.html#queryMateByRegionScan(htsjdk.samtools.SAMRecord, gngs.Region)" title="Method in ReadMateLookupActor">queryMateByRegionScan(SAMRecord, Region)</a></span> - Method in <a href="gngs/ReadMateLookupActor.html">ReadMateLookupActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/ClinVar.html#queryOverlapping(gngs.Region)" title="Method in ClinVar">queryOverlapping(Region)</a></span> - Method in <a href="DefaultPackage/ClinVar.html">ClinVar</a>
</dt><dd> <div class="block">Convenience method to return CNVS overlapping the specified region</div></dd>
<dt><span class="strong"><a href="gngs/CNVDatabase.html#queryOverlapping(gngs.Region)" title="Method in CNVDatabase">queryOverlapping(Region)</a></span> - Method in <a href="gngs/CNVDatabase.html">CNVDatabase</a>
</dt><dd> <div class="block">@return list of CNVS overlapping the specified region</div></dd>
<dt><span class="strong"><a href="gngs/DecipherCNVs.html#queryOverlapping(gngs.Region)" title="Method in DecipherCNVs">queryOverlapping(Region)</a></span> - Method in <a href="gngs/DecipherCNVs.html">DecipherCNVs</a>
</dt><dd> <div class="block">Convenience method to return CNVS overlapping the specified region</div></dd>
<dt><span class="strong"><a href="gngs/DGV.html#queryOverlapping(gngs.Region)" title="Method in DGV">queryOverlapping(Region)</a></span> - Method in <a href="gngs/DGV.html">DGV</a>
</dt><dd> <div class="block">return CNVS overlapping the specified region</div></dd>
<dt><span class="strong"><a href="gngs/FASTAIndex.html#querySequenceName(java.lang.String)" title="Method in FASTAIndex">querySequenceName(String)</a></span> - Method in <a href="gngs/FASTAIndex.html">FASTAIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#queryTabix(java.lang.String, int, int, groovy.lang.Closure)" title="Method in VCFIndex">queryTabix(String, int, int, Closure)</a></span> - Method in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#queryVariantCounts(Map, gngs.Variant, gngs.Allele, java.lang.String)" title="Method in VariantDB">queryVariantCounts(Map, Variant, Allele, String)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Return a set of counts of times the given variant has been 
 observed with in</div></dd>
</dl>

    
<a name="_R_"><!-- --></a>
<h2 class="title">R</h2>
<dl>

<dt><span class="strong"><a href="DefaultPackage/Overlap.html#r1" title="Field in Overlap">r1</a></span> - Field in <a href="DefaultPackage/Overlap.html">Overlap</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#r1AlignmentStart" title="Field in CompactReadPair">r1AlignmentStart</a></span> - Field in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#r1ReferenceName" title="Field in CompactReadPair">r1ReferenceName</a></span> - Field in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Overlap.html#r2" title="Field in Overlap">r2</a></span> - Field in <a href="DefaultPackage/Overlap.html">Overlap</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#r2AlignmentStart" title="Field in CompactReadPair">r2AlignmentStart</a></span> - Field in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CompactReadPair.html#r2ReferenceName" title="Field in CompactReadPair">r2ReferenceName</a></span> - Field in <a href="gngs/CompactReadPair.html">CompactReadPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTA.html#R" title="Field in FASTA">R</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangedData.html" title="Class in gngs">RangedData</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">RangedData represents a set of genomic regions with data attached. </div></dd>
<dt><span class="strong"><a href="gngs/RangedData.html#RangedData(java.io.Reader, int, int, int)" title="Constructor in RangedData">RangedData(Reader, int, int, int)</a></span> - Constructor in <a href="gngs/RangedData.html">RangedData</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html" title="Class in gngs">RangeIndex</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Indexes a set of ranges so that the ranges crossing any point
 can be identified quickly and easily.</div></dd>
<dt><span class="strong"><a href="gngs/PileupState.html#read" title="Field in PileupState">read</a></span> - Field in <a href="gngs/PileupState.html">PileupState</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadMateLookupActor.html" title="Class in gngs">ReadMateLookupActor</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">An actor that can accept <a href='../gngs/ResolveRequest.html' title='ResolveRequest'>ResolveRequest</a> messages to asynchronously 
 resolve the missing mates of read pairs.</div></dd>
<dt><span class="strong"><a href="gngs/ReadMateLookupActor.html#ReadMateLookupActor(htsjdk.samtools.SAMFileReader, gngs.MissingMateIndex, groovy.lang.Closure)" title="Constructor in ReadMateLookupActor">ReadMateLookupActor(SAMFileReader, MissingMateIndex, Closure)</a></span> - Constructor in <a href="gngs/ReadMateLookupActor.html">ReadMateLookupActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadPair.html" title="Interface in gngs">ReadPair</a></span> - Interface in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ReadStatsActor.html" title="Class in gngs.tools">ReadStatsActor</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadWindow.html" title="Class in gngs">ReadWindow</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#red(Closure)" title="Method in PDF">red(Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#reduce(groovy.lang.Closure)" title="Method in RangeIndex">reduce(Closure)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Merge all overlapping ranges together to make simplified regions
 representing all the ranges covered by any range in this RangeInde.</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#reduce(groovy.lang.Closure)" title="Method in Regions">reduce(Closure)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Simplify all overlapping regions down to a single region</div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html" title="Class in gngs">RefGenes</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A convenience class that is set up to parse the UCSC RefGene database
 as a RangedData object. </div></dd>
<dt><span class="strong"><a href="gngs/RefGenes.html#RefGenes(java.io.Reader)" title="Constructor in RefGenes">RefGenes(Reader)</a></span> - Constructor in <a href="gngs/RefGenes.html">RefGenes</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Region.html" title="Class in gngs">Region</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Region of a genome</div></dd>
<dt><span class="strong"><a href="gngs/Region.html#Region(java.util.Map, java.lang.String, int, int)" title="Constructor in Region">Region(Map, String, int, int)</a></span> - Constructor in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ScannableRegion.html#region" title="Field in ScannableRegion">region</a></span> - Field in <a href="gngs/ScannableRegion.html">ScannableRegion</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RegionComparator.html" title="Class in gngs">RegionComparator</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RegionIterator.html" title="Class in gngs">RegionIterator</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html" title="Class in gngs">Regions</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A set of genomic regions, each indexed by chromosome, start and end position.</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#Regions(java.util.Map, Iterable<IRegion>)" title="Constructor in Regions">Regions(Map, Iterable&lt;IRegion&gt;)</a></span> - Constructor in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Create a Regions from a list of regions</div></dd>
<dt><span class="strong"><a href="gngs/RegionScan.html" title="Class in gngs">RegionScan</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#regionsEndingAt(java.lang.String, int)" title="Method in Regions">regionsEndingAt(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Return a list of ranges that end exactly at the specified position.</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#regionsStartingAt(java.lang.String, int)" title="Method in Regions">regionsStartingAt(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Relationship.html" title="Class in gngs">Relationship</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Models a relationship between two people</div></dd>
<dt><span class="strong"><a href="gngs/RelationshipType.html" title="Enum in gngs">RelationshipType</a></span> - Enum in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Models a relationship between two individuals in a pedigree</div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#remove()" title="Method in Pileup">remove()</a></span> - Method in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#remove()" title="Method in PileupIterator">remove()</a></span> - Method in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#remove(groovy.lang.Range)" title="Method in RangeIndex">remove(Range)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Remove an existing range from the index. </div></dd>
<dt><span class="strong"><a href="gngs/RegionIterator.html#remove()" title="Method in RegionIterator">remove()</a></span> - Method in <a href="gngs/RegionIterator.html">RegionIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#remove(java.lang.String, groovy.lang.Range)" title="Method in Regions">remove(String, Range)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#removeFamily(java.lang.String)" title="Method in Pedigrees">removeFamily(String)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#renameFamily(java.lang.String, java.lang.String)" title="Method in Pedigrees">renameFamily(String, String)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#renameSample(java.lang.String, java.lang.String)" title="Method in VCF">renameSample(String, String)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Change the sample ids for this VCF</div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#renameSubject(java.lang.String, java.lang.String)" title="Method in Pedigree">renameSubject(String, String)</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#renameSubject(java.lang.String, java.lang.String)" title="Method in Pedigrees">renameSubject(String, String)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#replaceSamples(List<String>)" title="Method in VCF">replaceSamples(List&lt;String&gt;)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Replace the sample ids in the VCF with the given ones</div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#reset()" title="Method in Pedigree">reset()</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#resolve(int, gngs.SAMRecordPair)" title="Method in MissingMateIndex">resolve(int, SAMRecordPair)</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#resolveAsynchronously(gngs.SAMRecordPair, int)" title="Method in OrderedPairReader">resolveAsynchronously(SAMRecordPair, int)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#resolveExportSamples(java.util.List)" title="Method in VCFtoHTML">resolveExportSamples(List)</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ReadMateLookupActor.html#resolvePair(gngs.SAMRecordPair, gngs.Region)" title="Method in ReadMateLookupActor">resolvePair(SAMRecordPair, Region)</a></span> - Method in <a href="gngs/ReadMateLookupActor.html">ReadMateLookupActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ResolveRequest.html" title="Class in gngs">ResolveRequest</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/RestrictionEnzyme.html" title="Class in RestrictionEnzyme">RestrictionEnzyme</a></span> - Class in <a href="./RestrictionEnzyme/package-summary.html">RestrictionEnzyme</a>
</dt><dd><div class="block">A simple class for matching motifs in restriction enzymes to 
 FASTA sequence. </div></dd>
<dt><span class="strong"><a href="DefaultPackage/RestrictionEnzyme.html#RestrictionEnzyme(java.lang.String, java.lang.String, int, int, boolean)" title="Constructor in RestrictionEnzyme">RestrictionEnzyme(String, String, int, int, boolean)</a></span> - Constructor in <a href="DefaultPackage/RestrictionEnzyme.html">RestrictionEnzyme</a>
</dt><dd> <div class="block">Create a restriction enzyme object with given name, motif cut point and overhang.</div></dd>
<dt><span class="strong"><a href="gngs/FASTA.html#reverseComplement(java.lang.String)" title="Method in FASTA">reverseComplement(String)</a></span> - Method in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block">Compute the reverse complement of the given sequence of bases.</div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#reverseIterator()" title="Method in RangeIndex">reverseIterator()</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#reverseIteratorAt(int)" title="Method in RangeIndex">reverseIteratorAt(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Return an iterator that will return each range that starts
 at or before the given position, proceeding backwards
 through the genome, ordered by start position.</div></dd>
<dt><span class="strong"><a href="gngs/tools/ROCPoint.html" title="Class in gngs.tools">ROCPoint</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/AdjustZygosity.html#run()" title="Method in AdjustZygosity">run()</a></span> - Method in <a href="gngs/tools/AdjustZygosity.html">AdjustZygosity</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/AsyncMateResolveWorker.html#run()" title="Method in AsyncMateResolveWorker">run()</a></span> - Method in <a href="gngs/AsyncMateResolveWorker.html">AsyncMateResolveWorker</a>
</dt><dd> <div class="block">Main loop - runs forever</div></dd>
<dt><span class="strong"><a href="gngs/tools/BAM2Ped.html#run()" title="Method in BAM2Ped">run()</a></span> - Method in <a href="gngs/tools/BAM2Ped.html">BAM2Ped</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/DenovoRate.html#run()" title="Method in DenovoRate">run()</a></span> - Method in <a href="DefaultPackage/DenovoRate.html">DenovoRate</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DepthOfCoverage.html#run()" title="Method in DepthOfCoverage">run()</a></span> - Method in <a href="gngs/tools/DepthOfCoverage.html">DepthOfCoverage</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DownsampleFastq.html#run(java.lang.String, java.lang.String, java.lang.String, java.lang.String, double, long)" title="Method in DownsampleFastq">run(String, String, String, String, double, long)</a></span> - Method in <a href="gngs/tools/DownsampleFastq.html">DownsampleFastq</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ExtractFASTQ.html#run(java.io.Writer)" title="Method in ExtractFASTQ">run(Writer)</a></span> - Method in <a href="gngs/tools/ExtractFASTQ.html">ExtractFASTQ</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#run()" title="Method in Gaps">run()</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Pairs.html#run()" title="Method in Pairs">run()</a></span> - Method in <a href="gngs/tools/Pairs.html">Pairs</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Sex.html#run()" title="Method in Sex">run()</a></span> - Method in <a href="gngs/tools/Sex.html">Sex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SexKaryotyper.html#run()" title="Method in SexKaryotyper">run()</a></span> - Method in <a href="DefaultPackage/SexKaryotyper.html">SexKaryotyper</a>
</dt><dd> <div class="block">The primary logic is simply that lack of coverage of the Y
 chromosome indicates a female sample. </div></dd>
<dt><span class="strong"><a href="gngs/tools/SplitCommonRecords.html#run()" title="Method in SplitCommonRecords">run()</a></span> - Method in <a href="gngs/tools/SplitCommonRecords.html">SplitCommonRecords</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ToolBase.html#run()" title="Method in ToolBase">run()</a></span> - Method in <a href="gngs/ToolBase.html">ToolBase</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#run()" title="Method in VCFtoHTML">run()</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#run(boolean)" title="Method in VCFtoPedigree">run(boolean)</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VEPDiff.html#run(PrintWriter, def)" title="Method in VEPDiff">run(PrintWriter, def)</a></span> - Method in <a href="DefaultPackage/VEPDiff.html">VEPDiff</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_S_"><!-- --></a>
<h2 class="title">S</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#S" title="Field in FASTA">S</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html" title="Class in gngs">SAM</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Adds various Groovy idioms and convenience features to the 
 Picard SAMFileReader.</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#SAM(java.io.File)" title="Constructor in SAM">SAM(File)</a></span> - Constructor in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#sampleDosage(java.lang.String, int)" title="Method in Variant">sampleDosage(String, int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#sampleGenoType(java.lang.String)" title="Method in Variant">sampleGenoType(String)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#sampleIndex(java.lang.String)" title="Method in VCF">sampleIndex(String)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SamplesToPed.html" title="Class in SamplesToPed">SamplesToPed</a></span> - Class in <a href="./SamplesToPed/package-summary.html">SamplesToPed</a>
</dt><dd><div class="block">Helper utility that walks the user through turning a sample info source / file
 into a PED file with family relationships.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/SamplesToPed.html#SamplesToPed()" title="Constructor in SamplesToPed">SamplesToPed()</a></span> - Constructor in <a href="DefaultPackage/SamplesToPed.html">SamplesToPed</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordCategory.html" title="Class in gngs">SAMRecordCategory</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html" title="Class in gngs">SAMRecordPair</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Models a read pair, where one of the reads might be missing.</div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#save(java.io.Writer)" title="Method in Pedigrees">save(Writer)</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#save(java.util.Map, java.lang.String)" title="Method in Regions">save(Map, String)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairScanner.html#scan(gngs.SAM)" title="Method in PairScanner">scan(SAM)</a></span> - Method in <a href="gngs/pair/PairScanner.html">PairScanner</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ScannableRegion.html" title="Class in gngs">ScannableRegion</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#scheduleAsyncResolves()" title="Method in OrderedPairReader">scheduleAsyncResolves()</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="db/Schema.html" title="Class in db">Schema</a></span> - Class in <a href="./db/package-summary.html">db</a>
</dt><dd><div class="block">The default schema, which uses SQLite syntax</div></dd>
<dt><span class="strong"><a href="gngs/BaseCounts.html#score()" title="Method in BaseCounts">score()</a></span> - Method in <a href="gngs/BaseCounts.html">BaseCounts</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Overlap.html#score" title="Field in Overlap">score</a></span> - Field in <a href="DefaultPackage/Overlap.html">Overlap</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Assembler.html#scoreOverlap(java.lang.String, java.lang.String)" title="Method in Assembler">scoreOverlap(String, String)</a></span> - Method in <a href="DefaultPackage/Assembler.html">Assembler</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#searchForScannableRegion(gngs.SAMRecordPair, int, int)" title="Method in MissingMateIndex">searchForScannableRegion(SAMRecordPair, int, int)</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block">Search for a region that is guaranteed to contain the missing read from the specified pair,
 while also containing at least <code>minimumReads</code> reads that also need resolution.</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#segregatesWith(gngs.Pedigree)" title="Method in Variant">segregatesWith(Pedigree)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return a Map of fields from the genotype column corresponding
 to the given sample. </div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#selectRegions()" title="Method in VCFtoPedigree">selectRegions()</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/GRange.html#serialVersionUID" title="Field in GRange">serialVersionUID</a></span> - Field in <a href="gngs/GRange.html">GRange</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Region.html#serialVersionUID" title="Field in Region">serialVersionUID</a></span> - Field in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#setAlt(java.lang.String)" title="Method in Variant">setAlt(String)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Update the first alternate allele to the given value</div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#setMinMappingQuality(int)" title="Method in PileupIterator">setMinMappingQuality(int)</a></span> - Method in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#setR2(htsjdk.samtools.SAMRecord)" title="Method in SAMRecordPair">setR2(SAMRecord)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#setReadGroup(java.lang.String)" title="Method in SAMRecordPair">setReadGroup(String)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block">Do not static compile this as it relies on access to protected function</div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#setSamples(List<String>)" title="Method in Pedigree">setSamples(List&lt;String&gt;)</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#setTag(java.lang.String, java.lang.String)" title="Method in SAMRecordPair">setTag(String, String)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VEPConsequences.html#severityOf(java.lang.String)" title="Method in VEPConsequences">severityOf(String)</a></span> - Method in <a href="gngs/VEPConsequences.html">VEPConsequences</a>
</dt><dd> <div class="block">Returns an integer representing the severity of the consequence of a mutation. 
 </div></dd>
<dt><span class="strong"><a href="gngs/Sex.html" title="Enum in gngs">Sex</a></span> - Enum in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/SexKaryotyper.html" title="Class in SexKaryotyper">SexKaryotyper</a></span> - Class in <a href="./SexKaryotyper/package-summary.html">SexKaryotyper</a>
</dt><dd><div class="block">An algorithm based on simple heuristics for estimating the sex of a sample from
 sequencing coverage.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/SexKaryotyper.html#SexKaryotyper(gngs.SAM, gngs.Regions)" title="Constructor in SexKaryotyper">SexKaryotyper(SAM, Regions)</a></span> - Constructor in <a href="DefaultPackage/SexKaryotyper.html">SexKaryotyper</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Hash.html#sha1(java.lang.String)" title="Method in Hash">sha1(String)</a></span> - Method in <a href="gngs/Hash.html">Hash</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairActor.html#shutdown()" title="Method in OrderedPairActor">shutdown()</a></span> - Method in <a href="gngs/OrderedPairActor.html">OrderedPairActor</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SimpleLogFormatter.html" title="Class in gngs">SimpleLogFormatter</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">The default Java log former uses a format that is too verbose, so
 replace it with something more compact.</div></dd>
<dt><span class="strong"><a href="gngs/Allele.html#size(java.lang.String)" title="Method in Allele">size(String)</a></span> - Method in <a href="gngs/Allele.html">Allele</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/CoverageBlock.html#size()" title="Method in CoverageBlock">size()</a></span> - Method in <a href="gngs/CoverageBlock.html">CoverageBlock</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTQRead.html#size()" title="Method in FASTQRead">size()</a></span> - Method in <a href="gngs/FASTQRead.html">FASTQRead</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/MissingMateIndex.html#size()" title="Method in MissingMateIndex">size()</a></span> - Method in <a href="gngs/MissingMateIndex.html">MissingMateIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Region.html#size()" title="Method in Region">size()</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block">@return  the number of positions spanned by this region</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#size()" title="Method in Regions">size()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#size()" title="Method in SAM">size()</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Count the total number of reads in the SAM file</div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#size()" title="Method in Variant">size()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Returns the change in size of the genome caused by this variant's default allele.</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#sniffGenomeBuild()" title="Method in SAM">sniffGenomeBuild()</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Probe the given BAM file to make a guess about what genome build it is 
 generated from.</div></dd>
<dt><span class="strong"><a href="gngs/SnpEffInfo.html" title="Class in gngs">SnpEffInfo</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Support for parsing annotations from SnpEff</div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#spanOverlaps(gngs.Region)" title="Method in SAMRecordPair">spanOverlaps(Region)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/GRange.html#spans(groovy.lang.IntRange)" title="Method in GRange">spans(IntRange)</a></span> - Method in <a href="gngs/GRange.html">GRange</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Region.html#spans(gngs.IRegion)" title="Method in Region">spans(IRegion)</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block">@return  true iff this region fully encompasses the given region</div></dd>
<dt><span class="strong"><a href="gngs/tools/SplitCommonRecords.html" title="Class in gngs.tools">SplitCommonRecords</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block">Splits records at the same position in a VCF file to 
 separate VCF files. </div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#spoolStats()" title="Method in OrderedPairReader">spoolStats()</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pileup.html#start" title="Field in Pileup">start</a></span> - Field in <a href="gngs/Pileup.html">Pileup</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PileupIterator.html#start" title="Field in PileupIterator">start</a></span> - Field in <a href="gngs/PileupIterator.html">PileupIterator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#startingAt(int)" title="Method in RangeIndex">startingAt(int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#startingAt(java.lang.String, int)" title="Method in Regions">startingAt(String, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/PrefixTrie.html#startsWith(java.lang.String, int, int, int, int)" title="Method in PrefixTrie">startsWith(String, int, int, int, int)</a></span> - Method in <a href="gngs/PrefixTrie.html">PrefixTrie</a>
</dt><dd> <div class="block">Return all values that match the given key, allowing for the specified number
 of mismatches, insertions, deletions and leading optional bases.</div></dd>
<dt><span class="strong"><a href="trie/TrieNode.html#startsWith(java.lang.String, trie.TrieQuery, int)" title="Method in TrieNode">startsWith(String, TrieQuery, int)</a></span> - Method in <a href="trie/TrieNode.html">TrieNode</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/BEDToolsCoverage.html#stats(gngs.Region)" title="Method in BEDToolsCoverage">stats(Region)</a></span> - Method in <a href="DefaultPackage/BEDToolsCoverage.html">BEDToolsCoverage</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairScanner.html#stopActor(java.lang.String, groovyx.gpars.actor.Actor)" title="Method in PairScanner">stopActor(String, Actor)</a></span> - Method in <a href="gngs/pair/PairScanner.html">PairScanner</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/StopParsingVCFException.html" title="Class in gngs">StopParsingVCFException</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#stream(groovy.lang.Closure)" title="Method in SAM">stream(Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Subject.html" title="Class in gngs">Subject</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">An individual in a family pedigree</div></dd>
<dt><span class="strong"><a href="gngs/Subject.html#Subject(gngs.Subject)" title="Constructor in Subject">Subject(Subject)</a></span> - Constructor in <a href="gngs/Subject.html">Subject</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#subtract(gngs.Regions)" title="Method in Regions">subtract(Regions)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/RangeIndex.html#subtractFrom(int, int)" title="Method in RangeIndex">subtractFrom(int, int)</a></span> - Method in <a href="gngs/RangeIndex.html">RangeIndex</a>
</dt><dd> <div class="block">Subtract all the ranges in this range index from the given range and return
 the resulting list of ranges.</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#subtractFrom(java.lang.String, int, int)" title="Method in Regions">subtractFrom(String, int, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Remove all of the regions belonging to this <a href='../gngs/Regions.html' title='Regions'>Regions</a> object
 from the interval specified, and return the list of Ranges that results.</div></dd>
<dt><span class="strong"><a href="gngs/tools/VInfo.html#summarize(gngs.Variant)" title="Method in VInfo">summarize(Variant)</a></span> - Method in <a href="gngs/tools/VInfo.html">VInfo</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_T_"><!-- --></a>
<h2 class="title">T</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#T" title="Field in FASTA">T</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Table.html" title="Class in gngs.tools">Table</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#table(Closure)" title="Method in PDF">table(Closure)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#table(java.util.Map, List<String>, List<List>)" title="Method in Utils">table(Map, List&lt;String&gt;, List&lt;List&gt;)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block">A utility to print a table of values in a nice format for 
 output on a terminal. </div></dd>
<dt><span class="strong"><a href="DefaultPackage/TargetedCNVAnnotator.html" title="Class in TargetedCNVAnnotator">TargetedCNVAnnotator</a></span> - Class in <a href="./TargetedCNVAnnotator/package-summary.html">TargetedCNVAnnotator</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TargetedCNVAnnotator.html#TargetedCNVAnnotator(gngs.Regions, java.lang.String)" title="Constructor in TargetedCNVAnnotator">TargetedCNVAnnotator(Regions, String)</a></span> - Constructor in <a href="DefaultPackage/TargetedCNVAnnotator.html">TargetedCNVAnnotator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Constraint.html#test(gngs.Variant, List<Pedigree>)" title="Method in Constraint">test(Variant, List&lt;Pedigree&gt;)</a></span> - Method in <a href="gngs/Constraint.html">Constraint</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TestTrio.html" title="Class in TestTrio">TestTrio</a></span> - Class in <a href="./TestTrio/package-summary.html">TestTrio</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TestTrio.html#TestTrio(List<VCF>, gngs.VCF)" title="Constructor in TestTrio">TestTrio(List&lt;VCF&gt;, VCF)</a></span> - Constructor in <a href="DefaultPackage/TestTrio.html">TestTrio</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#testTrios()" title="Method in VCFtoPedigree">testTrios()</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TestUpdate.html" title="Class in TestUpdate">TestUpdate</a></span> - Class in <a href="./TestUpdate/package-summary.html">TestUpdate</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Time.html" title="Class in Time">Time</a></span> - Class in <a href="./Time/package-summary.html">Time</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Time.html#Time()" title="Constructor in Time">Time()</a></span> - Constructor in <a href="DefaultPackage/Time.html">Time</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#time(java.lang.String, groovy.lang.Closure)" title="Method in Utils">time(String, Closure)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Utils.html#timeMs(groovy.lang.Closure)" title="Method in Utils">timeMs(Closure)</a></span> - Method in <a href="gngs/Utils.html">Utils</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#title(java.lang.Object)" title="Method in PDF">title(Object)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#toAlignment(java.lang.String)" title="Method in PDF">toAlignment(String)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#toAnnovar(int)" title="Method in Variant">toAnnovar(int)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#toBED()" title="Method in VCF">toBED()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/PDF.html#toColor(java.lang.String)" title="Method in PDF">toColor(String)</a></span> - Method in <a href="DefaultPackage/PDF.html">PDF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Plot.html#toDataTable(java.lang.Iterable)" title="Method in Plot">toDataTable(Iterable)</a></span> - Method in <a href="DefaultPackage/Plot.html">Plot</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#toJson()" title="Method in Pedigree">toJson()</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#toJson()" title="Method in Pedigrees">toJson()</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Subject.html#toJson()" title="Method in Subject">toJson()</a></span> - Method in <a href="gngs/Subject.html">Subject</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#toJson(java.lang.String)" title="Method in Variant">toJson(String)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Return a JSON string representing the key details about this variant.
 </div></dd>
<dt><span class="strong"><a href="gngs/RangedData.html#toListMap()" title="Method in RangedData">toListMap()</a></span> - Method in <a href="gngs/RangedData.html">RangedData</a>
</dt><dd> <div class="block">Convert to a list of map objects</div></dd>
<dt><span class="strong"><a href="gngs/ToolBase.html" title="Class in gngs">ToolBase</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#toPairRegions(java.lang.String, int, int, int)" title="Method in SAM">toPairRegions(String, int, int, int)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Return read pairs from this SAM file that overlap the specified region
 as a Regions object - that is, as a set of genomic intervals. </div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#toPed(java.io.Writer)" title="Method in Pedigree">toPed(Writer)</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TestTrio.html#toPedigree()" title="Method in TestTrio">toPedigree()</a></span> - Method in <a href="DefaultPackage/TestTrio.html">TestTrio</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/RestrictionEnzyme.html#toRegex(java.lang.String)" title="Method in RestrictionEnzyme">toRegex(String)</a></span> - Method in <a href="DefaultPackage/RestrictionEnzyme.html">RestrictionEnzyme</a>
</dt><dd> <div class="block">Convert N's to wildcards and select IUPAC codes to appropriate character sets</div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordCategory.html#toRegion(htsjdk.samtools.BAMRecord)" title="Method in SAMRecordCategory">toRegion(BAMRecord)</a></span> - Method in <a href="gngs/SAMRecordCategory.html">SAMRecordCategory</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#toRegions(java.lang.String, int, int)" title="Method in SAM">toRegions(String, int, int)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Return reads from this SAM file that overlap the specified region
 as a Regions object - that is, as a set of genomic intervals.</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#toRegions()" title="Method in VCF">toRegions()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Allele.html#toString()" title="Method in Allele">toString()</a></span> - Method in <a href="gngs/Allele.html">Allele</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BaseCounts.html#toString()" title="Method in BaseCounts">toString()</a></span> - Method in <a href="gngs/BaseCounts.html">BaseCounts</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/BED.html#toString()" title="Method in BED">toString()</a></span> - Method in <a href="gngs/BED.html">BED</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/ConcordanceScore.html#toString()" title="Method in ConcordanceScore">toString()</a></span> - Method in <a href="gngs/tools/ConcordanceScore.html">ConcordanceScore</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/CpipeRunSet.html#toString()" title="Method in CpipeRunSet">toString()</a></span> - Method in <a href="DefaultPackage/CpipeRunSet.html">CpipeRunSet</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/InheritanceConcordance.html#toString()" title="Method in InheritanceConcordance">toString()</a></span> - Method in <a href="gngs/tools/InheritanceConcordance.html">InheritanceConcordance</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/pair/PairLocator.html#toString()" title="Method in PairLocator">toString()</a></span> - Method in <a href="gngs/pair/PairLocator.html">PairLocator</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#toString()" title="Method in Pedigree">toString()</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Region.html#toString()" title="Method in Region">toString()</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#toString()" title="Method in Regions">toString()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Relationship.html#toString()" title="Method in Relationship">toString()</a></span> - Method in <a href="gngs/Relationship.html">Relationship</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/RestrictionEnzyme.html#toString()" title="Method in RestrictionEnzyme">toString()</a></span> - Method in <a href="DefaultPackage/RestrictionEnzyme.html">RestrictionEnzyme</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#toString()" title="Method in SAMRecordPair">toString()</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SnpEffInfo.html#toString()" title="Method in SnpEffInfo">toString()</a></span> - Method in <a href="gngs/SnpEffInfo.html">SnpEffInfo</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Subject.html#toString()" title="Method in Subject">toString()</a></span> - Method in <a href="gngs/Subject.html">Subject</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TestTrio.html#toString()" title="Method in TestTrio">toString()</a></span> - Method in <a href="DefaultPackage/TestTrio.html">TestTrio</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#toString()" title="Method in Variant">toString()</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#toString()" title="Method in VCF">toString()</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCFSummaryStats.html#toString()" title="Method in VCFSummaryStats">toString()</a></span> - Method in <a href="gngs/VCFSummaryStats.html">VCFSummaryStats</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/XA.html#toString()" title="Method in XA">toString()</a></span> - Method in <a href="gngs/XA.html">XA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#transmissionRate(gngs.VCF, gngs.VCF)" title="Method in VCF">transmissionRate(VCF, VCF)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Calculate the rate at which variants are transmitted to this sample from p1,
 given p2 is the other parent.</div></dd>
<dt><span class="strong"><a href="gngs/RelationshipType.html#transmits(java.lang.String)" title="Method in RelationshipType">transmits(String)</a></span> - Method in <a href="gngs/RelationshipType.html">RelationshipType</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieMismatchCosts.html" title="Class in trie">TrieMismatchCosts</a></span> - Class in <a href="./trie/package-summary.html">trie</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieNode.html" title="Class in trie">TrieNode</a></span> - Class in <a href="./trie/package-summary.html">trie</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieNode.html#TrieNode()" title="Constructor in TrieNode">TrieNode()</a></span> - Constructor in <a href="trie/TrieNode.html">TrieNode</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="trie/TrieQuery.html" title="Class in trie">TrieQuery</a></span> - Class in <a href="./trie/package-summary.html">trie</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTQRead.html#trimEnd(int, int)" title="Method in FASTQRead">trimEnd(int, int)</a></span> - Method in <a href="gngs/FASTQRead.html">FASTQRead</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#trimSampleId(java.lang.String)" title="Method in VariantDB">trimSampleId(String)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/Trio.html" title="Class in Trio">Trio</a></span> - Class in <a href="./Trio/package-summary.html">Trio</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/TSV.html" title="Class in TSV">TSV</a></span> - Class in <a href="./TSV/package-summary.html">TSV</a>
</dt><dd><div class="block">This class is just an alias to bring the Graxxia TSV class
 into the default namespace.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/TSV.html#TSV(java.lang.String)" title="Constructor in TSV">TSV(String)</a></span> - Constructor in <a href="DefaultPackage/TSV.html">TSV</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#tx(Closure)" title="Method in VariantDB">tx(Closure)</a></span> - Method in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Execute the given closure in teh scope of a transaction, and roll it back if
 an exception occurs.</div></dd>
</dl>

    
<a name="_U_"><!-- --></a>
<h2 class="title">U</h2>
<dl>

<dt><span class="strong"><a href="gngs/Regions.html#unique()" title="Method in Regions">unique()</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Return a new Regions that contains the same regions as this one,
 but ensuring each identical region is only present once.</div></dd>
<dt><span class="strong"><a href="DefaultPackage/IlluminaFileNameParser.html#UNKNOWN" title="Field in IlluminaFileNameParser">UNKNOWN</a></span> - Field in <a href="DefaultPackage/IlluminaFileNameParser.html">IlluminaFileNameParser</a>
</dt><dd> <div class="block">Value assigned to fields that could not be determined</div></dd>
<dt><span class="strong"><a href="gngs/PairReaderStats.html#update(htsjdk.samtools.SAMRecord)" title="Method in PairReaderStats">update(SAMRecord)</a></span> - Method in <a href="gngs/PairReaderStats.html">PairReaderStats</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html#update(java.lang.String, groovy.lang.Closure)" title="Method in Variant">update(String, Closure)</a></span> - Method in <a href="gngs/Variant.html">Variant</a>
</dt><dd> <div class="block">Allows various fields to be updated and then synchronises the 
 rest of the data with those updated fields</div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#updateROC(gngs.Variant)" title="Method in VCFtoHTML">updateROC(Variant)</a></span> - Method in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/OrderedPairReader.html#updateStatsBlock(htsjdk.samtools.SAMRecord)" title="Method in OrderedPairReader">updateStatsBlock(SAMRecord)</a></span> - Method in <a href="gngs/OrderedPairReader.html">OrderedPairReader</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="db/Schema.html#upgradeLegacy(groovy.sql.Sql)" title="Method in Schema">upgradeLegacy(Sql)</a></span> - Method in <a href="db/Schema.html">Schema</a>
</dt><dd> <div class="block">This function upgrades a "legacy" database that was created as part of the
 MGHA project to the schema used by this class. </div></dd>
<dt><span class="strong"><a href="db/Schema.html#upgradeSchema(groovy.sql.Sql, int)" title="Method in Schema">upgradeSchema(Sql, int)</a></span> - Method in <a href="db/Schema.html">Schema</a>
</dt><dd> <div class="block">Execute the given statements to upgrade the schema</div></dd>
<dt><span class="strong"><a href="gngs/Utils.html" title="Class in gngs">Utils</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Miscellaneous utilities that I couldn't think to put anywhere else</div></dd>
</dl>

    
<a name="_V_"><!-- --></a>
<h2 class="title">V</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#V" title="Field in FASTA">V</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Pedigree.html#validate()" title="Method in Pedigree">validate()</a></span> - Method in <a href="gngs/Pedigree.html">Pedigree</a>
</dt><dd> <div class="block">Verify that the samples in the pedigree are internally consistent
 ie: mother must be female, father male, sample cannot be mother of itself, etc.</div></dd>
<dt><span class="strong"><a href="gngs/Pedigrees.html#validate()" title="Method in Pedigrees">validate()</a></span> - Method in <a href="gngs/Pedigrees.html">Pedigrees</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#validVariant(gngs.Variant)" title="Method in VCFtoPedigree">validVariant(Variant)</a></span> - Method in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Variant.html" title="Class in gngs">Variant</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Represents the genetic state at a specific locus in a genome
 ie: captures the information at a single line of a VCF file.
 </div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html" title="Class in VariantDB">VariantDB</a></span> - Class in <a href="./VariantDB/package-summary.html">VariantDB</a>
</dt><dd><div class="block">VariantDB implements a simple, embedded, variant tracking database that
 can be used to track variants identified in multiple samples and 
 sequencing runs. </div></dd>
<dt><span class="strong"><a href="DefaultPackage/VariantDB.html#VariantDB(java.lang.String)" title="Constructor in VariantDB">VariantDB(String)</a></span> - Constructor in <a href="DefaultPackage/VariantDB.html">VariantDB</a>
</dt><dd> <div class="block">Open or create a VariantDB using default settings and the given
 file name. </div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#variantsAt(java.lang.String, int)" title="Method in VCF">variantsAt(String, int)</a></span> - Method in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block">Return a list of variants starting at the given location</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html" title="Class in gngs">VCF</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">The VCF class supports simple parsing, filtering, querying and updating of
 VCF files.</div></dd>
<dt><span class="strong"><a href="gngs/VCF.html#VCF(Iterable<Variant>)" title="Constructor in VCF">VCF(Iterable&lt;Variant&gt;)</a></span> - Constructor in <a href="gngs/VCF.html">VCF</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html" title="Class in gngs">VCFIndex</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">Support for querying indexed VCF files via random access.</div></dd>
<dt><span class="strong"><a href="gngs/VCFIndex.html#VCFIndex(java.lang.String)" title="Constructor in VCFIndex">VCFIndex(String)</a></span> - Constructor in <a href="gngs/VCFIndex.html">VCFIndex</a>
</dt><dd> <div class="block">Load and index for the given VCF file</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFSimilarity.html" title="Class in VCFSimilarity">VCFSimilarity</a></span> - Class in <a href="./VCFSimilarity/package-summary.html">VCFSimilarity</a>
</dt><dd><div class="block">A simple measurement of similarity of different samples by counting the number of 
 shared vs different SNVs. </div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFSimilarity.html#VCFSimilarity(java.lang.String)" title="Constructor in VCFSimilarity">VCFSimilarity(String)</a></span> - Constructor in <a href="DefaultPackage/VCFSimilarity.html">VCFSimilarity</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VCFSummaryStats.html" title="Class in gngs">VCFSummaryStats</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A set of statistics summarizing how a VCF was filtered</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoDB.html" title="Script in VCFtoDB">VCFtoDB</a></span> - Script in <a href="./VCFtoDB/package-summary.html">VCFtoDB</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html" title="Class in gngs.tools">VCFtoHTML</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VCFtoHTML.html#VCFtoHTML(groovy.util.OptionAccessor)" title="Constructor in VCFtoHTML">VCFtoHTML(OptionAccessor)</a></span> - Constructor in <a href="gngs/tools/VCFtoHTML.html">VCFtoHTML</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html" title="Class in VCFtoPedigree">VCFtoPedigree</a></span> - Class in <a href="./VCFtoPedigree/package-summary.html">VCFtoPedigree</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VCFtoPedigree.html#VCFtoPedigree(List<String>)" title="Constructor in VCFtoPedigree">VCFtoPedigree(List&lt;String&gt;)</a></span> - Constructor in <a href="DefaultPackage/VCFtoPedigree.html">VCFtoPedigree</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/VEPConsequences.html" title="Class in gngs">VEPConsequences</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="DefaultPackage/VEPDiff.html" title="Class in VEPDiff">VEPDiff</a></span> - Class in <a href="./VEPDiff/package-summary.html">VEPDiff</a>
</dt><dd><div class="block">Finds differences in VEP annotations for identical variants in two VCFs</div></dd>
<dt><span class="strong"><a href="DefaultPackage/VEPDiff.html#VEPDiff(java.lang.String, java.lang.String)" title="Constructor in VEPDiff">VEPDiff(String, String)</a></span> - Constructor in <a href="DefaultPackage/VEPDiff.html">VEPDiff</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/VInfo.html" title="Class in gngs.tools">VInfo</a></span> - Class in <a href="./gngs/tools/package-summary.html">gngs.tools</a>
</dt><dd><div class="block"></div></dd>
</dl>

    
<a name="_W_"><!-- --></a>
<h2 class="title">W</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#W" title="Field in FASTA">W</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/Region.html#widen(int)" title="Method in Region">widen(int)</a></span> - Method in <a href="gngs/Region.html">Region</a>
</dt><dd> <div class="block">@param bp    number of base pairs to widen by</div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#widen(int)" title="Method in Regions">widen(int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Return a new Regions object that has <code>bp</code> bases
 added to the beginning and end of each interval. </div></dd>
<dt><span class="strong"><a href="gngs/Regions.html#window(gngs.Region, int)" title="Method in Regions">window(Region, int)</a></span> - Method in <a href="gngs/Regions.html">Regions</a>
</dt><dd> <div class="block">Return a window of n regions upstream and downstream of the 
 given region</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#withIterator(groovy.lang.Closure)" title="Method in SAM">withIterator(Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#withOrderedPairActor(java.util.Map, java.lang.String, groovy.lang.Closure)" title="Method in SAM">withOrderedPairActor(Map, String, Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block">Execute the given closure with an actor set to write ordered pairs to the given
 output file, based on this BAM file.</div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#withOrderedPairWriter(java.lang.String, boolean, groovy.lang.Closure)" title="Method in SAM">withOrderedPairWriter(String, boolean, Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/ProgressCounter.html#withProgress(java.util.Map, groovy.lang.Closure)" title="Method in ProgressCounter">withProgress(Map, Closure)</a></span> - Method in <a href="gngs/ProgressCounter.html">ProgressCounter</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#withReader(groovy.lang.Closure)" title="Method in SAM">withReader(Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAM.html#withWriter(java.lang.String, boolean, groovy.lang.Closure)" title="Method in SAM">withWriter(String, boolean, Closure)</a></span> - Method in <a href="gngs/SAM.html">SAM</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/FASTQRead.html#write(java.io.Writer)" title="Method in FASTQRead">write(Writer)</a></span> - Method in <a href="gngs/FASTQRead.html">FASTQRead</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/SAMRecordPair.html#write(java.io.Writer)" title="Method in SAMRecordPair">write(Writer)</a></span> - Method in <a href="gngs/SAMRecordPair.html">SAMRecordPair</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Table.html#writeCSV(List<Map>)" title="Method in Table">writeCSV(List&lt;Map&gt;)</a></span> - Method in <a href="gngs/tools/Table.html">Table</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Table.html#writeData(List<Map>, java.lang.String)" title="Method in Table">writeData(List&lt;Map&gt;, String)</a></span> - Method in <a href="gngs/tools/Table.html">Table</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#writeDiffs(gngs.Regions, gngs.Regions)" title="Method in Gaps">writeDiffs(Regions, Regions)</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#writeGaps(gngs.Regions)" title="Method in Gaps">writeGaps(Regions)</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/DepthOfCoverage.html#writeOutputs()" title="Method in DepthOfCoverage">writeOutputs()</a></span> - Method in <a href="gngs/tools/DepthOfCoverage.html">DepthOfCoverage</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Gaps.html#writeTable(gngs.Regions)" title="Method in Gaps">writeTable(Regions)</a></span> - Method in <a href="gngs/tools/Gaps.html">Gaps</a>
</dt><dd> <div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/tools/Table.html#writeTSV(List<Map>)" title="Method in Table">writeTSV(List&lt;Map&gt;)</a></span> - Method in <a href="gngs/tools/Table.html">Table</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_X_"><!-- --></a>
<h2 class="title">X</h2>
<dl>

<dt><span class="strong"><a href="gngs/XA.html" title="Class in gngs">XA</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block"></div></dd>
<dt><span class="strong"><a href="gngs/XPos.html" title="Class in gngs">XPos</a></span> - Class in <a href="./gngs/package-summary.html">gngs</a>
</dt><dd><div class="block">A representation of a genomic position as a long</div></dd>
</dl>

    
<a name="_Y_"><!-- --></a>
<h2 class="title">Y</h2>
<dl>

<dt><span class="strong"><a href="gngs/FASTA.html#Y" title="Field in FASTA">Y</a></span> - Field in <a href="gngs/FASTA.html">FASTA</a>
</dt><dd> <div class="block"></div></dd>
</dl>

    
<a name="_Z_"><!-- --></a>
<h2 class="title">Z</h2>
<dl>

</dl>

    
<a name="___"><!-- --></a>
<h2 class="title">_</h2>
<dl>

</dl>

    
</div>


<a href="#_A_">A</a>

<a href="#_B_">B</a>

<a href="#_C_">C</a>

<a href="#_D_">D</a>

<a href="#_E_">E</a>

<a href="#_F_">F</a>

<a href="#_G_">G</a>

<a href="#_H_">H</a>

<a href="#_I_">I</a>

<a href="#_J_">J</a>

<a href="#_K_">K</a>

<a href="#_L_">L</a>

<a href="#_M_">M</a>

<a href="#_N_">N</a>

<a href="#_O_">O</a>

<a href="#_P_">P</a>

<a href="#_Q_">Q</a>

<a href="#_R_">R</a>

<a href="#_S_">S</a>

<a href="#_T_">T</a>

<a href="#_U_">U</a>

<a href="#_V_">V</a>

<a href="#_W_">W</a>

<a href="#_X_">X</a>

<a href="#_Y_">Y</a>

<a href="#_Z_">Z</a>

<a href="#___">_</a>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
    <!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="overview-summary.html">Overview</a></li>
        <li>Package</li>
        <li>Class</li>
        <li><a href="deprecated-list.html">Deprecated</a></li>
        <li class="navBarCell1Rev">Index</li>
        <li><a href="help-doc.html">Help</a></li>
    </ul>
</div>
<div class="subNav">
    <ul class="navList">
        <li><a href="./index.html?index-all.html" target="_top">Frames</a></li>
        <li><a href="index-all.html" target="_top">No Frames</a></li>
    </ul>
    <div class="aboutLanguage"><em>Groovy Documentation</em></div>
    <a name="skip-navbar_bottom">
        <!--   -->
    </a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
