<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- **************************************************************** -->
<!-- *  PLEASE KEEP COMPLICATED EXPRESSIONS OUT OF THESE TEMPLATES, * -->
<!-- *  i.e. only iterate & print data where possible. Thanks, Jez. * -->
<!-- **************************************************************** -->

<html>
<head>
<!-- Generated by groovydoc (2.1.0) on Tue Jul 29 16:04:20 EST 2014 -->
<title>SampleInfo (groovy-ngs-utils API)</title>
<meta name="date" content="2014-07-29">
<meta http-equiv="Content-Type" content="text/html; charset=MacRoman">
<link href="../groovy.ico" type="image/x-icon" rel="shortcut icon">
<link href="../groovy.ico" type="image/x-icon" rel="icon">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<script type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SampleInfo (groovy-ngs-utils API)";
    }
}
</script>
<noscript>
</noscript>

</head><body onload="windowTitle();" bgcolor="white">
<hr>

<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>

  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <!--<TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  -->
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
<b>Groovy Documentation</b>
</EM></TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><!--<FONT SIZE="-2">
&nbsp;<A HREF="../groovy/lang/ExpandoMetaClass.ExpandoMetaProperty.html" title="class in groovy.lang"><B>PREV CLASS</B></A>&nbsp;

&nbsp;<A HREF="../groovy/lang/GroovyClassLoader.html" title="class in groovy.lang"><B>NEXT CLASS</B></A></FONT>--></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?DefaultPackage/SampleInfo.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="SampleInfo.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<script type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-frame.html"><B>All Classes</B></A>');
  }
  //-->
</script>
<noscript>
  <A HREF="../allclasses-frame.html"><B>All Classes</B></A>
</noscript>

</FONT></TD>

</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#property_summary">PROPERTY</A>&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#prop_detail">PROPERTY</A>&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<!-- ========= END OF TOP NAVBAR ========= -->


<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>

<span CLASS="ClassTitleFont">[Groovy] Class SampleInfo</span></H2>
<pre>java.lang.Object
  <img src='../inherit.gif'>SampleInfo
</pre><hr>
<PRE>class SampleInfo

</PRE>

<P>
 Meta data about a sample.
 <p>
 Designed to be compatible with the MGHA sample information format.
 
</P>
<hr>


<!-- =========== NESTED CLASS SUMMARY =========== -->

<A NAME="nested_summary"><!-- --></A>


<!-- =========== ENUM CONSTANT SUMMARY =========== -->

<A NAME="enum_constant_summary"><!-- --></A>


<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>


<!-- =========== PROPERTY SUMMARY =========== -->

<A NAME="property_summary"><!-- --></A>

    <TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
    <TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
    <TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
    <B>Property Summary</B></FONT></TH>
    </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>static&nbsp;List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#MG_COLUMNS">MG_COLUMNS</A></B></CODE>
            <BR>
            <P>MGHA redefined column order and contents to have a lot of things not of interest to others,
 so have a separate mapping for them.</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>static&nbsp;List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#SIMPLE_COLUMNS">SIMPLE_COLUMNS</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#analysisContact">analysisContact</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#batch">batch</A></B></CODE>
            <BR>
            <P>Id of batch in which the sample was sequenced</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#captureDates">captureDates</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Consanguinity</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#consanguinity">consanguinity</A></B></CODE>
            <BR>
            <P>Whether the sample is consanguinous</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>float</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#dnaConcentrationNg">dnaConcentrationNg</A></B></CODE>
            <BR>
            <P>DNA quality in nanograms</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#dnaDates">dnaDates</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>float</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#dnaQuality">dnaQuality</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>float</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#dnaQuantity">dnaQuantity</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Ethnicity</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#ethnicity">ethnicity</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Map</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#fileMappings">fileMappings</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Map</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#files">files</A></B></CODE>
            <BR>
            <P>List of files containing data specific to this sample, indexed by file types:
  - fastq
  - coverage (output from coverageBed)
  - vcf
  - bam</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Map</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#geneCategories">geneCategories</A></B></CODE>
            <BR>
            <P>List of genes prioritised for the sample</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#institution">institution</A></B></CODE>
            <BR>
            <P>Hospital or organization responsible for the patient from which the sample originated</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#library">library</A></B></CODE>
            <BR>
            <P>The library</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#machineIds">machineIds</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>float</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#meanCoverage">meanCoverage</A></B></CODE>
            <BR>
            <P>Mean coverage as reported by sequencing provider</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#pedigree">pedigree</A></B></CODE>
            <BR>
            <P>The pedigree of the family</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#sample">sample</A></B></CODE>
            <BR>
            <P>Sample name</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>SampleType</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#sampleType">sampleType</A></B></CODE>
            <BR>
            <P>Whether the sample type is normal or tumor</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#sequencingContact">sequencingContact</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#sequencingDates">sequencingDates</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Sex</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#sex">sex</A></B></CODE>
            <BR>
            <P>The sex of the sample</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#target">target</A></B></CODE>
            <BR>
            <P>Target (flagship) name</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#variantsFile">variantsFile</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
    </TABLE>
    &nbsp;


<!-- =========== ELEMENT SUMMARY =========== -->



<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>


<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>

    <TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
    <TR CLASS="TableHeadingColor">
    <TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2" CLASS="ClassHeadingFont">
    <B>Method Summary</B></FONT></TH>
    </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>static&nbsp;<a href='../Map.html'>Map</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#fromFiles(List)">fromFiles</a></b>(<a href='../List.html'>List</a> files)</CODE>
            <BR>
            <P>Create a list of SampleInfo objects from provided files that
 contain sample information in the header data.</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>void</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#indexFileType(java.lang.String, java.lang.String)">indexFileType</a></b>(java.lang.String index, java.lang.String ending)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>void</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#indexFileTypes()">indexFileTypes</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>static&nbsp;<a href='../Date.html'>Date</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#parseDate(java.lang.String)">parseDate</a></b>(java.lang.String dateValue)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>static&nbsp;java.lang.Object</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#parse_mg_sample_info(def)">parse_mg_sample_info</a></b>(java.lang.Object fileName)</CODE>
            <BR>
            <P>Parse the given file to extract sample info, where the file is in the
 extended Melbourne Genomics Health Alliance format.</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>static&nbsp;java.lang.Object</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#parse_sample_info(def, def)">parse_sample_info</a></b>(java.lang.Object fileName, java.lang.Object columns = SIMPLE_COLUMNS)</CODE>
            <BR>
            <P>Parse the given file to extract sample information</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#toString()">toString</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#toTsv()">toTsv</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>void</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#validate()">validate</a></b>()</CODE>
            <BR>
            <P>Validates that text fields are in the correct format.</P>
        </TD>
        </TR>
    
    </TABLE>
    &nbsp;
<table BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
                <tr CLASS="TableSubHeadingColor"><th ALIGN="left" COLSPAN="2">
                <b>Methods inherited from class java.lang.Object</b>
                </th></tr>
                <tr class="TableRowColor"><td colspan='2'>java.lang.Object#wait(), java.lang.Object#wait(long), java.lang.Object#wait(long, int), java.lang.Object#equals(java.lang.Object), java.lang.Object#toString(), java.lang.Object#hashCode(), java.lang.Object#getClass(), java.lang.Object#notify(), java.lang.Object#notifyAll()</td></tr>
                </table>
                &nbsp;

<P>

<!-- ============ ENUM CONSTANT DETAIL ========== -->

<A NAME="enum_constant_detail"><!-- --></A>


<!-- =========== FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>


<!-- =========== PROPERTY DETAIL =========== -->

<A NAME="prop_detail"><!-- --></A>

    <TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
    <TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
    <TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
    <B>Property Detail</B></FONT></TH>
    </TR>
    </TABLE>
    
        <A NAME="MG_COLUMNS"><!-- --></A><H3>MG_COLUMNS</H3>
        <PRE>static&nbsp;List <B>MG_COLUMNS</B></PRE>
        <DL>
        <DD> MGHA redefined column order and contents to have a lot of things not of interest to others,
 so have a separate mapping for them.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="SIMPLE_COLUMNS"><!-- --></A><H3>SIMPLE_COLUMNS</H3>
        <PRE>static&nbsp;List <B>SIMPLE_COLUMNS</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="analysisContact"><!-- --></A><H3>analysisContact</H3>
        <PRE>String <B>analysisContact</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="batch"><!-- --></A><H3>batch</H3>
        <PRE>String <B>batch</B></PRE>
        <DL>
        <DD> Id of batch in which the sample was sequenced 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="captureDates"><!-- --></A><H3>captureDates</H3>
        <PRE>List <B>captureDates</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="consanguinity"><!-- --></A><H3>consanguinity</H3>
        <PRE>Consanguinity <B>consanguinity</B></PRE>
        <DL>
        <DD> Whether the sample is consanguinous 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="dnaConcentrationNg"><!-- --></A><H3>dnaConcentrationNg</H3>
        <PRE>float <B>dnaConcentrationNg</B></PRE>
        <DL>
        <DD> DNA quality in nanograms 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="dnaDates"><!-- --></A><H3>dnaDates</H3>
        <PRE>List <B>dnaDates</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="dnaQuality"><!-- --></A><H3>dnaQuality</H3>
        <PRE>float <B>dnaQuality</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="dnaQuantity"><!-- --></A><H3>dnaQuantity</H3>
        <PRE>float <B>dnaQuantity</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="ethnicity"><!-- --></A><H3>ethnicity</H3>
        <PRE>Ethnicity <B>ethnicity</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="fileMappings"><!-- --></A><H3>fileMappings</H3>
        <PRE>Map <B>fileMappings</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="files"><!-- --></A><H3>files</H3>
        <PRE>Map <B>files</B></PRE>
        <DL>
        <DD> List of files containing data specific to this sample, indexed by file types:
  - fastq
  - coverage (output from coverageBed)
  - vcf
  - bam
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="geneCategories"><!-- --></A><H3>geneCategories</H3>
        <PRE>Map <B>geneCategories</B></PRE>
        <DL>
        <DD> List of genes prioritised for the sample 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="institution"><!-- --></A><H3>institution</H3>
        <PRE>String <B>institution</B></PRE>
        <DL>
        <DD> Hospital or organization responsible for the patient from which the sample originated 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="library"><!-- --></A><H3>library</H3>
        <PRE>String <B>library</B></PRE>
        <DL>
        <DD> The library 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="machineIds"><!-- --></A><H3>machineIds</H3>
        <PRE>List <B>machineIds</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="meanCoverage"><!-- --></A><H3>meanCoverage</H3>
        <PRE>float <B>meanCoverage</B></PRE>
        <DL>
        <DD> Mean coverage as reported by sequencing provider 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="pedigree"><!-- --></A><H3>pedigree</H3>
        <PRE>String <B>pedigree</B></PRE>
        <DL>
        <DD> The pedigree of the family 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="sample"><!-- --></A><H3>sample</H3>
        <PRE>String <B>sample</B></PRE>
        <DL>
        <DD> Sample name 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="sampleType"><!-- --></A><H3>sampleType</H3>
        <PRE>SampleType <B>sampleType</B></PRE>
        <DL>
        <DD> Whether the sample type is normal or tumor 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="sequencingContact"><!-- --></A><H3>sequencingContact</H3>
        <PRE>String <B>sequencingContact</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="sequencingDates"><!-- --></A><H3>sequencingDates</H3>
        <PRE>List <B>sequencingDates</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="sex"><!-- --></A><H3>sex</H3>
        <PRE>Sex <B>sex</B></PRE>
        <DL>
        <DD> The sex of the sample 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="target"><!-- --></A><H3>target</H3>
        <PRE>String <B>target</B></PRE>
        <DL>
        <DD> Target (flagship) name 
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="variantsFile"><!-- --></A><H3>variantsFile</H3>
        <PRE>String <B>variantsFile</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
    &nbsp;


<!-- =========== ELEMENT DETAIL =========== -->

<A NAME="element_detail"><!-- --></A>


<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>


<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>

    <TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
    <TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
    <TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
    <B>Method Detail</B></FONT></TH>
    </TR>
    </TABLE>
    
        <A NAME="fromFiles(List)"><!-- --></A><H3>
        fromFiles</H3>
        <PRE>static&nbsp;<a href='../Map.html'>Map</a> <B>fromFiles</B>(<a href='../List.html'>List</a> files)</PRE>
        <DL>
        <DD> Create a list of SampleInfo objects from provided files that
 contain sample information in the header data. <DL><DT><B>Parameters:</B></DT><DD>files</DD></DL><DL><DT><B>Returns:</B></DT><DD></DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="indexFileType(java.lang.String, java.lang.String)"><!-- --></A><H3>
        indexFileType</H3>
        <PRE>void <B>indexFileType</B>(java.lang.String index, java.lang.String ending)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="indexFileTypes()"><!-- --></A><H3>
        indexFileTypes</H3>
        <PRE>void <B>indexFileTypes</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="parseDate(java.lang.String)"><!-- --></A><H3>
        parseDate</H3>
        <PRE>static&nbsp;<a href='../Date.html'>Date</a> <B>parseDate</B>(java.lang.String dateValue)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="parse_mg_sample_info(def)"><!-- --></A><H3>
        parse_mg_sample_info</H3>
        <PRE>static&nbsp;java.lang.Object <B>parse_mg_sample_info</B>(java.lang.Object fileName)</PRE>
        <DL>
        <DD> Parse the given file to extract sample info, where the file is in the
 extended Melbourne Genomics Health Alliance format. <DL><DT><B>Parameters:</B></DT><DD>fileName</DD></DL><DL><DT><B>Returns:</B></DT><DD></DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="parse_sample_info(def, def)"><!-- --></A><H3>
        parse_sample_info</H3>
        <PRE>static&nbsp;java.lang.Object <B>parse_sample_info</B>(java.lang.Object fileName, java.lang.Object columns = SIMPLE_COLUMNS)</PRE>
        <DL>
        <DD> Parse the given file to extract sample information
     <DL><DT><B>Returns:</B></DT><DD>List of (Map) objects defining properties of each
          sample: 
              <li>sample name (sample), 
              <li>FastQ files (files), 
              <li>Name of flagship (target)
              <li>Genes to be classed as high priority (genes)</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="toString()"><!-- --></A><H3>
        toString</H3>
        <PRE>java.lang.String <B>toString</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="toTsv()"><!-- --></A><H3>
        toTsv</H3>
        <PRE>java.lang.String <B>toTsv</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="validate()"><!-- --></A><H3>
        validate</H3>
        <PRE>void <B>validate</B>()</PRE>
        <DL>
        <DD> Validates that text fields are in the correct format.
 <p>
 This format validation is specific to melbourne genomics.
	 
        </DD>
        <P>
        </DL>
        <HR>
    
    &nbsp;


<!-- ========= END OF CLASS DATA ========= -->
<p>Groovy Documentation</p>
<hr>

</body>
</html>
