<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- **************************************************************** -->
<!-- *  PLEASE KEEP COMPLICATED EXPRESSIONS OUT OF THESE TEMPLATES, * -->
<!-- *  i.e. only iterate & print data where possible. Thanks, Jez. * -->
<!-- **************************************************************** -->

<html>
<head>
<!-- Generated by groovydoc (2.1.0) on Tue Jul 29 16:04:20 EST 2014 -->
<title>Variant (groovy-ngs-utils API)</title>
<meta name="date" content="2014-07-29">
<meta http-equiv="Content-Type" content="text/html; charset=MacRoman">
<link href="../groovy.ico" type="image/x-icon" rel="shortcut icon">
<link href="../groovy.ico" type="image/x-icon" rel="icon">
<link rel="stylesheet" type="text/css" href="../stylesheet.css" title="Style">
<script type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Variant (groovy-ngs-utils API)";
    }
}
</script>
<noscript>
</noscript>

</head><body onload="windowTitle();" bgcolor="white">
<hr>

<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>

  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <!--<TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  -->
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
<b>Groovy Documentation</b>
</EM></TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><!--<FONT SIZE="-2">
&nbsp;<A HREF="../groovy/lang/ExpandoMetaClass.ExpandoMetaProperty.html" title="class in groovy.lang"><B>PREV CLASS</B></A>&nbsp;

&nbsp;<A HREF="../groovy/lang/GroovyClassLoader.html" title="class in groovy.lang"><B>NEXT CLASS</B></A></FONT>--></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../index.html?DefaultPackage/Variant.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="Variant.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<script type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../allclasses-frame.html"><B>All Classes</B></A>');
  }
  //-->
</script>
<noscript>
  <A HREF="../allclasses-frame.html"><B>All Classes</B></A>
</noscript>

</FONT></TD>

</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#property_summary">PROPERTY</A>&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#prop_detail">PROPERTY</A>&nbsp;|&nbsp;CONSTR&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<!-- ========= END OF TOP NAVBAR ========= -->


<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>

<span CLASS="ClassTitleFont">[Groovy] Class Variant</span></H2>
<pre>java.lang.Object
  <img src='../inherit.gif'>Variant
</pre><dl><dt><b>All Implemented Interfaces:</b> </dt><dd><a href='../DefaultPackage/IRegion.html'>IRegion</a></dd></dl><hr>
<PRE>class Variant

</PRE>

<P>
 Represents the genetic state at a specific locus in a genome
 ie: captures the information at a single line of a VCF file.
 Note that this can include multiple alleles that may be present
 at the site, having different start and end position.
 <p>
 The Variant class includes support for accessing the structured
 annotations and attributes included in the VCF format and popular
 annotation tools such as VEP, Annovar and SnpEFF. See <a href='#getSnpEffInfo()'>getSnpEffInfo()</a>
 and <a href='#getVepInfo()'>getVepInfo()</a> for accessors that return parsed annotation
 information.
 <p>
 Although Variant instances can be constructed manually, typically
 you will query them from VCF or VCFIndex classes, and this
 is required for some operations because the VCF header is required to 
 parse and understand some of the fields (for example, sample names). Without
 a header many operations still work, but you may experience IllegalStateException
 exceptions if you call methods that depend on the header.
 <p>
 One of the most useful operations is to determine the dosage (ie: heterozygosity, number of copies)
 of a variant in a particular sample. Eg:
 <pre>
 VCF vcf = new VCF.parse("test.vcf")
 
 // Find all the variants that affect sample MYSAMPLE in any way
 List<Variant> mySampleVariants = vcf.grep {  Variant v -> v.sampleDosage("MYSAMPLE")>0 }
 
 // Find all the homozygous variants (for autosomal + female X chromosomes) 
 List<Variant> mySampleVariants = vcf.grep {  Variant v -> v.sampleDosage("MYSAMPLE") == 2 }
 ...
 </pre>
 
 Some support is implemented for understanding pedigrees via the Pedigree class.
 If a pedigree is set on the VCF class from which the Variant originates, 
 then one can query variants to find variants that segregate with families or conditions.
 <p>
 A Variant also implements the IRange interface and thus you can use
 them seamlessly with a RegionSource. For example, to check
 if a Variant falls within a BED file works simply:
 <pre>
 Variant v = ...
 BED bed = new BED("test.bed").load()
 if(v in bed) 
  println "Variant v falls in the ranges included in the BED file"
 </pre>
 
 The Variant class supports working with the full genotype data of each sample, and
 the genotype fields are parsed for you. For example, to check if every sample is below
 a particular genotype quality:
 <pre>
 if(v.genoTypes.every { it.GQ < 5.0f })
     println "Quality is low for every sample!"
 </pre>
 <DL><DT><B>Authors:</B></DT><DD>simon.sadedin@mcri.edu.au</DD></DL>
</P>
<hr>


<!-- =========== NESTED CLASS SUMMARY =========== -->

<A NAME="nested_summary"><!-- --></A>


<!-- =========== ENUM CONSTANT SUMMARY =========== -->

<A NAME="enum_constant_summary"><!-- --></A>


<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>


<!-- =========== PROPERTY SUMMARY =========== -->

<A NAME="property_summary"><!-- --></A>

    <TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
    <TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
    <TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
    <B>Property Summary</B></FONT></TH>
    </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#alt">alt</A></B></CODE>
            <BR>
            <P>The sequence for the first alternate allele. </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>byte</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#altByte">altByte</A></B></CODE>
            <BR>
            <P>The first character of the first alternate allele in byte form. </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#alts">alts</A></B></CODE>
            <BR>
            <P>List of alternate alleles represented by the variant.</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Region</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#cachedRegion">cachedRegion</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#chr">chr</A></B></CODE>
            <BR>
            <P>The chromosome on which the variant falls</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#dosages">dosages</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#filter">filter</A></B></CODE>
            <BR>
            <P>The filter field as loaded from the VCF file.</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#genoTypeFields">genoTypeFields</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#genoTypes">genoTypes</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>VCF</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#header">header</A></B></CODE>
            <BR>
            <P>The VCF to which this Variant is linked. </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#id">id</A></B></CODE>
            <BR>
            <P>ID column as per VCF spec. </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#info">info</A></B></CODE>
            <BR>
            <P>The whole INFO field as a raw string. </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Map</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#infos">infos</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#line">line</A></B></CODE>
            <BR>
            <P>The original line from which this variant was parsed</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>Set</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#pedigrees">pedigrees</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>int</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#pos">pos</A></B></CODE>
            <BR>
            <P>The position at which the reference bases (<a href='#ref'>ref</a> attribute) starts.
 </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>float</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#qual">qual</A></B></CODE>
            <BR>
            <P>The QUAL field as loaded from the VCF file</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#ref">ref</A></B></CODE>
            <BR>
            <P>The reference allele at the position. </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>boolean</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#snpEffDirty">snpEffDirty</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>List</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#snpEffInfo">snpEffInfo</A></B></CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
        <CODE>String</CODE></FONT></TD>
        <TD>
            <CODE><B><A HREF="#type">type</A></B></CODE>
            <BR>
            <P>One of SNP, INS, DEL or SV indicating the type of change represented by the first 
 alternate allele.</P>
        </TD>
        </TR>
    
    </TABLE>
    &nbsp;


<!-- =========== ELEMENT SUMMARY =========== -->



<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>


<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>

    <TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
    <TR CLASS="TableHeadingColor">
    <TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2" CLASS="ClassHeadingFont">
    <B>Method Summary</B></FONT></TH>
    </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#applyDEL(java.lang.String, int)">applyDEL</a></b>(java.lang.String sequence, int position)</CODE>
            <BR>
            <P>Apply this deletion to the specified sequence at the specified position</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#applyINS(java.lang.String, int)">applyINS</a></b>(java.lang.String sequence, int position)</CODE>
            <BR>
            <P>Apply this deletion to the specified sequence at the specified position</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#applySNP(java.lang.String, int)">applySNP</a></b>(java.lang.String sequence, int position)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#applyTo(java.lang.String, int)">applyTo</a></b>(java.lang.String sequence, int position)</CODE>
            <BR>
            <P>Apply this variant to mutate the given sequence at the specified position</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#convertType(java.lang.String, java.lang.String)">convertType</a></b>(java.lang.String refSeq, java.lang.String altSeq)</CODE>
            <BR>
            <P>Convert reference and alternate allele strings into a 
 mutation type, being one of "SNP","INS","DEL","GAIN"
 or "LOSS", with the latter two representing CNVs.</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#displaySequenceDifference(java.lang.String, int)">displaySequenceDifference</a></b>(java.lang.String sequence, int position)</CODE>
            <BR>
            <P>A debug function, displays the sequence difference and returns the 
 new sequence</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>int</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#equalsAnnovar(java.lang.String, int, java.lang.String)">equalsAnnovar</a></b>(java.lang.String chr, int pos, java.lang.String obs)</CODE>
            <BR>
            <P>Return true if the given Annovar formattd position and observation
 match those of this variant</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>int</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#findAlleleIndex(Allele)">findAlleleIndex</a></b>(<a href='../DefaultPackage/Allele.html'>Allele</a> other)</CODE>
            <BR>
            <P>Return the index of the allele (if any) that matches the give other variant</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>void</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#fixGenoTypeOrder()">fixGenoTypeOrder</a></b>()</CODE>
            <BR>
            <P>VCF requires genotype to be the 1st field, but some tools (R, grrrr) write it in whatever order they 
 feel like. </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../List.html'>List</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getAlleles()">getAlleles</a></b>()</CODE>
            <BR>
            <P>Return a list of Allele objects representing alleles present on this line of the VCF</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../List.html'>List</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getAllelesAndTypes()">getAllelesAndTypes</a></b>()</CODE>
            <BR>
            <P>Return a list of each allele and its type.
 </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getConsequence(int)">getConsequence</a></b>(int alleleIndex)</CODE>
            <BR>
            <P>Return the consequence of the specified allele, at the moment,
 from VEP annotations (later, from others). </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../List.html'>List</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getDosages()">getDosages</a></b>()</CODE>
            <BR>
            <P>Return list of dosages (number of copies of allele) for 
 each sample for the first alternate allele.
 </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../List.html'>List</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getDosages(int)">getDosages</a></b>(int alleleIndex)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../List.html'>List</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getGenes(java.lang.String)">getGenes</a></b>(java.lang.String minVEPCons)</CODE>
            <BR>
            <P>Return the list of genes impacted by this variant, in a manner that is
 neutral to the annotator used (VEP and SnpEFF supported)</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../Map.html'>Map</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getInfo()">getInfo</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../DefaultPackage/SnpEffInfo.html'>SnpEffInfo</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getMaxEffect()">getMaxEffect</a></b>()</CODE>
            <BR>
            <P>Return the most impactful SnpEff effect for this variant</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../Map.html'>Map</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getMaxVep()">getMaxVep</a></b>()</CODE>
            <BR>
            <P>Return the details of the most severe VEP consequence, as ranked by
 VEPConsequences#RANKED_CONSEQUENCES#RANKED_CONSEQUENCES.</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>float</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getMaxVepMaf()">getMaxVepMaf</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../Set.html'>Set</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getPedigrees()">getPedigrees</a></b>()</CODE>
            <BR>
            <P>Return a list of all the pedigrees that contain this variant</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../IntRange.html'>IntRange</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getRange()">getRange</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../DefaultPackage/Region.html'>Region</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getRegion()">getRegion</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../List.html'>List</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getSnpEffInfo()">getSnpEffInfo</a></b>()</CODE>
            <BR>
            <P>Return a list of SnpEffInfo objects, each describing
 a separate SnpEff effect caused by the variant
 @return</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../List.html'>List</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#getVepInfo()">getVepInfo</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>boolean</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#isCase(IRegion)">isCase</a></b>(<a href='../DefaultPackage/IRegion.html'>IRegion</a> r)</CODE>
            <BR>
            <P>Return true if this variant overlaps the given range</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>boolean</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#isSV()">isSV</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>static&nbsp;<a href='../DefaultPackage/Variant.html'>Variant</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#parse(java.lang.String)">parse</a></b>(java.lang.String line)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>int</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#sampleDosage(java.lang.String)">sampleDosage</a></b>(java.lang.String sampleName)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>int</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#sampleDosage(java.lang.String, int)">sampleDosage</a></b>(java.lang.String sampleName, int alleleIndex)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../Map.html'>Map</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#sampleGenoType(java.lang.String)">sampleGenoType</a></b>(java.lang.String sampleName)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>boolean</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#segregatesWith(Pedigree)">segregatesWith</a></b>(<a href='../DefaultPackage/Pedigree.html'>Pedigree</a> ped)</CODE>
            <BR>
            <P>Return a Map of fields from the genotype column corresponding
 to the given sample. </P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>void</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#setAlt(java.lang.String)">setAlt</a></b>(java.lang.String alt)</CODE>
            <BR>
            <P>Update the first alternate allele to the given value</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>int</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#size()">size</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE><a href='../Map.html'>Map</a></CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#toAnnovar(int)">toAnnovar</a></b>(int alleleIndex = 0)</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#toJson()">toJson</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>java.lang.String</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#toString()">toString</a></b>()</CODE>
            <BR>
            <P></P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>void</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#update(Closure)">update</a></b>(<a href='../Closure.html'>Closure</a> c)</CODE>
            <BR>
            <P>Update using a default description.</P>
        </TD>
        </TR>
    
        <TR BGCOLOR="white" CLASS="TableRowColor">
        <TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1" CLASS="ClassItemFont">
            <CODE>void</CODE></FONT>
        </TD>
        <TD>
            <CODE><b><a href="#update(java.lang.String, Closure)">update</a></b>(java.lang.String desc, <a href='../Closure.html'>Closure</a> c)</CODE>
            <BR>
            <P>Allows various fields to be updated and then synchronises the 
 rest of the data with those updated fields</P>
        </TD>
        </TR>
    
    </TABLE>
    &nbsp;
<table BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
                <tr CLASS="TableSubHeadingColor"><th ALIGN="left" COLSPAN="2">
                <b>Methods inherited from class java.lang.Object</b>
                </th></tr>
                <tr class="TableRowColor"><td colspan='2'>java.lang.Object#wait(), java.lang.Object#wait(long), java.lang.Object#wait(long, int), java.lang.Object#equals(java.lang.Object), java.lang.Object#toString(), java.lang.Object#hashCode(), java.lang.Object#getClass(), java.lang.Object#notify(), java.lang.Object#notifyAll()</td></tr>
                </table>
                &nbsp;

<P>

<!-- ============ ENUM CONSTANT DETAIL ========== -->

<A NAME="enum_constant_detail"><!-- --></A>


<!-- =========== FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>


<!-- =========== PROPERTY DETAIL =========== -->

<A NAME="prop_detail"><!-- --></A>

    <TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
    <TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
    <TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
    <B>Property Detail</B></FONT></TH>
    </TR>
    </TABLE>
    
        <A NAME="alt"><!-- --></A><H3>alt</H3>
        <PRE>String <B>alt</B></PRE>
        <DL>
        <DD> The sequence for the first alternate allele. This is a convenience for the common case where 
 the first alternate allele is the only one of interest. In general you should be aware that
 more than one alternate Allele represented by a single Variant.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="altByte"><!-- --></A><H3>altByte</H3>
        <PRE>byte <B>altByte</B></PRE>
        <DL>
        <DD> The first character of the first alternate allele in byte form. This is a special case
 optimization to allow for fast access when interoperating with Picard (which wants to see
 bytes, not char or Strings).
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="alts"><!-- --></A><H3>alts</H3>
        <PRE>String <B>alts</B></PRE>
        <DL>
        <DD> List of alternate alleles represented by the variant.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="cachedRegion"><!-- --></A><H3>cachedRegion</H3>
        <PRE>Region <B>cachedRegion</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="chr"><!-- --></A><H3>chr</H3>
        <PRE>String <B>chr</B></PRE>
        <DL>
        <DD> The chromosome on which the variant falls
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="dosages"><!-- --></A><H3>dosages</H3>
        <PRE>List <B>dosages</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="filter"><!-- --></A><H3>filter</H3>
        <PRE>String <B>filter</B></PRE>
        <DL>
        <DD> The filter field as loaded from the VCF file.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="genoTypeFields"><!-- --></A><H3>genoTypeFields</H3>
        <PRE>String <B>genoTypeFields</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="genoTypes"><!-- --></A><H3>genoTypes</H3>
        <PRE>List <B>genoTypes</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="header"><!-- --></A><H3>header</H3>
        <PRE>VCF <B>header</B></PRE>
        <DL>
        <DD> The VCF to which this Variant is linked. This enables the variant to intelligently
 parse INFO fields and be aware of sample names. If the header is not provided,
 many functions still work, but some functions will be disabled.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="id"><!-- --></A><H3>id</H3>
        <PRE>String <B>id</B></PRE>
        <DL>
        <DD> ID column as per VCF spec. Usually this contains the rsID (dbSNP) identifer
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="info"><!-- --></A><H3>info</H3>
        <PRE>String <B>info</B></PRE>
        <DL>
        <DD> The whole INFO field as a raw string. Users will generally call <a href='#getInfo()'>getInfo()</a> to get
 this field in parsed form rather than accessing this field directly.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="infos"><!-- --></A><H3>infos</H3>
        <PRE>Map <B>infos</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="line"><!-- --></A><H3>line</H3>
        <PRE>String <B>line</B></PRE>
        <DL>
        <DD> The original line from which this variant was parsed
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="pedigrees"><!-- --></A><H3>pedigrees</H3>
        <PRE>Set <B>pedigrees</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="pos"><!-- --></A><H3>pos</H3>
        <PRE>int <B>pos</B></PRE>
        <DL>
        <DD> The position at which the reference bases (<a href='#ref'>ref</a> attribute) starts.
 Note that this is <i>not</i> necessarily the start of the genetic change represented
 by the variant.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="qual"><!-- --></A><H3>qual</H3>
        <PRE>float <B>qual</B></PRE>
        <DL>
        <DD> The QUAL field as loaded from the VCF file
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="ref"><!-- --></A><H3>ref</H3>
        <PRE>String <B>ref</B></PRE>
        <DL>
        <DD> The reference allele at the position. Note that as per VCF spec, this is not necessarily
 the start of the change indicated by the Variant. For the start of actual changes,
 you need to query the Allele objects and look at the Allele#start#start and
 Allele#end#end attributes.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="snpEffDirty"><!-- --></A><H3>snpEffDirty</H3>
        <PRE>boolean <B>snpEffDirty</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="snpEffInfo"><!-- --></A><H3>snpEffInfo</H3>
        <PRE>List <B>snpEffInfo</B></PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="type"><!-- --></A><H3>type</H3>
        <PRE>String <B>type</B></PRE>
        <DL>
        <DD> One of SNP, INS, DEL or SV indicating the type of change represented by the first 
 alternate allele. 
     
        </DD>
        <P>
        </DL>
        <HR>
    
    &nbsp;


<!-- =========== ELEMENT DETAIL =========== -->

<A NAME="element_detail"><!-- --></A>


<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>


<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>

    <TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
    <TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
    <TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
    <B>Method Detail</B></FONT></TH>
    </TR>
    </TABLE>
    
        <A NAME="applyDEL(java.lang.String, int)"><!-- --></A><H3>
        applyDEL</H3>
        <PRE>java.lang.String <B>applyDEL</B>(java.lang.String sequence, int position)</PRE>
        <DL>
        <DD> Apply this deletion to the specified sequence at the specified position
 <p>NOTE: this variant MUST be a deletion
     <DL><DT><B>Parameters:</B></DT><DD>sequence</DD><DD>position</DD></DL><DL><DT><B>Returns:</B></DT><DD>given sequence mutated to include this deletion</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="applyINS(java.lang.String, int)"><!-- --></A><H3>
        applyINS</H3>
        <PRE>java.lang.String <B>applyINS</B>(java.lang.String sequence, int position)</PRE>
        <DL>
        <DD> Apply this deletion to the specified sequence at the specified position
 <p>NOTE: this variant MUST be a deletion
     <DL><DT><B>Parameters:</B></DT><DD>sequence</DD><DD>position</DD></DL><DL><DT><B>Returns:</B></DT><DD>given sequence mutated to include this deletion</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="applySNP(java.lang.String, int)"><!-- --></A><H3>
        applySNP</H3>
        <PRE>java.lang.String <B>applySNP</B>(java.lang.String sequence, int position)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="applyTo(java.lang.String, int)"><!-- --></A><H3>
        applyTo</H3>
        <PRE>java.lang.String <B>applyTo</B>(java.lang.String sequence, int position)</PRE>
        <DL>
        <DD> Apply this variant to mutate the given sequence at the specified position
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="convertType(java.lang.String, java.lang.String)"><!-- --></A><H3>
        convertType</H3>
        <PRE>@CompileStatic
java.lang.String <B>convertType</B>(java.lang.String refSeq, java.lang.String altSeq)</PRE>
        <DL>
        <DD> Convert reference and alternate allele strings into a 
 mutation type, being one of "SNP","INS","DEL","GAIN"
 or "LOSS", with the latter two representing CNVs. <DL><DT><B>Parameters:</B></DT><DD>refSeq</DD><DD>altSeq</DD></DL><DL><DT><B>Returns:</B></DT><DD></DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="displaySequenceDifference(java.lang.String, int)"><!-- --></A><H3>
        displaySequenceDifference</H3>
        <PRE>java.lang.String <B>displaySequenceDifference</B>(java.lang.String sequence, int position)</PRE>
        <DL>
        <DD> A debug function, displays the sequence difference and returns the 
 new sequence
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="equalsAnnovar(java.lang.String, int, java.lang.String)"><!-- --></A><H3>
        equalsAnnovar</H3>
        <PRE>@CompileStatic
int <B>equalsAnnovar</B>(java.lang.String chr, int pos, java.lang.String obs)</PRE>
        <DL>
        <DD> Return true if the given Annovar formattd position and observation
 match those of this variant
 <p>
 Annovar outputs non-standard formatting that makes it difficult to trace
 an Annovar variant back to it's VCF source. This method implements
 the tricky logic to compare an Annovar variant to a VCF equivalent
 and say if they are the same.
     <DL><DT><B>Returns:</B></DT><DD>if the Annovar variant is not the same variant as this VCF variant,
         returns 0. If it is the same, returns the number of the alternate
         allele (if the variant has only one alternate, the return value will 
         be 1).</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="findAlleleIndex(Allele)"><!-- --></A><H3>
        findAlleleIndex</H3>
        <PRE>int <B>findAlleleIndex</B>(<a href='../DefaultPackage/Allele.html'>Allele</a> other)</PRE>
        <DL>
        <DD> Return the index of the allele (if any) that matches the give other variant <DL><DT><B>Parameters:</B></DT><DD>other</DD></DL><DL><DT><B>Returns:</B></DT><DD></DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="fixGenoTypeOrder()"><!-- --></A><H3>
        fixGenoTypeOrder</H3>
        <PRE>void <B>fixGenoTypeOrder</B>()</PRE>
        <DL>
        <DD> VCF requires genotype to be the 1st field, but some tools (R, grrrr) write it in whatever order they 
 feel like. This function corrects it.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getAlleles()"><!-- --></A><H3>
        getAlleles</H3>
        <PRE><a href='../List.html'>List</a> <B>getAlleles</B>()</PRE>
        <DL>
        <DD> Return a list of Allele objects representing alleles present on this line of the VCF
 <p>
 Note: currently this computes the results on the fly and they are not cached, so
       use with caution in computationaly intensive situations
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getAllelesAndTypes()"><!-- --></A><H3>
        getAllelesAndTypes</H3>
        <PRE><a href='../List.html'>List</a> <B>getAllelesAndTypes</B>()</PRE>
        <DL>
        <DD> Return a list of each allele and its type.
 Yes, VCF allows multiple types (INS,DEL) to be on the same line of 
 a VCF file.<DL><DT><B>deprecated:</B></DT><DD>use getAlleles() instead</DD></DL>
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getConsequence(int)"><!-- --></A><H3>
        getConsequence</H3>
        <PRE>java.lang.String <B>getConsequence</B>(int alleleIndex)</PRE>
        <DL>
        <DD> Return the consequence of the specified allele, at the moment,
 from VEP annotations (later, from others). If multiple consequences
 are present for the same allele, then the most severe consequence is 
 returned.
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getDosages()"><!-- --></A><H3>
        getDosages</H3>
        <PRE><a href='../List.html'>List</a> <B>getDosages</B>()</PRE>
        <DL>
        <DD> Return list of dosages (number of copies of allele) for 
 each sample for the first alternate allele. <DL><DT><B>Returns:</B></DT><DD></DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getDosages(int)"><!-- --></A><H3>
        getDosages</H3>
        <PRE><a href='../List.html'>List</a> <B>getDosages</B>(int alleleIndex)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getGenes(java.lang.String)"><!-- --></A><H3>
        getGenes</H3>
        <PRE><a href='../List.html'>List</a> <B>getGenes</B>(java.lang.String minVEPCons)</PRE>
        <DL>
        <DD> Return the list of genes impacted by this variant, in a manner that is
 neutral to the annotator used (VEP and SnpEFF supported)
     <DL><DT><B>Parameters:</B></DT><DD><code>minVEPCons</code> -   the VEP impact level above which genes should
                      be returned</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getInfo()"><!-- --></A><H3>
        getInfo</H3>
        <PRE>@CompileStatic
<a href='../Map.html'>Map</a> <B>getInfo</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getMaxEffect()"><!-- --></A><H3>
        getMaxEffect</H3>
        <PRE><a href='../DefaultPackage/SnpEffInfo.html'>SnpEffInfo</a> <B>getMaxEffect</B>()</PRE>
        <DL>
        <DD> Return the most impactful SnpEff effect for this variant
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getMaxVep()"><!-- --></A><H3>
        getMaxVep</H3>
        <PRE><a href='../Map.html'>Map</a> <B>getMaxVep</B>()</PRE>
        <DL>
        <DD> Return the details of the most severe VEP consequence, as ranked by
 VEPConsequences#RANKED_CONSEQUENCES#RANKED_CONSEQUENCES.
     <DL><DT><B>Returns:</B></DT><DD>A map with key value pairs of VEP annotation fields.</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getMaxVepMaf()"><!-- --></A><H3>
        getMaxVepMaf</H3>
        <PRE>@CompileStatic
float <B>getMaxVepMaf</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getPedigrees()"><!-- --></A><H3>
        getPedigrees</H3>
        <PRE><a href='../Set.html'>Set</a> <B>getPedigrees</B>()</PRE>
        <DL>
        <DD> Return a list of all the pedigrees that contain this variant
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getRange()"><!-- --></A><H3>
        getRange</H3>
        <PRE><a href='../IntRange.html'>IntRange</a> <B>getRange</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getRegion()"><!-- --></A><H3>
        getRegion</H3>
        <PRE>@CompileStatic
<a href='../DefaultPackage/Region.html'>Region</a> <B>getRegion</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getSnpEffInfo()"><!-- --></A><H3>
        getSnpEffInfo</H3>
        <PRE><a href='../List.html'>List</a> <B>getSnpEffInfo</B>()</PRE>
        <DL>
        <DD> Return a list of SnpEffInfo objects, each describing
 a separate SnpEff effect caused by the variant <DL><DT><B>Returns:</B></DT><DD></DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="getVepInfo()"><!-- --></A><H3>
        getVepInfo</H3>
        <PRE><a href='../List.html'>List</a> <B>getVepInfo</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="isCase(IRegion)"><!-- --></A><H3>
        isCase</H3>
        <PRE>boolean <B>isCase</B>(<a href='../DefaultPackage/IRegion.html'>IRegion</a> r)</PRE>
        <DL>
        <DD> Return true if this variant overlaps the given range
     
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="isSV()"><!-- --></A><H3>
        isSV</H3>
        <PRE>@CompileStatic
boolean <B>isSV</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="parse(java.lang.String)"><!-- --></A><H3>
        parse</H3>
        <PRE>static&nbsp;<a href='../DefaultPackage/Variant.html'>Variant</a> <B>parse</B>(java.lang.String line)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="sampleDosage(java.lang.String)"><!-- --></A><H3>
        sampleDosage</H3>
        <PRE>@CompileStatic
int <B>sampleDosage</B>(java.lang.String sampleName)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="sampleDosage(java.lang.String, int)"><!-- --></A><H3>
        sampleDosage</H3>
        <PRE>@CompileStatic
int <B>sampleDosage</B>(java.lang.String sampleName, int alleleIndex)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="sampleGenoType(java.lang.String)"><!-- --></A><H3>
        sampleGenoType</H3>
        <PRE>@CompileStatic 
    /**
     * Return a Map of fields from the genotype column corresponding
     * to the given sample. Common fields include:
     * <li>AD - allele depth (list of numeric values)
     * <li>GQ - genotype qualities for each genotype (string value)
     * <li>GT - list of actual genotypes for sample for each allele (string)
     * <li>DP - total depth (integer)
     * @param sampleName
     */
<a href='../Map.html'>Map</a> <B>sampleGenoType</B>(java.lang.String sampleName)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="segregatesWith(Pedigree)"><!-- --></A><H3>
        segregatesWith</H3>
        <PRE>boolean <B>segregatesWith</B>(<a href='../DefaultPackage/Pedigree.html'>Pedigree</a> ped)</PRE>
        <DL>
        <DD> Return a Map of fields from the genotype column corresponding
 to the given sample. Common fields include:
 <li>AD - allele depth (list of numeric values)
 <li>GQ - genotype qualities for each genotype (string value)
 <li>GT - list of actual genotypes for sample for each allele (string)
 <li>DP - total depth (integer)
     <DL><DT><B>Parameters:</B></DT><DD>sampleName</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="setAlt(java.lang.String)"><!-- --></A><H3>
        setAlt</H3>
        <PRE>void <B>setAlt</B>(java.lang.String alt)</PRE>
        <DL>
        <DD> Update the first alternate allele to the given value
     <DL><DT><B>Parameters:</B></DT><DD><code>alt</code> -    Alternate sequence of bases. Note that this should conform to VCF 
              spec, eg: for deletions it is expected at least one base of context 
              should be provided, and this should be consistent with the <a href='#pos'>pos</a>
              field.</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="size()"><!-- --></A><H3>
        size</H3>
        <PRE>@CompileStatic
int <B>size</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="toAnnovar(int)"><!-- --></A><H3>
        toAnnovar</H3>
        <PRE><a href='../Map.html'>Map</a> <B>toAnnovar</B>(int alleleIndex = 0)</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="toJson()"><!-- --></A><H3>
        toJson</H3>
        <PRE>java.lang.String <B>toJson</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="toString()"><!-- --></A><H3>
        toString</H3>
        <PRE>java.lang.String <B>toString</B>()</PRE>
        <DL>
        <DD>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="update(Closure)"><!-- --></A><H3>
        update</H3>
        <PRE>void <B>update</B>(<a href='../Closure.html'>Closure</a> c)</PRE>
        <DL>
        <DD> Update using a default description. 
 <p>
 Call this method to prepare the Variant for updating and
 then make your updates within the Closure that you pass. After 
 the closure exits, other internal fields that are impacted by your changes
 will be synchronized with the changes you made.
 <p>
 Not all fields are supported! see <a href='#update(java.lang.String, Closure)'>update(String, Closure)</a>
 <p>
 It's generally bad form to add INFO fields without annotating what you've done.
 Prefer to use <a href='#update(java.lang.String, Closure)'>update(String, Closure)</a> and pass in a description
 over this method.
     <DL><DT><B>Parameters:</B></DT><DD><code>c</code> -      Closure within updates can be made.</DD></DL>
        </DD>
        <P>
        </DL>
        <HR>
    
        <A NAME="update(java.lang.String, Closure)"><!-- --></A><H3>
        update</H3>
        <PRE>void <B>update</B>(java.lang.String desc, <a href='../Closure.html'>Closure</a> c)</PRE>
        <DL>
        <DD> Allows various fields to be updated and then synchronises the 
 rest of the data with those updated fields
 <p>
 Call this method to prepare the Variant for updating and
 then make your updates within the Closure that you pass. After 
 the closure exits, other internal fields that are impacted by your changes
 will be synchronized with the changes you made. 
 <p>
 Not all fields are supported! see the ones that are set below.
 <p>
 The only update to snpEFF information is to remove individual 
 annotations.
     
        </DD>
        <P>
        </DL>
        <HR>
    
    &nbsp;


<!-- ========= END OF CLASS DATA ========= -->
<p>Groovy Documentation</p>
<hr>

</body>
</html>
