<!--

     Licensed to the Apache Software Foundation (ASF) under one
     or more contributor license agreements.  See the NOTICE file
     distributed with this work for additional information
     regarding copyright ownership.  The ASF licenses this file
     to you under the Apache License, Version 2.0 (the
     "License"); you may not use this file except in compliance
     with the License.  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

     Unless required by applicable law or agreed to in writing,
     software distributed under the License is distributed on an
     "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
     KIND, either express or implied.  See the License for the
     specific language governing permissions and limitations
     under the License.

-->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>

<title>gngs.tools (groovy-ngs-utils API)</title>
<meta name="keywords" content="gngs.tools package">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<link rel ="stylesheet" TYPE="text/css" HREF="../../stylesheet.css" title="Style">
<link href="../../groovy.ico" type="image/x-icon" rel="shortcut icon">
<link href="../../groovy.ico" type="image/x-icon" rel="icon">
<script type="text/javascript">
function windowTitle()
{
    parent.document.title="gngs.tools (groovy-ngs-utils API)";
}
</script>
<noscript>
</noscript>

</head>

<body class="center" onload="windowTitle();">

<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
    <!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
    <!--   -->
</a>
<a name="navbar_top_firstrow"><!-- --></a>
<ul class="navList" title="Navigation">
    <li><a href="../../overview-summary.html">Overview</a></li>
    <li class="navBarCell1Rev">Package</li>
    <li>Class</li>
    <li><a href="../../deprecated-list.html">Deprecated</a></li>
    <li><a href="../../index-all.html">Index</a></li>
    <li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>

<div class="subNav">
    <ul class="navList">
        <li><a href="../../index.html?gngs/tools/package-summary.html" target="_top">Frames</a></li>
        <li><a href="package-summary.html" target="_top">No Frames</a></li>
    </ul>
</div>

<!-- ========= END OF TOP NAVBAR ========= -->

<div class="header">
    <h1 class="title">Package gngs.tools</h1>
</div>


<h2 title="Package gngs.tools Description" class="title">Package gngs.tools Description</h2>
<div class="header"> Command line tools that are supported by the <code>gngstool</code> command
 
 </div>


<div class="contentContainer">




    <div class="summary">
        <ul class="blockList">
            <li class="blockList">
                <table border="0" cellpadding="3" cellspacing="0" summary="Class Summary">
                    <caption><span>Class Summary</span><span class="tabEnd">&nbsp;</span></caption>
                    <tbody>
                    <tr>
                        <th class="colFirst" scope="col">Class</th>
                        <th class="colLast" scope="col">Description</th>
                    </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="AlleleNumber.html" title="class in gngs/tools">
                                    AlleleNumber
                                </a></strong>
                            </td>
                            <td>A utility class to contain some functionality for computing allele numbers / counts</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CoveragePosition.html" title="class in gngs/tools">
                                    CoveragePosition
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="CustomInfo.html" title="class in gngs/tools">
                                    CustomInfo
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="DeletionPlot.html" title="class in gngs/tools">
                                    DeletionPlot
                                </a></strong>
                            </td>
                            <td> <DL><DT><B>Authors:</B></DT><DD>Simon Sadedin</DD></DL></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Delfin.html" title="class in gngs/tools">
                                    Delfin
                                </a></strong>
                            </td>
                            <td>A simple exon-by-exon deletion finder</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="DownsampleFASTQ.html" title="class in gngs/tools">
                                    DownsampleFASTQ
                                </a></strong>
                            </td>
                            <td>A simple tool for downsampling paired end FASTQ to achieve lower coverage.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ExtractFASTQ.html" title="class in gngs/tools">
                                    ExtractFASTQ
                                </a></strong>
                            </td>
                            <td>Extract reads as pairs from a sorted BAM and writes them out to standard 
 output in interleaved format.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="FASTQDedupeActor.html" title="class in gngs/tools">
                                    FASTQDedupeActor
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="FastQIdent.html" title="class in gngs/tools">
                                    FastQIdent
                                </a></strong>
                            </td>
                            <td>Find sequences not present in a raw fastq file</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="GapAnnotator.html" title="class in gngs/tools">
                                    GapAnnotator
                                </a></strong>
                            </td>
                            <td>Annotates regions with gene, transcript, CDS overlap and distance information, as well
 as custom annotation to gene associations ("panels") and sub-panels 
 (groups of genes within a panel)</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="GeneCovStats.html" title="class in gngs/tools">
                                    GeneCovStats
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Genotype.html" title="class in gngs/tools">
                                    Genotype
                                </a></strong>
                            </td>
                            <td>A simple tool that measures the status of predefined SNPs in a 
 a BAM file, where SNPs are provided as a VCF file.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="HeaderCat.html" title="class in gngs/tools">
                                    HeaderCat
                                </a></strong>
                            </td>
                            <td>A utility to concatenate files that may have headers which 
 should only be included once at the top.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="LCSHPlot.html" title="class in gngs/tools">
                                    LCSHPlot
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="LiftOver.html" title="class in gngs/tools">
                                    LiftOver
                                </a></strong>
                            </td>
                            <td>Simple tool that translates between genome coordinates with automatic
 download and caching of the chain files.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="MeanCoverageEstimator.html" title="class in gngs/tools">
                                    MeanCoverageEstimator
                                </a></strong>
                            </td>
                            <td>Estimates the mean coverage of a BAM file quickly through random sampling.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="MergePartitionedVCFs.html" title="class in gngs/tools">
                                    MergePartitionedVCFs
                                </a></strong>
                            </td>
                            <td>A tool to merge together multiple VCFs that contain variants from distinct regions
 but which may be interleaved between different VCFs</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="MultiCov.html" title="class in gngs/tools">
                                    MultiCov
                                </a></strong>
                            </td>
                            <td>A command line tool to rapidly calculate per-base coverage across multiple BAM files
 with extended support for coverage normalisation and variability calculation.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Pairs.html" title="class in gngs/tools">
                                    Pairs
                                </a></strong>
                            </td>
                            <td>Extracts read pairs from BAM or CRAM files sorted in coordinate order.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="PedTrio.html" title="class in gngs/tools">
                                    PedTrio
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ReadSpans.html" title="class in gngs/tools">
                                    ReadSpans
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SampleCorrelationPlot.html" title="class in gngs/tools">
                                    SampleCorrelationPlot
                                </a></strong>
                            </td>
                            <td>Computes pairwise SNP correlation for a set of input VCFs, outputting as a plot
 or a table in TSV format.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Sex.html" title="class in gngs/tools">
                                    Sex
                                </a></strong>
                            </td>
                            <td>Estimates the sex of a sample from one or more VCF file and prints to the console.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="ShearingKmerCounter.html" title="class in gngs/tools">
                                    ShearingKmerCounter
                                </a></strong>
                            </td>
                            <td>Tool to count the frequency of kmers at the start
 of each read in a BAM file. </td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SortFASTH10X.html" title="class in gngs/tools">
                                    SortFASTH10X
                                </a></strong>
                            </td>
                            <td>Utility to sort interleaved 10X FASTH format</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SplitFASTQ10X.html" title="class in gngs/tools">
                                    SplitFASTQ10X
                                </a></strong>
                            </td>
                            <td>Utility to split paired end FASTQ and write it out in interleaved 10X format</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="SplitLiveFASTQ.html" title="class in gngs/tools">
                                    SplitLiveFASTQ
                                </a></strong>
                            </td>
                            <td>Splits FASTQ in block-gzipped files which can be written live simultaneously
 while this tool is streaming them out.</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Table.html" title="class in gngs/tools">
                                    Table
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="Toolbase9.html" title="class in gngs/tools">
                                    Toolbase9
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="TrimBarcode10X.html" title="class in gngs/tools">
                                    TrimBarcode10X
                                </a></strong>
                            </td>
                            <td>Utility to filter 10X barcode off the R1 FASTQ</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCFMergeSource.html" title="class in gngs/tools">
                                    VCFMergeSource
                                </a></strong>
                            </td>
                            <td>Utility class for tracking state of a specific VCF in the merge process</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCFtoDB.html" title="class in gngs/tools">
                                    VCFtoDB
                                </a></strong>
                            </td>
                            <td>Utility to import variants to SQLite database rows</td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VCFtoPed.html" title="class in gngs/tools">
                                    VCFtoPed
                                </a></strong>
                            </td>
                            <td>Simple utility to extract sample id and sex from VCF file, 
 and then write that information as a PED file. </td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VInfo.html" title="class in gngs/tools">
                                    VInfo
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>
<tr class="altColor">
                            <td class="colOne">
                                <strong><a href="VariantInfo.html" title="class in gngs/tools">
                                    VariantInfo
                                </a></strong>
                            </td>
                            <td></td>
                        </tr>

                    </tbody>
                </table>
            </li>
            </ul>
        </div>
    






</div>

<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
    <!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
    <!--   -->
</a>
    <ul class="navList" title="Navigation">
        <li><a href="overview-summary.html">Overview</a></li>
        <li class="navBarCell1Rev">Package</li>
        <li>Class</li>
        <li><a href="deprecated-list.html">Deprecated</a></li>
        <li><a href="index-all.html">Index</a></li>
        <li><a href="help-doc.html">Help</a></li>
    </ul>
</div>
<div class="subNav">
    <div>
        <ul class="navList">
            <li><a href="index.html?deprecated-list.html" target="_top">Frames</a></li>
            <li><a href="deprecated-list.html" target="_top">No Frames</a></li>
        </ul>
    </div>

    <div class="aboutLanguage"><em>Groovy Documentation</em></div>
    <a name="skip-navbar_bottom">
        <!--   -->
    </a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

</body>
</html>
